# Per-_and_polyfluoroalkyl_substances_PFAS_as_enviro

**Source PDF**: Per-_and_polyfluoroalkyl_substances_PFAS_as_enviro.pdf

---

## Page 1

[image]
View PDF VersionPrevious ArticleNext Article
[image]
[image] Open Access Article
[image] This Open Access Article is licensed under a Creative
Commons Attribution-Non Commercial 3.0 Unported Licence
DOI: 10.1039/D4VA00359D (Paper) Environ. Sci.: Adv., 2025, 4,
1444-1476
Per- and polyfluoroalkyl
substances (PFAS) as
environmental drivers of
antimicrobial resistance:
insights from genome
sequences of Klebsiella
grimontii and Citrobacter
braakii isolated from
contaminated soil†
Matteo Calcagnile [image]*a, Andrea Giuliano [image]bc, Maurizio
Salvatore Tredici a, Davide Gualandris [image]d, Davide Rotondo
[image]d, Antonio Calisi [image]d, Chiara Leo [image]e, Margherita
Martelli [image]e, Anna Rocchi [image]e, Knud Erik Klint f,
Francesco Dondero‡ [image]d and Pietro Alifano‡ [image]a
aDepartment of Experimental Medicine (DiMes), University of
Salento, Lecce, 73100, Italy. E-mail:
matteo.calcagnile@unisalento.it
bDepartment of Biological and Environmental Sciences and

## Page 2

Technologies (DiSTeBA), University of Salento, Lecce, 73100, Italy
cDepartment of Medical Biotechnology, University of Siena, Siena,
53100, Italy
dDepartment of Science and Technological Innovation (DISIT),
University of Eastern Piedmont “Amedeo Avogadro”, Alessandria,
15121, Italy
ePolo d'Innovazione di Genomica Genetica e Biologia SRL, NGS &
Bioinformatic Laboratory, 53100, Siena, Italy
fGeo, Maglebjergvej, 2800 Kgs. Lyngby, Copenhagen, Denmark
Received 7th October 2024 , Accepted 3rd July 2025
First published on 4th July 2025
Abstract
Per- and polyfluoroalkyl substances (PFAS) are man-made
chemicals widely used for industrial applications since the 1940s.
PFAS are extremely persistent in the environment, to the extent that
they have earned the reputation of ‘forever chemicals’. There is
growing evidence that PFAS have a significant impact on the
biodiversity, composition, and activity of microbial communities. In
this study, we hypothesized that these compounds may increase the
abundance of antibiotic-resistant bacteria. To investigate this
hypothesis, we employed Winogradsky columns to study the
microbial community's response to PFAS-contaminated soil from the
Albäck fire drill site (Trelleborg, Sweden). Column amendment with
a high amount of perfluorooctanoic acid (PFOA) led to selective
growth, in the aqueous phase of the columns, of Klebsiella grimontii
and Citrobacter braakii, two emerging opportunistic facultative
anaerobic pathogens. Whole-genome sequencing of K. grimontii
Tre-B and C. braakii Tre-T isolates revealed numerous antibiotic
resistance genes (ARGs), with a notable prevalence of resistance to
fluoroquinolones. Among these genes are those encoding multidrug
efflux systems that confer resistance to a wide range of toxic
compounds such as antibiotics, surfactants, dyes, detergents, and
disinfectants. Both strains contain a large set of features involved in
the degradation of aromatic and halogenated compounds, and other
recalcitrant chemicals. K. grimontii Tre-B is characterized by the

## Page 3

presence of an IncR-group plasmid (named pKGTreB) containing
many genes involved in resistance to arsenic, copper, mercury, and
silver. This strain also contains a choline utilization (cut) bacterial
microcompartment (BMC) locus, which has been implicated in
various human diseases as a source of trimethylamine (TMA).
Understanding the genomes of these two bacterial strains provides
insights into the molecular mechanisms responsible for their
pathogenicity, antibiotic resistance, resistance to biocides, and
heavy metal tolerance. In this study we also show that when the
two bacteria were grown with PFOA, their resistance to certain
aminoglycosides, fluoroquinolones and macrolides increased, and
we found that transcript levels of the kpnF, kpnG, adeF, and oqxA
antibiotic-resistance genes of K. grimontii Tre-B increased as a
function of PFOA concentration, whereas acrA was upregulated
only at low PFOA concentrations. These results indicate that PFOA,
in addition to selecting specific groups of bacteria, may increase
antibiotic resistance through upregulation of specific antibiotic
resistance genes and suggest that these genes may also be involved
in bacterial resistance to PFAS. Through the exploration of these
mechanisms, we can gain valuable insights into how environmental
pollutants, such as PFAS and other contaminants, may contribute to
the development of antimicrobial resistance.

## Page 4

Environmental significance
Per- and polyfluoroalkyl substances (PFAS), persistent
environmental pollutants, disrupt soil microbial communities.
PFAS accumulation in soil affects microbial diversity and function.
Soil microorganisms play a crucial role in nutrient cycling,
decomposition of organic matter, and maintaining overall soil
stability. This study examined PFAS-contaminated soil using
microcosms contaminated with perfluorooctanoic acid (PFOA) and
identified Klebsiella grimontii Tre-B and Citrobacter braakii Tre-T
as dominant bacteria. Both strains carried antibiotic resistance
genes and virulence factors, suggesting PFAS contamination may
promote the proliferation of potentially harmful, drug-resistant
microbes. In addition, PFOA exposure causes an increase in the
transcription levels of antibiotic-resistance genes of K. grimontii
Tre-T and C. braakii Tre-B. This finding suggests a possible
environmental mechanism by which PFAS can influence public
health.
Introduction
Per and polyfluoroalkyl substances (PFAS) are man-made chemicals
that comprise a large group of compounds that have an alkyl chain
backbone, typically 4 to 16 carbon atoms in length, and a
functional moiety (primarily carboxylate, sulfonate, or
phosphonate).1 The two most widely known PFAS contain an eightcarbon backbone, including perfluorooctanoic acid (PFOA) and
perfluorooctane sulfonate (PFOS). PFAS, introduced in the latter
half of the 20th century, have found extensive applications in
hundreds of industrial and consumer products.2 These applications
span from water and stain-resistant coatings for textiles, leather,
upholstery, and carpets to oil-resistant coatings for food-contactapproved paper. PFAS have also been used in various other
industries, including chromium electroplating as mist suppressants,
as surfactants in electronic etching baths, as photographic
emulsifiers, in aviation hydraulic fluids, and in the manufacturing
of paints, adhesives, waxes, polishes and fire-fighting foams.3–5 The

## Page 5

industrial success of PFAS, which resulted in an estimated annual
production of thousands of tons at the turn of the century, can be
attributed to their specific chemical and physical properties that
make them ideal surfactants.4 Additionally, the prevailing notion of
these substances being biologically inert significantly contributed to
their widespread use. However, recent studies have revealed that
these compounds are extraordinarily persistent in the environment,
earning them the moniker ‘forever chemicals.’ Furthermore, they
have been shown to exert various biological effects, with adverse
consequences for human health and ecosystems on a global
scale.4,6–15 For these reasons over the past decade, PFASs have
been increasingly regulated. Currently, a few PFAS congeners and
their salts or precursors are listed in the international Stockholm
Convention under Annex A-elimination (PFHxS, PFOA) or Brestriction (PFOS), so their production and use is banned or
restricted worldwide. In view of the regulations EU2020/784 and
2019/1021 in Europe PFOA should only be present in real life as a
trace contaminant while the production and use of PFOS is still
legal in a single industrial application, as a mist suppressant for
non-decorative hard chrome plating (VI) in closed loop systems.
C9-14 chain PFCA (and their related products) are restricted under
REACH in the EU/EEA from February 2023. Despite these legal
barriers, there are thousands of PFAS hotspots. ‘The Forever
Pollution Project’ website (2023)16 has estimated there to be
approximately 17[image]000 confirmed sites globally in Europe,
along with an additional 21[image]000 presumptive contamination
sites resulting from current or past industrial activities. These sites
are particularly prevalent around fluoropolymer plants but also
military and drill sites as well as airports due to the extensive use of
Aqueous Film Forming Foams (AFFFs) for fire extinguishing.
PFAS are generally resistant to microbial degradation,17 and there
is a growing body of evidence that PFOA and PFOS have a profound
impact on the structure and function of microbial communities in
diverse ecosystems.18–24 On the other hand, there is little evidence
on the real ability of microorganisms to effectively metabolize these
compounds, although some models on the catabolism of these
substances have been proposed.15,25 PFAS contain high-energy
carbon-fluorine bonds that occur very rarely in microbial chemistry
and most importantly, the end-product of biodegradation, fluoride,

## Page 6

can be very toxic to microorganisms.26 Although some microbial
metabolism with minimal PFAS degradation (defluorination)
activity such as the anaerobic Acidimicrobium sp A6 strain, and
Pseudomonades with bioaccumulation capabilities have been
described (see Shahsavari et al., 2021 (ref. 27)), information
regarding the mode of action is scarce and bioremediation for
environmental cleanup should not be deemed a practical option at
present.25 The impact of PFAS on the biodiversity, composition and
activity of microbial communities has aroused particular concern.
Indeed, the structure and correct functioning of microbial
communities are crucial in the balance of biogeochemical cycles,
pollutant decomposition, chemical transformation, food chain.28–30
There is substantial evidence indicating that prolonged exposure to
PFAS in soils, sediments, and vadose regions results in a marked
reduction in biodiversity.31,32 Additionally, this exposure tends to
favor the enrichment of specific bacterial phyla, notably
Proteobacteria, Acidobacteria, and Actinobacteria, which exhibit
higher resistance to PFAS compared to other phyla.18,22,33 This is
possibly due to a different architecture of cell wall (i.e., negativelycharged outer membrane in Proteobacteria), or a higher ability to
cope with oxidative damage and/or DNA damage, or also an ability
to extrude PFAS from the cells or immobilize these compounds in a
biofilm.
Interestingly, experiments in microcosms have recently revealed
that exposure to PFOA may significantly increase the abundance of
antibiotic resistance genes (ARGs) and human bacterial pathogens
(HBPs) raising further alarm for human health.34 Studies conducted
on conjugative strains of E. coli harboring the RP4 plasmid show
that the spread of ARGs in PFAS-polluted environments may be due
to the ability of PFASs to promote conjugative transfer of the ARG
plasmid as a consequence of inducing oxidative stress, increasing
cell membrane permeability, and stimulating excretion of
extracellular polymeric substances that promote conjugative
transfer.35,36 PFAS have been also shown to increase transformation
frequencies in Acinetobacter baylyi, a naturally competent
bacterium commonly found in aquatic environment, thereby
contributing to the spread of plasmid-borne antibiotic resistance
genes.37 Mechanistically, this increase in transformation
frequencies was imputed to increased cell envelope permeability,

## Page 7

biofilm formation, reactive oxygen species production, and
upregulation of DNA uptake genes.37 In addition, PFOA and PFOS
have been shown to promote long-term plasmid stability and induce
the expression of ARGs.37
The evidence that PFAS contamination may act as a driver for
selection of environmental ARBs and promote the spread of ARGs is
particularly worrisome as, in recent years, the spread of multidrugresistant bacterial infections has raised global concerns.38–40 The
World Health Organization41 (WHO, 2023) and the European
Commission42,43(EC, 2023) identified antimicrobial resistance
(AMR) as a transboundary health threat – a One Health concern –
encompassing human health, animal health, plant health, and
environmental aspects, with impacts on food and nutrition security,
economic development, and equity within societies.44–46 The
spread of AMR appears to be linked to factors associated with
climate change and chemical contamination, as indicated by recent
studies.47–49 Therefore, it is imperative to gain a comprehensive
understanding of the underlying mechanisms and drivers of
antimicrobial resistance in order to effectively tackle it.
In this research, we present a comprehensive analysis of the whole
genome sequences, obtained with a sequencing depth of 200×,
along with pertinent traits and characteristics of two bacterial
strains, namely Klebsiella grimontii and Citrobacter braakii. These
strains were isolated from microcosm experiments prepared using
soil samples obtained from the Albäck fire drill site in Trelleborg,
Sweden, known for its residual contamination by PFAS congeners
originating from heavy and light Aqueous Film-Forming Foams
(AFFFs). This investigation provides insights into the genomic and
functional attributes of these bacterial isolates, shedding light on
their potential roles in the context of PFAS-contaminated
environments. Characterized by abundance of antibiotic, surfactant,
dye, detergent, disinfectant, and heavy metal resistance genes, K.
grimontii and C. braakii, both opportunistic human pathogens, offer
an avenue for future research. This exploration may shed light, even
at a mechanistic level, on potential links between PFAS
contamination and the global spread of bacterial and antibiotic
resistance.

## Page 8

Materials and methods
Site characterization and sampling procedure of the
Trelleborg site
The study area is the old Albäck landfill in Trelleborg, Sweden (Fig.
1). This former unlined domestic/industrial landfill has been
utilized as a Fire Drill Site (FDS) for many years and has a
significant history of Aqueous Film Forming Foam (AFFF)
contamination in the underground water and soil.
[image]
Fig. 1 Location of the Trelleborg site and position of the
monitoring wells B1–B7 and B1B (Fig. 2) and existing
wells, W1–W3. Cross section A–A′ was constructed for
visualizing the geological settings on the site (Fig. 3).
Initially, 7 monitoring wells were drilled (B1–B7) for geological/
chemical characterization. Soil and water sampling in and outside
the firefighting site were completed in June 2022. In September
2022 a supplementary borehole B1B was drilled to 10 m depth
immediately adjacent to the concrete floor that firefighters used as
an exercise area and for washing tools from foam (Fig. 1). The
drilling was performed using a hydraulic drill equipped with a 100
mm rotary steel auger, and a steel casing was installed during the
drilling. Samples were described directly on the steel auger in 2 m
sections. Then samples were collected for each 50 cm, carefully
using steel tools cleaned with Ethanol between each sampling. The
samples were stored in Rilsan plastic bags, which are diffusion-tight
and PFAS-controlled by Eurofins, and kept cool in cooling boxes at
4 °C in the dark.
For the purposes of this study, the sample obtained at a depth of 7.5
m, consisting of the peaty organic soil, was selected for analysis.
Upon completion of the drilling process, water from the subsurface
aquifer was sampled using a sand filter and polypropylene plastic
pipes.
After 16 hours, a newly installed submersible pump, equipped with
a polypropylene plastic tube, was lowered into the well, and water

## Page 9

was brought up to the surface for sampling purposes. All pumps and
tubes have been tested for PFAS emission previously, and were
accepted for sampling PFAS infested water samples.
Soil and water samples were placed in appropriate 100 mL
containers made of polystyrene and high-density polyethylene
(HDPE), respectively, certified as PFAS-free by the supplier Eurofins
Laboratories Denmark. Subsequently, the samples were dispatched
to the same laboratory for PFAS analysis.
PFAS analysis
PFAS analysis (sum of 17 congeners) in soil samples were carried
out by means of LC-MS/MS according to DIN 38414-14. The
following congeners were evaluated: PFBA (perfluorobutanoic acid),
PFBS (perfluorobutane sulfonic acid), PFPeA (perfluoropentanoic
acid), PFPeS (perfluoropentane sulfonic acid), PFHxA
(perfluorohexanoic acid), PFHxS (perfluorohexane sulphonic acid),
PFHpA (perfluoroheptanoic acid), PFHpS (perfluoroheptane sulfonic
acid), PFOA (perfluorooctanoic acid), PFOS (perfluorooctane
sulfonic acid), 6:2 FTS (fluorotelomer sulfonate), PFOSA
(perfluorooctane sulfonamide), PFNA (perfluorononanoic acid),
PFNS (perfluorononanesulfonic acid), PFDA (perfluorodecanoic
acid), PFDS (perfluorodecanesulphonic acid), PFUnDA
(perfluorodecanoic acid), PFUnDS (perfluorodecane sulphonic acid),
PFDoDA (perfluorodecanoic acid), PFDoDS (perfluorodecane
sulphonic acid), PFTrDA (perfluorotridecanoic acid), PFTrDS
(perfluorotridecane sulphonic acid). Limit of detection was 0.1 ppb
for all congeners except PFOA and PFOS, 0.05 ppb; PFNS, 0.2 ppb;
PFUnDS, PFDoDS, PFTrDS, 1 ppb. Determination of dry residue and
water content was carried out according by means of
thermogravimetric analysis according to the Swedish standard SSEN 12880.
PFAS analysis (sum of 22 congeners) in water samples were carried
out by means of LC-MS/MS according to DIN38407-42. PFBA
(perfluorobutanoic acid), PFBS (perfluorobutane sulfonic acid),
PFPeA (perfluoropentanoic acid), PFPeS (perfluoropentane sulfonic
acid), PFHxA (perfluorohexanoic acid), PFHxS (perfluorohexane
sulphonic acid), PFHpA (perfluoroheptanoic acid), PFHpS

## Page 10

(perfluoroheptane sulfonic acid), PFOA (perfluorooctanoic acid),
PFOS (perfluorooctane sulfonic acid), 6:2 FTS (fluorotelomer
sulfonate), PFOSA (perfluorooctane sulfonamide), PFNA
(perfluorononanoic acid), PFNS (perfluorononanesulfonic acid),
PFDA (perfluorodecanoic acid), PFDS (perfluorodecanesulphonic
acid), PFUnDA (perfluorodecanoic acid), PFUnDS (perfluorodecane
sulphonic acid), PFDoDA (perfluorodecanoic acid), PFDoDS
(perfluorodecane sulphonic acid), PFTrDA (perfluorotridecanoic
acid), PFTrDS (perfluorotridecane sulphonic acid). Limit of
detection was 0.3 ppt for all congeners except PFOS, 0.2 ppt; PFBA,
0.6 ppt; PFUnDS, PFDoDS, PFTrDS, 1 ppt.
Installation of Winogradsky columns and microbial
isolation
Two sets of Winogradsky columns (microcosms) were set up using
soil sampled 7.5 m deep at the PFAS-contaminated Trelleborg site
B1 (Fig. 1). The columns were structured, from top to bottom, in: (i)
an aqueous phase consisting of 25 mL of PFAS contaminated
groundwater taken from the same site described above; (ii) a first
layer of 50 mL consisting of 50 g of soil (“top layer”); (iii) a second
layer of 50 mL consisting of 50 g of soil, 0.25 g of CaCO3, 0.5 g of
Na2SO4 and 1 g of soy flour (“bottom layer”). In one set of columns
(PFOA columns), 5 mL of 40 mg per mL PFOA in 50% isopropanol
(VWR, Radnor, Pennsylvania, USA) (final concentration of PFOA: 2
mg mL−1; final concentration isopropanol: 2.5%) was added to the
upper and lower layers. In the second set of columns (CTL
columns), 5 mL of 50% isopropanol was added as control. The
columns were incubated for 2 months at room temperature, then
serial dilutions of the aqueous phase were plated onto LB agar
medium [NaCl 10 g, tryptone 10 g (BD Difco™ Bacto™, Franklin
Lakes, New Jersey, USA), yeast extract 5 g (BD Difco™ Bacto™,
Franklin Lakes, New Jersey, USA), agar 15 g (BD Difco™ Bacto™,
Franklin Lakes, New Jersey, USA), distilled water up to 1 L]
containing 2 mg per mL PFOA (stock solution: of 40 mg per mL
PFOA in 50% isopropanol) and incubated for 48 h at 28 °C. A
workflow of the sampling and analysis processes carried out on the
microcosms is provided in Fig. S1.† Unless otherwise specified, the
reagents were sourced from Sigma-Aldrich (Merck, Darmstadt,
Germany).

## Page 11

DNA extraction from bacterial isolates
The DNA was extracted as previously reported.50 Briefly, bacterial
isolates were grown in 250 mL flasks containing 50 mL LB broth.
The flasks were incubated at 37 °C and 180 rpm. When the
absorbance at 600 nm of the culture broth was 0.6, the growth was
stopped by incubating the cultures on ice. The bacteria were
collected using a centrifuge (Eppendorf, Hamburg, Germany) at
4000 rpm, 4 °C, and 30 min, and frozen for 24 h. The pellet was
resuspended in SET Buffer [75 mM NaCl, 25 mM EDTA, 20 mM
Tris–HCl pH 7.5], lysozyme was added to a final concentration of 1
mg mL−1 (w/v) and the samples were incubated for 60 min at 37
°C. After this time, proteinase K at a concentration of 0.5 mg mL−1
(w/v) and SDS (VWR, Radnor, Pennsylvania, USA) at 1% (v/v)
were added. Samples were incubated at 55 °C for 60 min. Total
nucleic acids were extracted by
phenol[image]:[image]chloroform[image]:[image]isoamylic alcohol
(25[image]:[image]24[image]:[image]1 [v/v/v]) and RNase A
(final concentration 15 μg mL−1 (w/v)) was used to remove RNA
(15 min at 37 °C). The nucleic acid extraction procedure was
repeated to eliminate RNase A. The DNA was precipitated by
adding cold ethanol and Na-acetate pH 7.3 M and incubating the
samples at −20 °C overnight. A centrifuge was carried out to
collect the pellet (10.000 rpm, 15 min, 4 °C), which was then
washed with 80% ethanol, left to dry at room temperature, and
resuspended in 100 μL of sterile water. The quality and the
concentration of DNA samples were assessed by electrophoresis
analysis and UV-spectrophotometry (NanoDrop®, ND-1000
Spectrophotometer, Thermo Fisher Scientific, Waltham,
Massachusetts, USA). Unless otherwise specified, the reagents were
sourced from Sigma-Aldrich (Merck, Darmstadt, Germany).
Screening of bacterial isolates by repetitive extragenic
palindromic sequence-based PCR (rep-PCR)
PFOA-resistant colonies were characterized at the molecular level
using the repetitive extragenic palindromic sequences-based
polymerase chain reaction (REP-PCR) method.50 The DNA was
extracted as previously reported and used as a template to amplify
the REP sequence using the BoxA1-R primer (5′-

## Page 12

CTACGGCAAGGCGACGCTGACG-3′). The PCR reaction was
implemented using 1 μL of template DNA, 1.25 μL of BoxA1-R
primer, 2.5 μL of 10× buffer S (VWR, Radnor, Pennsylvania, USA),
2.5 μL of DMSO, 0.5 μL Taq DNA polymerase (VWR, Radnor,
Pennsylvania, USA) and sterile water to a final volume of 25 μL.
The PCR cycles used were the following: (1) initial denaturation (95
°C, 7 min); (2) denaturation (95 °C, 1 min); (3) annealing (52 °C, 1
min); (4) extension (65 °C, 8 min); (5) final extension (65 °C, 16
min). Steps 2, 3, and 4 were repeated 30 times. The amplicons were
analyzed by electrophoresis analysis (1% (w/v) agarose gel, 75 eV,
1× TBE buffer). The electrophoresis gel image was acquired using
the ChemiDoc Imaging System (BIO-RAD, Hercules, California,
USA). Unless otherwise specified, the reagents were sourced from
Sigma-Aldrich (Merck, Darmstadt, Germany).
Minimum inhibitory concentration (MIC) and
minimum bactericidal concentration (MBC) tests
Minimum inhibitory concentration (MIC) experiments were
performed as previously described.51E. coli strain FB8 was used as a
reference strain. This is an E. coli reference strain, also known as
GC2553, K12S (Luria) or UTH1038, deposited at the University of
Texas/Houston stock culture collection, genotype F-l-/lS, ftsR1, and
is a prototrophic strain.52–54 The MIC of ampicillin, metals, and
PFOA were determined using 24-well plates and LB broth. Briefly,
the wells of the multiwell plates (CytoOne®, Milan Italy) were
filled with 1 mL of LB broth inoculated with bacterial isolates (one
colony in 10 mL of LB broth). 2 mL of inoculated broth was added
to the first well. One volume of the molecule to be tested was added
to the first well so that the final concentration was 500 μg mL−1
(w/v) for ampicillin; 100 mM for chromium (Cr), aluminum (Al),
cobalt (Co), nickel (Ni), copper (Cu) zinc (Zn) and 320 mM for
silver (Ag). Subsequently, serial dilutions 1 to 2 were carried out.
For the metals Cr, Al, Co, Ni, Cu, and Zn 10 dilutions were carried
out (from 100 mM to 0.2 mM). For ampicillin, 11 dilutions were
carried out (from 500 μg mL−1 to 0.5 μg mL−1). E. coli ATCC 25922
(NCTC 12241, CIP 76.24, DSM 1103, CCUG 17620, CECT 434), an
Eucast routine quality control strain, was used to assess the
accuracy and reliability of antibiotic susceptibility testing. For
silver, 14 dilutions were carried out (from 320 mM to 0.04 mM).

## Page 13

The compounds used to perform the MICs of the metals were the
following: KCr(SO4)2·12H2O, Al2(SO4)3·8H2O, CoCl2·6H2O,
NiCl2·6H2O, CuCl2·2H2O, ZnCl2, AgNO3. The MIC for fluoride was
performed similarly to the previous ones and sodium fluoride (NaF)
was used. The NaF powder was solubilized in LB at a concentration
of 1 M. 2 mL of this solution was used to fill the 1st well.
Subsequently, serial 1 to 2 dilutions were made using LB as a
diluent. The MIC for PFOA was performed by filling 10 wells of the
multiwell plate with inoculated LB broth and adding a volume of
PFOA solution so that the final concentration of the compound in
the wells was: 10 mg mL−1 for the 1st well, 9 mg mL−1 for second
well, 8 mg mL−1 for third well, 7 mg mL−1 for fourth well, 6 mg
mL−1 for fifth well, 5 mg mL−1 for sixth well, 4 mg mL−1 for
seventh well, 3 mg mL−1 for the eighth well, 2 mg mL−1 for the
ninth well, and 1 mg mL−1 for the tenth well. The PFOA stock
solution used in this experiment had a concentration of 200 mg
mL−1 (w/v). PFOA was solubilized in a mixture of 50% isopropyl
alcohol in water. To control and normalize the MIC data, the
solubilization solution (50% isopropyl alcohol in water) was used to
set up a MIC experiment without the PFOA. In each MIC experiment
setup, one well containing only LB (positive control) was foreseen.
The experiments were repeated three times. In addition, the
minimal bactericidal concentration (MBC) was measured after the
MIC. For each experiment, a 100 μL aliquot was taken from each
well and washed 2 times with sterile LB. The samples were
centrifuged (3000 rpm, 5 min) to collect the pellet. The pellet
obtained after the washings was resuspended in 100 μL of sterile
LB, and 10 μL of suspension was plated on LB agar. The plates
obtained in this way were incubated at 37 °C. After 24 h, growth
was observed. The MBC was considered the lowest concentration of
substance for which no growth was observed. Unless otherwise
specified, the reagents were sourced from Sigma-Aldrich (Merck,
Darmstadt, Germany).
Antimicrobial susceptibility of C. braakii Tre-T and K.
grimontii Tre-B in the presence of PFOA
The antimicrobial susceptibility of C. braakii Tre-T and K. grimontii
Tre-B was assessed by the Kirby–Bauer method following the
guidelines established by EUCAST v.15 (2025). Bacterial strains

## Page 14

were grown on Mueller–Hinton agar plates (BD Difco™ Bacto™,
Franklin Lakes, New Jersey, USA) supplemented with an
isopropanol-PFOA solution (final concentration of PFOA of 2 μg
mL−1, 20 μg mL−1, 200 μg mL−1, 2 mg mL−1) or isopropanol as a
control. The inoculum was standardized to approximately 1.5 ×
108 CFU mL−1. Sterile swabs were used to inoculate the surface of
the agar plates with each bacterial suspension. Then, antibiotic
disks (Liofilchem, Roseto degli Abruzzi, Italy) were placed onto the
agar surface using sterile forceps. Plates were incubated at 37 ± 1
°C for 18 hours. This method was used to determine the
susceptibility to the following antibiotics: ampicillin 10 μg
(AMP10), cefoperazone 30 μg (CAZ30), ceftazidime 30 μg (CFP30),
amikacin 30 μg (AK30), tobramycin 10 μg (TOB10), pefloxacin 5 μg
(PEF5), pipemidic acid 20 μg (PI20), azithromycin 15 μg (AZM15),
tetracycline 30 μg (TE30), and trimethoprim-sulfamethoxazol 25 μg
(SXT25). The quality of the antibiotic disks and agar plates was
assessed using E. coli ATCC 25922. Unless otherwise specified, the
reagents were sourced from Sigma-Aldrich (Merck, Darmstadt,
Germany).
Antibiotic resistance gene transcription levels in the
presence of PFOA (RT-qPCR)
K. grimontii Tre-B was grown in LB broth supplemented with PFOA
to measure the impact of PFOA on the transcription levels of the
antibiotic resistance genes. Briefly, K. grimontii Tre-B was
inoculated in 10 mL of LB broth and the culture was incubated at
37 °C and 180 rpm for 18 h. Then, 50 mL flasks were inoculated
1[image]:[image]100 and incubated under the same conditions for
5 h, when the bacterial culture reached an OD600nm of 0.6. These
flasks contained 10 mL of LB broth supplemented with an
isopropanol–PFOA solution (final concentration of PFOA of 2 μg
mL−1, 20 μg mL−1, 200 μg mL−1) or isopropanol as a control. After
the growth, bacterial cultures were centrifuged at 10[image]000
rpm for 1 minute, and then the supernatant was discarded. The
RNA was extracted using the Aurum™ Total RNA Mini Kit according
to the manufacturer's instructions and the extracts were quantified
using UV spectrophotometry (NanoDrop®, ND-1000 spectrophotometer). Reverse transcription and qPCR reactions were
performed using protocols, reagents, and instruments previously

## Page 15

described.55 All the experiments were repeated three times. The
primers used for this analysis were: acrA_F (5′-
GCGCTAACAGGATGTGACGAC-3′) and acrA_R (5′-
ACCTGAGGACGAACTTCCGC-3′), adeF_F (5′-
ATCACCGGATTAATCGCCATC-3′) and adeF_R (5′-
CAGCGACGGATTTCATGTAC-3′); oqxA_F (5′-
TACTCTCCGCGCTCCTCGTC-3′) and oqxA_R (5′-
CTCCTGACCGTCGGTGTAATTC-3′); kpnF_F (5′-
GCTGGCGCTGGCTATCGCGC-3′) and kpnF_R (5′-
GGCAATGGTCGCCGCGATGC-3′); kpnG_F (5′-
GCGTCACTTCGAAGAGACCG-3′) and kpnG_R (5′-
GTCTGGTCGAGGGTGACCAG-3′); and rpoB_F (5′-
ATGGTTTACTCCTATACCGAG-3′) and rpoB_R (5′-
CCTGAAGGGCAGTATGGTCTGG). Unless otherwise specified, the
reagents were sourced from BIO-RAD (Hercules, California, USA).
Statistical significance of the results was calculated using Student's
t-test.
16S rRNA metabarcoding
The aqueous phases and the layer of surface soil soaked in water
were collected from microcosms biostimulated or not with PFOA
(Fig. S1†). The water and soil mixtures were used to extract DNA
using the E.Z.N.A.® Soil DNA Kit (Omega Bio Tek, Norcross, GA,
USA). The quality and concentration of the extracts were measured
using Nanodrop® (ND-1000 spectro-photometer). The
amplification, sequencing (Illumina MiSeq, San Diego, California,
USA), and taxonomic assignment process were performed as
previously described.56 The Greengenes database (gg_13_5)
(https://greengenes.lbl.gov/) was used for taxonomic assignment.57
Whole genome sequencing: preparation of DNA
libraries and Illumina sequencing
Bacterial strains were cultured in 250 mL flasks containing 50 mL of
LB broth. The flasks were incubated at 37 °C and 180 rpm. The
growth of the bacteria was measured by monitoring the absorbance
at 600 nm (V-10 PLUS spectrophotometer, ONDA). When the
absorbance reached a value of 0.6, bacterial growth was stopped by
incubating the flasks on ice. DNA extraction was performed as

## Page 16

previously described. The DNA libraries were prepared according to
the Illumina DNA Prep kit following the manufacturer's instructions
(Illumina, San Diego, California, USA). Their quality was checked
using the Fragment Analyzer™ High Sensitivity Small Fragment
(Agilent Technologies, Santa Clara, California, USA) and Qubit® 4.0
Fluorometer (Thermo Fisher Scientific, Waltham, Massachusetts,
USA). Finally, the libraries were loaded onto an Illumina MiSeq
platform and sequenced with 2 × 150 bp paired-end run and V2
chemistry at the Polo GGB sequencing facility (Polo d'Innovazione
di Genomica, Genetica e Biologia, Siena, Italy).
De novo genome assembly
The quality of the demultiplexed samples was checked using the
program FastQC v0.11.9.58 The reads were then de novo assembled
using the assembler MaSuRCA v5.0.2.59 The completeness and
contiguity of the final assembly was evaluated using BUSCO
(v5.2.2, with database bacteria_odb10)60 based on the
evolutionarily informed expectations of gene content from nearuniversal single-copy orthologs. The quality of the assembly was
assessed using the program QUAST 5.0.2.61
Genomic data analysis
The tools Prokka v 1.14.6 (ref. 62) and DFAST v1.2.18 (ref. 63)
were applied to identify main features of the sample genomes,
including architecture, composition and functions. The taxonomic
identification was performed using the program DFAST_QC63 using
as a comparison the Genome Taxonomy Database (GTDB).
The program Resistance Gene Identifier v6.0.3 (ref. 64) was used to
identify resistance genes by comparing the assembled genomes to
the Comprehensive Antibiotic Resistance Database (CARD). The
tools PathogenFinder65,66 and VirulenceFinder-2.0 (ref. 67) were
further used to detect other virulence factors along the assembled
Tre-B and Tre-T genomes. The online tool AntiSMASH 7.0 (ref. 68)
was used to identify clusters containing genes involved in the
secondary metabolism. Finally, the presence of mobile genetic
elements in the genomes was analyzed using the
MobileElementFinder tool.69

## Page 17

Co-occurrence and genome neighborhood analysis
The Enzymatic Similarity Tool (EFI-EST) was used to calculate the
co-occurrence of genes close to the cutC (OKNGJBID_00805), eutB
(OKNGJBID_01616) and pduE (OKNGJBID_04294) genes of K.
grimontii Tre-B and the eutB (LBFIJIGF_02983) and pduE
(LBFIJIGF_03409) genes of C. braakii Tre-T.70,71 Initially using the
protein sequences as input, a sequence similarity network (SSN)
was generated (option A: Blast), with an alignment score cutoff of
20. The resulting full network was submitted to EFI-GNT to
generate a Genome Neighborhood Networks (GNNs) useful to
analyze the co-occurrence and neighboring of Pfam families.70,71
This tool also produces Genome Neighborhood Diagrams (GNDs)
which have been used to visualize conserved genes.70,71 SSN and
GNN were analyzed with Cytoscape (v 3.10.0).72 The composition
of the generated datasets was manually analyzed to identify the
most numerous bacterial genera showing conservation in the
genomic region under analysis. EFI-EST and EFI-GNT analysis was
also performed using the putative fluoride ion transporter (crcB)
gene of both bacterial isolates: K. grimontii Tre-B
(OKNGJBID_03358) and C. braakii Tre-T (LBFIJIGF_01727). The
genes from cut, eut and pdu loci were also used to confirm the
annotation using BLAST and as reference the Salmonella enterica
subsp. enterica serovar Typhimurium str. LT2 (NCBI reference
sequence: NC_003197.2) and Escherichia coli 536 complete
sequences (NCBI reference sequence: NC_008253.1). The spread and
conservation of the locus cut in K. grimontii specie were analyzed
manually using the NCBI nucleotide database. Gene maps were
drawn using Illustrator for Biological Sequences 2.0.73
Results
Geological setting of the Trelleborg site and analysis
of PFAS
A sampling campaign was conducted at the firefighting site situated
near the old Albäck landfill in Sweden, which serves as one of the
demonstration units within the European project SCENARIOS
(https://scenarios-project.eu) aimed at assessing the impact of PFAS

## Page 18

congeners on the environment. Central to our inquiry is the detailed
depiction presented in Fig. 1, where borehole B1 is positioned
strategically directly above the elevated exercise area, which has
been exposed for decades to 3 M light water AFFF containing PFOS
and other PFAS precursors. This firefighting foam is specifically
designed for combating hydrocarbon and polar solvent fuel fires.
Various samples were obtained from depths ranging from 0 m to 9
m.
Based on the descriptions, a lithological log was constructed (Fig.
2). The log indicates that the upper 1.0 m consists of a sandy matrix
with a dark brown fill containing large amounts of plastic waste.
From 1.0 m onwards, the soil becomes richer in clay, and by 1.5 m,
waste makes up more than 50% of the volume, with the fill
primarily consisting of plastic, concrete, glass, and wood. At a depth
of 4.0 m, the waste becomes sandier, with remnants of tires, rock
wool, and even more wood, plastic, and glass. For these reasons, it
is important to consider that there may be interactions between
PFAS and co-contaminants. It has been demonstrated that certain
contaminants, such as microplastics, can serve as long-range
transport media for PFAS.74,75 Additionally, organic solvents can
impede the chemical transformation of PFAS,76 while heavy metals
promote their adsorption into the soil.77 At 6.8 m below ground
surface (b.g.s.), the drill reaches the groundwater table, and at 7.0
m, the waste layer ends. Here, the natural soil begins, consisting of
approximately 70 cm of dark brown organic peat. Beneath this peat
layer, there is a 20 cm thick layer of laminated freshwater clay,
poor in calcium carbonate (CaCO3), containing strings of silt and
fine sand. At 7.9 m b.g.s., a nearly 2 m thick layer of fine to
medium grey meltwater sand, rich in CaCO3, appears. Finally, from
9.7 m to the bottom at 10 m b.g.s., the drill encounters sandy,
gravelly clay till. This till is grey, indicating reduced conditions, and
firm, classified as basal clay till deposited under a transgressing
glacier during the last stage of the Weichselian glaciation, around
13[image]000 BP. In Fig. 3, a cross-section from the Albäck River to
the firefighting facility at well B1B is presented. The cross-section
reveals that the area consists of a 55-million-year-old Danien
limestone basement, located between 7 and 4 meters below the
reference elevation, according to the Danish Vertical Reference
1990 (DVR 90). During glacial times, glaciers eroded the limestone,

## Page 19

and basal clay till was deposited directly on top of it. After the
glacier's retreat, a meltwater river eroded the clay till and deposited
meltwater sand in an ancient riverbed. In the post-glacial period
(less than 11[image]000 BP), freshwater sand and clay were
deposited on top of the meltwater sediments in the riverbed. Over
time, vegetation accumulated in this meandering river system,
forming layers of peat, clay, and sand. In modern times, human
activities have led to waste being deposited in parts of this old river
system. Eventually, parts of the waste deposit site were reclaimed,
and a firefighting facility was constructed on top of the former
landfill. It may be concluded that the meltwater sand is
hydraulically well connected throughout the area and that the
freshwatersand is following channels in a classic meander
riversystem with creation of oxbowlakes and lagunes with
freshwaterclay/laminated silt/fine sand, that grows into peat bogs.
Accordingly the infiltrating land fill percolate may potentially
spread to the Albäck River downgradient to the west and south.
[image]
Fig. 2 Lithological log from well B1B. The log exhibits a
detailed description of the soil samples collected
including grain-size distribution and textural properties
of the samples. Specific samples for this study were
collected from the 70 cm thick peat layer between 7 and
7.7 m b.g.s.
[image]
Fig. 3 Cross-section and geological settings from the
Albech River to the landfill site.
Table 1 displays the results of chemical analyses, detailing the
concentrations of PFAS observed at the 7.5 m sampling height of
borehole B1. Despite the concentration of PFAS found in this
specific layer was not particularly high, i.e. sum of PFAS (17
congeners, see Materials and methods) 56.97 ppb, this depth
harbors a rich, dark-brown organic peat layer located directly
beneath the water table, where percolates from the upper strata
accumulate. Below this organic layer, the borehole's geological

## Page 20

composition comprises laminated clay with silt interspersed with
sand stringers, which serves to support the peat layer. Notably, this
clay layer, primarily composed of impermeable clay, is instrumental
in accumulating PFAS present in leachate or groundwater due to its
impermeability. Additionally, the distinct layer features grey
meltwater sand and exhibits a low concentration of calcium
carbonate (CaCO3).
Table 1 Results of chemical analyses, detailing the concentrations
of PFAS observed at the 7.5 m sampling height of borehole B1
Parameters
Unit
B1–7.5
m
Dry weight
%
78.2
PFBA (perfluorbutanoicacid)
μg per kg
dw
0.23
PFBS (perfluorbutansulfonicacid)
μg per kg
dw
0.41
PFPeA (perfluorpentanoicacid)
μg per kg
dw
1.1
PFPeS (perfluorpentansulfonicacid)
μg per kg
dw
0.38
PFHxA (perfluorhexanoic acid)
μg per kg
dw
0.96
PFHxS (perfluorhexansulfonoic acid)
μg per kg
dw
4.9
PFHpA (perfluorheptanoic acid)
μg per kg
dw
0.29
PFHpS (perfluorheptansulfonoic acid)
μg per kg
dw
0.55
PFOA (perfluoroctanoic acid)
μg per kg
dw
0.68
PFOS (perfluoroctansulfonoic acid)
μg per kg
dw
45
6[image]:[image]2 FTS
(fluortelomersulfonat)
μg per kg
dw
1.5

## Page 21

PFOSA (perfluoroctansulfonamid)
μg per kg
dw
0.33
PFNA (perfluornonanoic acid)
μg per kg
dw
<0.10
PFNS (perfluornonansulfonoic acid)
μg per kg
dw
<0.20
PFDA (perfluordekanoic acid)
μg per kg
dw
0.31
PFDS (perflordekanesulfonoic acid)
μg per kg
dw
0.18
PFUnDA (perfluorundecanoic acid)
μg per kg
dw
0.15
PFUnDS (perfluorundecansulfonoic acid)
μg per kg
dw
<1.0
PFDoDA (perfluordodecanoic acid)
μg per kg
dw
0.24
PFDoDS (perfluordodekansulfonoic acid)
μg per kg
dw
<1.0
PFTrDA (perfluortridekanoic acid)
μg per kg
dw
<0.10
PFTrDS (perfluortridekansulfonoic acid)
μg per kg
dw
<1.0
Sum af PFAS 4 excl. LOQ
μg per kg
dw
51
Sum af PFAS excl. LOQ
μg per kg
dw
57
Experimental setup of Winogradsky columns,
characterization of microbial communities, and
isolation of bacteria
Winogradsky columns78 were used as microcosms to challenge
microbial communities from the PFAS-contaminated Trelleborg site
B1 (Fig. 1–3 and Table 1) in an attempt to isolate PFAS-resistant

## Page 22

microorganisms and study the underlying mechanisms of resistance.
Two sets of Winogradsky columns (without or with 2 mg per mL
PFOA) were set up using soil sampled at 7.5 m of depth, and the
columns were structured in an aqueous phase, a “top layer” and a
“bottom layer” as described in Materials and methods. The columns
were incubated for 2 months at room temperature. Then the “top
layer” and the aqueous phase of the columns were analyzed by
culture-based and culture-independent approaches in this study.
Serial dilutions of the aqueous phase of the columns and native
control soil were plated onto LB agar plates containing 2 mg per mL
PFOA and in control LB agar plates without PFOA to determine
microbial counts (Fig. 4). Bacteria resistant to PFOA were found in
both the aqueous phase and the “top layer” of the PFOA-containing
Winogradsky columns. In contrast, no bacteria resistant to PFOA
were isolated from the “bottom layer” of the columns. In particular,
in the aqueous phase of the PFOA-containing Winogradsky columns,
nearly identical microbial counts (approximately 7 × 106 CFU
mL−1) were detected on PFOA-containing and PFOA-free LB agar
plates (Fig. 4A and B, green). In the aqueous phase of the PFOA-free
Winogradsky columns, slightly lower microbial counts
(approximately 2.5 × 106 CFU mL−1) were observed on PFOA-free
LB agar plates, whereas no colonies were observed on PFOAcontaining LB agar plates (Fig. 4A and B, violet). In native soil,
microbial counts of 3 × 105 CFU mL−1 were recorded on PFOA-free
LB agar plates and no colonies on PFOA-containing LB agar plates
(Fig. 4A and B, grey).
[image]
Fig. 4 CFU quantification and molecular characterization
of the isolates. (A and B) The bacterial load in the
microcosms was analyzed using the serial dilution
method and LB agar medium (A) or LB agar medium
supplemented with PFOA (2 mg mL−1) (B). (C) The two
bacterial morphotypes isolated using PFOA-enriched LB
agar were counted separately to estimate the abundance
of Tre-B morphotype versus Tre-T morphotype. (D) 5
colonies of each morphotype were analyzed by rep-PCR
which demonstrated that the colonies were groupable in
two genomic fingerprints. MWM = molecular weight
marker.

## Page 23

PFOA-containing LB agar plates were then used to isolate PFOAresistant microorganisms. Colonies exhibited two distinct
morphotypes, designated Tre-B and Tre-T, and a visual analysis of
each morphotype showed a clear prevalence of Tre-T
(approximately 7 × 106 CFU mL−1) over Tre-B (approximately 2 ×
105 CFU mL−1) (Fig. 4C). Five colonies for each morphotype were
then analyzed by rep-PCR (Fig. 4D). The results demonstrated
identical profiles within each morphotype and different profiles
between the two morphotypes suggesting the presence of two
distinct taxa. Therefore, two bacterial isolates, one for each
morphotype, were subjected to whole-genome sequencing. MIC
experiments showed that the Tre-B isolate was slightly more
resistant than the Tre-T isolate to PFOA and the reference E. coli
strain FB8. MIC values were 8 mg mL−1 for the Tre-B isolate, 7 mg
mL−1 for the Tre-T isolate, and 6 mg mL−1 for E. coli FB8 (Fig.
S2†). In contrast, the MIC measured using fluoride (NaF) was the
same for K. grimontii Tre-B, C. braakii Tre-T, and E. coli FB8.
However, the isolates K. grimontii Tre-B and C. braakii showed an
MBC value of 1 M. This value for E. coli FB8 was half (0.5 M) (Fig.
S2†).
To characterize the structure of the microbial communities in the
PFOA-containing and PFOA-free Winogradsky columns, total DNA
was extracted from the aqueous phase and the “top layer” of the
columns and subject to 16S rRNA metabarcoding analysis (Fig. 5).
The bacterial communities were analyzed at three different
taxonomic levels: phyla, orders, and families.
[image]
Fig. 5 Composition of microbial communities in the
PFOA-containing and PFOA-free Winogradsky columns:
phyla level (A), order level (B), family level (C).
In the PFOA-free columns, the most representative phyla
(abundance >0.5%) at phylum level (Fig. 5A) were Firmicutes,
Proteobacteria, Bacteroidetes, Chloroflexi, Spirochaetes,
Actinobacteria, Synergistetes, Cloacimonetes, Armatimonadetes,
Euryarchaeota, Tenericutes, and Deinococcus/unclassified Thermus.
In the PFOA-containing columns, the most representative phyla

## Page 24

were Proteobacteria, Firmicutes, Actinobacteria, Bacteroidetes,
Chloroflexi, Cyanobacteria/Chloroplast, Acidobacteria,
Euryarchaeota, and Synergistetes. Interestingly, Proteobacteria were
much more abundant in the PFOA-containing columns than in the
PFOA-free columns (81.11% vs. 22.56%). Actinobacteria,
Cyanobacteria/Chloroplast, Acidobacteria, Planctomycetes,
Chlamydiae, Candidatus_Saccharibacteria, Parcubacteria, and
Atribacteria were also more abundant in the PFOA-containing
columns than in PFOA-free columns. In contrast, Firmicutes and
Bacteroidetes were more abundant in PFOA-free columns than in
PFOA-containing columns (42.22% vs. 9.70%; 16.52% vs. 1.41%).
Chloroflexi, Synergistetes, Spirochaetes, and Cloacimonetes were
also more abundant in the PFOA-free columns than in PFOAcontaining columns.
At order level (Fig. 5B), the majority of Proteobacteria in the PFOAcontaining columns were Enterobacteriales (79%), while this order
was much less represented in the PFOA-free columns (0.037%).
Rhizobiales, within the Proteobacteria phylum, were also more
abundant in the PFOA-containing columns. Conversely,
Caulobacterales, also belonging to the Proteobacteria phylum, was
more abundant in the PFOA-free columns. Among Firmicutes, the
orders Clostridiales, Bacillales, and Selenomonadales were much
more represented abundant in the PFOA-free columns than in the
PFOA-containing columns. In contrast, the order Lactobacillales was
more abundant in the PFOA-containing columns (2%) than in the
PFOA-free columns (0.43%). Other orders that were more abundant
in the PFOA-free columns compared to the PFOA-containing
columns were Bacteroidales, Anaerolineales, Desulfovibrionales,
Desulfobacterales, Flavobacteriales, and Erysipelotrichales.
At family level (Fig. 5C), the most abundant family in the PFOAcontaining columns was Enterobacteriaceae (79%). The relative
abundance of this family was very low in the PFOA-free columns
(0.037%). Carnobacteriaceae also were more represented in the
PFOA-containing columns (1.87%) then in the PFOA-free columns
(0.34%). In contrast, the relative abundance of Lachnospiraceae
(phylum Firmicutes) was similar (about 2%) in both columns.
Conversely, other families were more represented in the PFOA-free
columns (>2%) than in PFOA-containing columns, including

## Page 25

Caulobacteraceae, Porphyromonadaceae, Anaerolineaceae,
Clostridiales_Incertae_Sedis_XI, Desulfovibrionaceae,
Ruminococcaceae, Planococcaceae, Gracilibacteraceae,
Desulfobulbaceae, Bacillaceae, Spirochaetaceae,
Acidaminococcaceae and Peptococcaceae.
The high relative abundance of Enterobacteriaceae in the PFOAcontaining columns was consistent with the culture-based analysis
that led to the isolation of two strains (Tre-B and Tre-T) of this
family.
Whole genome sequence and characterization of
Klebsiella grimontii strain Tre-B
A total of 7[image]348[image]974 reads were sequenced and the
genome assembly of strain Tre-B was represented by 36 contigs
with a N50 value of 456[image]983 bp and a N75 value of
248[image]966 bp; the largest contig was 841[image]358 bp. The
average coverage was >90-fold. The combined length was
5[image]886[image]764 bp with a G + C content of 55.9% (Table
2). Based on Genome Taxonomy Database, genome sequence of
strain Tre-B showed the highest identity (99.26%) with the that of
Klebsiella grimontii reference strain 06D021 (accession
GCA900200035.1). Rapid genome annotation was performed with
Prokka.62 Annotation features identified with Prokka v1.14.6 (ref.
62) includes 5354 DNA coding sequences (CDSs), 20 rRNAs, 81
tRNAs (Table 2). Annotation features identified with DFAST v5.0.2
(ref. 63) identified a total of 5384 CDSs, 9 rRNAs, 81 tRNAs and
one CRISPR locus (Table 2). According to these results, the
annotation was manually implemented, and results are reported in
Table S1.†
Table 2 Features of Klebsiella grimontii strain Tre-B and Citrobacter
braakii strain Tre-T genome sequences, and features of pKGTreB
plasmid revealed by MobileElementFinder. Number of CDSs, rRNAs,
and tRNAs show the two values resulted from Prokka/DFAST
annotations
Feature
Value (Tre-B) Value (Tre-T) Value (pKGTreB)
Total sequence
length (bp)
5[image]887[image]560
4[image]936[image]210
87[image]115

## Page 26

Number of
contigs
36
28
6
Largest contig
(bp)
841[image]3581[image]837[image]299
50[image]714
N 50 (bp)
456[image]983484[image]74750[image]714
Gap ratio (%)
0.452242
0.000000
0.000000
GC content (%)
55.9
52.2
51.9
Number of CDSs 5354/5384
4581/4563
88
Average protein
length
320.0
319.5
260.80
Coding ratio (%) 87.8
88.6
79.0
Number of
rRNAs
20/9
9/9
0
Number of
tRNAs
81/81
75/75
0
Number of
CRISPRs
1
1
0
IncR (position:
nt, reverse/
forward)
44[image]742–
44[image]992
ISEcl1 (position:
nt, reverse/
forward)
38[image]327–
39[image]662
forward
ISKpn34
(position: nt,
reverse/forward)
14[image]837–
16[image]025
reverse
IS903 (position:
nt, reverse/
forward)
1139–2195
forward
IS903 (position:
nt, reverse/
forward)
753–1805
forward
IS26 (position:
nt, reverse/
forward)
2–821 forward

## Page 27

K. grimontii is a newly identified species closely related to
Klebsiella oxytoca.79K. oxytoca has a chromosomally encoded βlactamase gene (blaOXY) that confers resistance to amino- and
carboxypenicillins.80 This gene diversified in parallel to
housekeeping genes in species closely related to K. oxytoca, and
variants blaOXY-1 to blaOXY-7 allowed to classify these closely related
bacteria into seven phylogenetic lineages named from Ko1 to Ko7.
K. oxytoca corresponds to phylogroup Ko2, while K. grimontii
corresponds to phylogroup Ko6.79 Resistance Gene Identifier64
predicted the presence of blaOXY-6-1 gene in strain Tre-B (Table 3)
confirming the correct assignment to the species K. grimontii.
Consistent with this result, MIC experiments showed that growth of
K. grimontii Tre-B was not inhibited by 500 mg per mL ampicillin
(Fig. S2†). Resistance Gene Identifier also predicted acquired
resistance to very wide range of antibiotics, disinfecting agents and
antiseptics (Tables 3 and S2†). In particular, a plethora of genes are
involved in resistance to fluoroquinolones, coding for different
antibiotic efflux pumps belonging to the major facilitator
superfamily (MFS) (KpnGH-TolC) or to the resistance-nodulationcell division (RND) superfamily (AdeFGH, OqxA, AcrAB-TolC) and
to MFS and RND antibiotic efflux pump regulators (Tables 3 and
S2†). It may be also noted the presence of a gene
(OKNGJBID03127) encoding methyl viologen (i.e., the herbicide
Paraquat) resistance protein SmvA, an MFS transporter.
Table 3 Acquired resistance to antimicrobial compounds in K.
grimontii Tre-B and C. braakii Tre-T predicted by Resistance Gene
Identifier
BestHitARO
Drug class
TreB
TreT
adeF
Fluoroquinolone antibiotic;
tetracycline antibiotic
+
−
ArnT
Peptide antibiotic
+
−
eptB
Peptide antibiotic
+
−

## Page 28

fosA5
Fluoroquinolone antibiotic;
aminoglycoside antibiotic;
phosphonic acid antibiotic
+
−
Klebsiella pneumoniae
KpnG
Macrolide antibiotic;
fluoroquinolone antibiotic;
aminoglycoside antibiotic;
carbapenem; cephalosporin;
penam; peptide antibiotic;
penem
+
−
Klebsiella pneumoniae
KpnH
Macrolide antibiotic;
fluoroquinolone antibiotic;
aminoglycoside antibiotic;
carbapenem; cephalosporin;
penam; peptide antibiotic;
penem
+
−
LptD
Peptide antibiotic;
aminocoumarin antibiotic;
rifamycin antibiotic
+
−
Morganella morganii
gyrB conferring
resistance to
fluoroquinolones
Fluoroquinolone antibiotic
+
−
oqxA
Fluoroquinolone antibiotic;
glycylcycline; tetracycline
antibiotic; diaminopyrimidine
antibiotic; nitrofuran antibiotic
+
−
OXY-6-1
Monobactam; cephalosporin;
penam
+
−
qacJ
Disinfecting agents and
antiseptics
+
−
baeR
Aminoglycoside antibiotic;
aminocoumarin antibiotic
+
+
CRP
Macrolide antibiotic;
fluoroquinolone antibiotic;
penam
+
+
emrR
Fluoroquinolone antibiotic
+
+

## Page 29

Escherichia coli
AcrAB-TolC with MarR
mutations conferring
resistance to
ciprofloxacin and
tetracycline
Fluoroquinolone antibiotic;
cephalosporin; glycylcycline;
penam; tetracycline antibiotic;
rifamycin antibiotic; phenicol
antibiotic; disinfecting agents
and antiseptics
+
+
Escherichia coli EF-Tu
mutants conferring
resistance to
pulvomycin
Elfamycin antibiotic
+
+
Escherichia coli UhpT
with mutation
conferring resistance
to fosfomycin
Phosphonic acid antibiotic
+
+
H-NS
Macrolide antibiotic;
fluoroquinolone antibiotic;
cephalosporin; cephamycin;
penam; tetracycline antibiotic
+
+
Haemophilus
influenzae PBP3
conferring resistance
to beta-lactam
antibiotics
Cephalosporin; cephamycin;
penam
+
+
Klebsiella pneumoniae
KpnE
Macrolide antibiotic;
aminoglycoside antibiotic;
cephalosporin; tetracycline
antibiotic; peptide antibiotic;
rifamycin antibiotic; disinfecting
agents and antiseptics
+
+
Klebsiella pneumoniae
KpnF
Macrolide antibiotic;
aminoglycoside antibiotic;
cephalosporin; tetracycline
antibiotic; peptide antibiotic;
rifamycin antibiotic; disinfecting
agents and antiseptics
+
+
leuO
nucleoside antibiotic;
disinfecting agents and
antiseptics
+
+

## Page 30

marA
Fluoroquinolone antibiotic;
monobactam; carbapenem;
cephalosporin; glycylcycline;
cephamycin; penam;
tetracycline antibiotic; rifamycin
antibiotic; phenicol antibiotic;
penem; disinfecting agents and
antiseptics
+
+
msbA
nitroimidazole antibiotic
+
+
rsmA
Fluoroquinolone antibiotic;
diaminopyrimidine antibiotic;
phenicol antibiotic
+
+
vanG
Glycopeptide antibiotic
+
+
CMY-70
Cephamycin
−
+
emrB
Fluoroquinolone antibiotic
−
+
Escherichia coli acrA
Fluoroquinolone antibiotic;
cephalosporin; glycylcycline;
penam; tetracycline antibiotic;
rifamycin antibiotic; phenicol
antibiotic; disinfecting agents
and antiseptics
−
+
Escherichia coli EF-Tu
mutants conferring
resistance to
pulvomycin
Elfamycin antibiotic
−
+
Escherichia coli GlpT
with mutation
conferring resistance
to fosfomycin
Phosphonic acid antibiotic
−
+
Escherichia coli mdfA
Tetracycline antibiotic;
disinfecting agents and
antiseptics
−
+
kdpE
Aminoglycoside antibiotic
−
+
mdtB
Aminocoumarin antibiotic
−
+
mdtG
Phosphonic acid antibiotic
−
+
PmrF
Peptide antibiotic
−
+

## Page 31

Whole genome sequence of K. grimontii Tre-B showed that it has
the potential to be a human pathogen. Analysis by antiSMASH 7.0
(ref. 68) revealed that it contains the entire biosynthetic gene
cluster for the potent cytotoxin kleboxymycin (Fig. 6 and Table 4),
while genes for other virulence factors were identified by
VirulenceFinder-2.0 Server67 and PathogenFinder.65,66
VirulenceFinder-2.0 Server identified the gene encoding the
lipoprotein NlpI precursor as virulence factor according to previous
data showing an involvement of lipoprotein NlpI in the virulence of
adherent invasive Escherichia coli strain isolated from a patient
with Crohn's disease.81 PathogenFinder identified a long list of
putative virulence factors and provided a probability score of being
a human pathogen of 0.858 (as a reference, the probability score of
Salmonella enterica sv. Typhimurium LT2 is 0.937). Among the
virulence factors identified by PathogenFinder, adhesins, fimbrial
systems, flagellin, invasins, cell invasion proteins, secretion system
structural proteins and effectors, LPS modification enzymes,
hemolysins, iron uptake systems, phospholipases, and proteins
involved in host sialic acid metabolism and uptake were found.
[image]
Fig. 6 Kleboxymycin gene cluster revealed by antiSMASH
in the genome of K. grimontii Tre-B (A) and aryl polyene
gene cluster revealed by antiSMASH in the genome of C.
braakii Tre-T (B).
Table 4 Secondary metabolites in K. grimontii Tre-B and C. braakii
Tre-T predicted by antiSMASH
Region
Type
From
To
Most similar
known
cluster
Similarity Strain
Region
12.1
Thiopeptide
245[image]295
271[image]573
O-Antigen
saccharide
14%
Tre-B
Region
24.1
RiPP
27[image]708
36[image]416
Tre-B

## Page 32

Region
28.1
NRPS
70[image]758
113[image]466
Kleboxymycin
NRP
100%
Tre-B
Region
28.2
T1PKS, NRPmetallophore,
NRPS
176[image]031
239[image]201
Yersiniabactin
NPR +
polyketide
16%
Tre-B
Region
29.1
NRPmetallophore
148[image]067
202[image]054
Enterobactin
NPR
100%
Tre-B
Region
4.1
Aryl polyene
280[image]804
324[image]400
APE Ec
94%
Tre-T
Region
12.1
NRPmetallophore,
NRPS
275[image]693
329[image]439
Enterobactin
NRP
100%
Tre-T
Region
16.1
Thiopeptide
1[image]792[image]500
1[image]818[image]790
O-Antigen
saccharide
14%
Tre-T
Genes coding for structural components of metabolosomes were
also found. Metabolosomes are bacterial microcompartments
(BMCs) forming polyhedral bodies, which consist of a single-layer
proteinaceous shell that encapsulates both enzymes and metabolites
facilitating specific catabolic pathways in a protected microenvironment.82 In general, these pathways are characterized by the
presence of oxygen sensitive metal co-factor containing enzymes,
such as coenzyme B12-dependent and glycyl-radical enzymes, and
BMCs may facilitate these pathways by an O2 exclusion
mechanism.83,84 The most well studied metabolosomes are the 1,2propanediol utilization (pdu), the ethanolamine utilization (eut),
the choline utilization (cut), and the glycyl radical propanediol
(grp) catabolic BMCs, which are found in several strains of
Salmonella enterica and Escherichia coli82 (Fig. 7, S3 and Table
S3†).
[image]
Fig. 7 Genetic maps of cut, pdu and eut loci of K.
grimontii Tre-B, and pdu and eut loci of C. braakii Tre-T.

## Page 33

K. grimontii Tre-B contains three BMC loci (Table S1†): pdu BMC
locus, eut BMC locus and choline utilization (cut) BMC locus (Fig. 7
and Table S3†) characterized in E. coli 536 and Proteus
mirabilis.85,86cut BMC loci were identified in 21 of 25 fully
sequenced K. grimontii genomes, suggesting a broad distribution
among strains of this species (Table S4†). cut enzymes choline
trimethylamine lyase (CutC) and its activating enzyme (CutD) were
also found in K. pneumoniae,87 and structural shell BMC proteins
have been recently characterized in this microorganism.88cut BMCs
have been implicated in diseases in humans, because they
catabolize choline to trimethylamine (TMA) plus ethanol or acetate,
and they are a major source of TMA in the intestine. TMA is further
metabolized in the liver to trimethylamine-N-oxide (TMAO), whose
high levels have been associated with various diseases, including
non-alcoholic fatty liver disease,89 cardiovascular disease and
atherosclerosis,90,91 kidney disease,92 and diabetes.93
On the other hand, the presence of dmsA coding for dimethyl
sulfoxide/trimethylamine N-oxide reductase in the genome of K.
grimontii Tre-B (Fig. 8 and Table S1†) may allow this
microorganism to use TMAO as an alternative electron acceptor in
anaerobic respiration,94 and in the genome of K. grimontii Tre-B it
may be noted the presence of dmsA coding for dimethyl sulfoxide/
trimethylamine N-oxide reductase, a molybdopterin-dependent
oxidoreductase. dmsA maps in a genomic region encompassing
genes involved in biosynthesis of lipoic acid from octanoic acid
(lipA, lipB, pagP), fluoride ion transport (crcB), general stress
(uspG), cold stress (cspE), oxidative stress (ahpF, ahpC), glutathione
metabolism (ybeM) and methionine metabolism and savage
pathway (ybdO, ybdM, ybdL, mntC, mntD) (Fig. 8). In particular,
the fluoride ion transport may be relevant for resistance to PFOA in
presence of activities that defluorinate this compound.
[image]

## Page 34

Fig. 8 Genetic mapping and conservation of proteincoding regions involved in lipoic acid metabolism and
fluoride transport. (A) Genetic maps of K. grimontii Tre-B
and C. braakii Tre-T. The red arrow highlights the CrcB
protein (F− specific ion channel). (B and C) Cooccurrence of conserved genes. The data set was
generated by inputting the CrcB protein sequence of K.
grimontii Tre-B (OKNGJBID_03358) and C. braakii Tre-T
(LBFIJIGF_01727) and using the EFI-EST and EFI-GNT
tools. (D) Most abundant bacterial genera in the data set
obtained from EFI-EST/EFI-GNT.
MobileElementFinder69 detected the presence of IncR-group
plasmid and numerous mobile genetic elements in the genome of K.
grimontii Tre-B. The IncR-group plasmid (named pKGTreB) has a
total length of 87[image]115 bp and contains 88 predicted CDSs,
including numerous insertion sequences (ISEcl1, ISKpn34, IS903,
IS26) (Table 2), and, notably, many genes involved in resistance to
arsenic, copper, mercury and silver (Table 5), and a gene coding for
a putative glycosyl transferase (epsJ) that in Bacillus subtilis 168
has been involved in biofilm matrix formation.95 pKGTreB also
contains genes coding for proteins involved in maltose/maltodexrin
transport and metabolism, and type II toxin-antitoxin system (vapB/
vapC). Resistance against silver was confirmed in laboratory
experiments. In particular, while the MIC of silver nitrate was the
same in K. grimontii Tre-B and reference E. coli FB8 (0.63 mM),
and lower in Citrobacter braakii Tre-T (0.08 mM), the MBC of this
compound was much higher in K. grimontii Tre-B (320 mM) as
compared to E. coli FB8 (0.15 mM), and lower in C. braakii Tre-T (5
mM) (Fig. S2†). In contrast, the MIC values of chromium,
aluminum, nickel, and copper were the same in K. grimontii Tre-B
and C. braakii Tre-T.
Table 5 Relevant CDSs of pKGTreB plasmid
Locustag
(genome)
Locustag
(plasmid)
CDS
(bp)
Gene
Product
OKNGJBID02184 MGA88/89 924
IS5 family transposase
IS903

## Page 35

OKNGJBID02185 MGA58
2967
Tn3-like element
ISPa38 family
transposase
OKNGJBID02186 MGA59
291
Nucleotidyltransferase
OKNGJBID02187 MGA60
402
DUF86 domaincontaining protein
OKNGJBID02188 MGA61
357
Cupin domaincontaining protein
OKNGJBID02189 MGA62
327
Hypothetical protein
OKNGJBID02190 MGA63
501
Hypothetical protein
OKNGJBID02191 MGA64
372
Hypothetical protein
OKNGJBID02192 MGA65
558
hin
Recombinase family
protein
OKNGJBID02193 MGA66
750
epsJ
Glycosyltransferase
EpsJ
OKNGJBID02194 MGA67
828
usp
Universal stress protein
OKNGJBID02195 MGA68
1479
SulP family inorganic
anion transporter
OKNGJBID02196 MGA69
249
Recombinase family
protein
OKNGJBID02197 MGA70
2985
Tn3-like element TnAs3
family transposase
OKNGJBID02198
123
Hypothetical protein
OKNGJBID02199 MGA71
126
Hypothetical protein
OKNGJBID02200 MGA72
795
IS3 family transposase
ISKpn34
OKNGJBID02201 MGA73
2796
Tn3-like element Tn3
family transposase
OKNGJBID02202 MGA74
573
Recombinase family
protein
OKNGJBID02203 MGA75
405
Hypothetical protein
OKNGJBID02204 MGA76
426
arsC
Glutaredoxindependent arsenate
reductase

## Page 36

OKNGJBID02205 MGA77
1290 arsB
Arsenite efflux
transporter membrane
subunit ArsB
OKNGJBID02206 MGA78
1752 arsA
Arsenite efflux
transporter ATPase
subunit ArsA
OKNGJBID02207 MGA79
363
arsD
Arsenite efflux
transporter
metallochaperone ArsD
OKNGJBID02208 MGA80
354
arsR
As(III)-sensing
metalloregulatory
transcriptional
repressor ArsR
OKNGJBID02209 MGA82
501
ftnA
Non-heme ferritin-like
protein
OKNGJBID02210 MGA90
705
IS6-like element IS26
family transposase
OKNGJBID02211
1890
Tn3 family transposase
OKNGJBID02212 MGA84
1326 amyB Alpha-amylase family
glycosyl hydrolase
OKNGJBID02213 MGA85
1188 malE Maltose/maltodextrin
ABC transporter
substrate-binding
protein MalE
OKNGJBID02214 MGA90
705
IS6-like element IS26
family transposase
OKNGJBID02215 MGA1
558
hin
Recombinase family
protein
OKNGJBID02216 MGA2
213
DUF3330 domaincontaining protein
OKNGJBID02217 MGA3
237
merE Broad-spectrum
mercury transporter
MerE
OKNGJBID02218 MGA4
366
merD Mercury resistance coregulator MerD
OKNGJBID02219 MGA5
1686 merA Mercury(II) reductase

## Page 37

OKNGJBID02220 MGA6
426
merC Organomercurial
transporter MerC
OKNGJBID02221 MGA7
276
merP Mercury resistance
system periplasmic
binding protein MerP
OKNGJBID02222 MGA8
411
merT Mercuric ion
transporter MerT
OKNGJBID02223 MGA9
456
merR Hg(II)-responsive
transcriptional
regulator
OKNGJBID02224 MGA10
1083
IS110 family
transposase
OKNGJBID02225 MGA12
1524
Group II intron reverse
transcriptase/maturase
OKNGJBID02226 MGA14
285
IS3 family transposase
OKNGJBID02227 MGA15
417
vapC Type II toxin-antitoxin
system VapC family
toxin
OKNGJBID02228 MGA16
231
vapB Type II toxin-antitoxin
system VapB family
antitoxin
OKNGJBID02229 MGA17
378
Transposase
OKNGJBID02230 MGA18
348
IS66 family insertion
sequence element
accessory protein TnpB
OKNGJBID04814 MGA19
690
IS66-like element ISEc8
family transposase
OKNGJBID04815 MGA20
180
parD Type II toxin-antitoxin
system ParD family
antitoxin
OKNGJBID04816 MGA21
99
Hypothetical protein
OKNGJBID04817 MGA22
435
pcoE Copper resistance
system
metallochaperone PcoE

## Page 38

OKNGJBID04818 MGA23
1401 pcoS Copper resistance
membrane spanning
protein PcoS
OKNGJBID04819 MGA24
681
pcoR Copper response
regulator transcription
factor PcoR
OKNGJBID04820 MGA25
930
pcoD Copper resistance inner
membrane protein
PcoD
OKNGJBID04821 MGA26
381
pcoC Copper resistance
system
metallochaperone PcoC
OKNGJBID04822 MGA27
897
pcoB Copper resistance outer
membrane transporter
PcoB
OKNGJBID04823 MGA28
1818 pcoA Multicopper oxidase
PcoA
OKNGJBID04824 MGA29
450
Copper resistance
protein
OKNGJBID04825 MGA30
738
Peptidoglycan DDmetalloendopeptidase
family protein
OKNGJBID04826 MGA31
198
DUF2933 domaincontaining protein
OKNGJBID04827 MGA32
2442 silP
Ag(+)-translocating Ptype ATPase SilP
OKNGJBID04828 MGA33
441
DUF411 domaincontaining protein
OKNGJBID04829 MGA34
3147 silA
Cu(+)/Ag(+) efflux
RND transporter
permease subunit SilA
OKNGJBID04830 MGA35
1293 silB
Cu(+)/Ag(+) efflux
RND transporter
periplasmic adaptor
subunit SilB

## Page 39

OKNGJBID04831 MGA36
354
cusF
Cation efflux system
protein CusF
OKNGJBID04832 MGA37
1386 silC
Cu(+)/Ag(+) efflux
RND transporter outer
membrane channel SilC
OKNGJBID04833 MGA38
681
silR
Copper/silver response
regulator transcription
factor SilR
OKNGJBID04834 MGA39
1476 silS
Copper/silver sensor
histidine kinase SilS
OKNGJBID04835 MGA40
432
silE
Silver-binding protein
SilE
OKNGJBID04836 MGA41
234
Hypothetical protein
OKNGJBID04837 MGA42
915
HNH endonuclease
OKNGJBID04838
255
Hypothetical protein
OKNGJBID04839 MGA43
396
Hypothetical protein
OKNGJBID04840 MGA44
882
Hypothetical protein
OKNGJBID04841 MGA45
564
Hypothetical protein
OKNGJBID04842 MGA46
582
Hypothetical protein
OKNGJBID04843 MGA47
351
Hypothetical protein
OKNGJBID04844 MGA48
744
Hypothetical protein
OKNGJBID04845 MGA49
777
Site-specific integrase
OKNGJBID04846 MGA50
258
Hypothetical protein
OKNGJBID04847 MGA51
867
repE
Replication initiation
protein RepE
OKNGJBID04848 MGA52
270
Hypothetical protein
OKNGJBID04849 MGA53
1206 parA AAA family ATPase
OKNGJBID04850 MGA54
975
parB ParB family protein
OKNGJBID04851 MGA56
276
IS1-like element
transposase
OKNGJBID04852 MGA57
267
IS1 family transposase
OKNGJBID04853 MGA57
267
IS1 family transposase

## Page 40

OKNGJBID04854 MGA56
276
IS1-like element
transposase
The genome sequence of K. grimontii Tre-B also showed that it has
a considerable potential for degradation of a wide array of aromatic
compounds and recalcitrant chemicals (Table 6). A total of 14
monooxygenase- and 18 dioxygenase-encoding genes were
annotated in the genome sequence, which are involved in different
pathways, including: (i) degradation of exogenous pyrimidines as
the sole nitrogen source; (ii) degradation of homo-protocatechuate;
(iii) degradation of 4-hydroxyphenylacetate; (iv) degradation of
benzoate and 2-halo (F, Br, Cl, I)-benzoate to catechol; (v)
degradation of catechol to beta-ketoadipate; (vi) degradation of
beta-ketoadipate to succinyl-CoA and acetyl-CoA; (vii) degradation
of 4-hydroxybenzoate to protocatechuate; (viii) degradation of 3hydroxybenzoate via gentisate to pyruvate and fumarate; (ix)
degradation of nitrilotriacetate (NTA) to iminodiacetate and
glyoxylate; (x) degradation of aliphatic sulfonates to utilize
dimethyl sulfide and methanesulfonate as a carbon and energy and/
or sulfur source; (xi) degradation of taurine as an alternative sulfur
source for growth in the absence of sulfate; (xii) degradation and
metabolism of quercetin and other plant flavonoids. It may be also
noted the presence of duplicated genes coding for validamycin A
dioxygenase that is responsible for transformation of the antibiotic
validamycin A to validamycin B, which is less active (Table 6).
Table 6 Pathways for degradation of aromatic compounds and
recalcitrant chemicals as inferred from K. grimontii Tre-B genome
sequence
Locustag
CDS
length
(bp)
Gene
Product
P
OKNGJBID00372 639
rutR
HTH-type transcriptional
regulator RutR
Degrad
exogen
pyrimid
sole nit

## Page 41

OKNGJBID00373 1092
rutA
Pyrimidine monooxygenase
RutA
OKNGJBID00374 711
rutB1
Peroxyureidoacrylate/
ureidoacrylate amidohydrolase
RutB
OKNGJBID00375 393
rutC1
Putative aminoacrylate peracid
reductase RutC
OKNGJBID00376 804
rutD
Putative aminoacrylate
hydrolase RutD
OKNGJBID00377 591
rutE
Putative malonic semialdehyde
reductase RutE
OKNGJBID00378 495
rutF
FMN reductase (NADH) RutF
OKNGJBID00379 1323
rutG
Putative pyrimidine permease
RutG
OKNGJBID00754 633
hpcE1 Homoprotocatechuate
catabolism bifunctional
isomerase/decarboxylase
Degrad
protoca
OKNGJBID00755 765
hpcE2 Homoprotocatechuate
catabolism bifunctional
isomerase/decarboxylase
OKNGJBID00756 1467
betB1
NAD/NADP-dependent betaine
aldehyde dehydrogenase
OKNGJBID00757 858
hpcB
3,4-Dihydroxyphenylacetate 2,3dioxygenase
OKNGJBID00758 381
hpcD
5-Carboxymethyl-2hydroxymuconate deltaisomerase
OKNGJBID00759 804
hpcG
2-Oxo-hept-4-ene-1,7-dioate
hydratase
OKNGJBID00760 792
hpcH
4-Hydroxy-2-oxo-heptane-1,7dioate aldolase
OKNGJBID00763 1563
hpaB
4-Hydroxyphenylacetate 3monooxygenase oxygenase
component
Degrad
hydrox

## Page 42

OKNGJBID00764 513
hpaC
4-Hydroxyphenylacetate 3monooxygenase reductase
component
OKNGJBID03120 1017
benC
Benzoate 1,2-dioxygenase
electron transfer component
Degrad
benzoa
(F, Br, C
to catec
OKNGJBID03121 486
cbdB
2-Halobenzoate 1,2-dioxygenase
small subunit
OKNGJBID03122 1383
cbdA
2-Halobenzoate 1,2-dioxygenase
large subunit
OKNGJBID03123 927
catA
Catechol 1,2-dioxygenase
Degrad
catecho
ketoadi
OKNGJBID03124 291
catC
Muconolactone delta-isomerase
OKNGJBID03125 1119
catB
Muconate cycloisomerase 1
OKNGJBID05375 768
catD
3-Oxoadipate enol-lactonase 2
OKNGJBID03126 798
pcaR1 Pca regulon regulatory protein
Degrad
protoca
beta-ke
OKNGJBID05370 807
pcaR2 Pca regulon regulatory protein
OKNGJBID05374 1353
pcaB
3-Carboxy-cis,cis-muconate
cycloisomerase
OKNGJBID03206 741
pcaH
Protocatechuate 3,4-dioxygenase
beta chain
OKNGJBID03207 621
pcaG
Protocatechuate 3,4-dioxygenase
alpha chain
OKNGJBID05371 687
pcaI
3-Oxoadipate CoA-transferase
subunit A
Degrad
ketoadi
succiny
acetyl-C
OKNGJBID05372 657
pcaJ
3-Oxoadipate CoA-transferase
subunit B
OKNGJBID05373 1203
pcaF
Beta-ketoadipyl-CoA thiolase

## Page 43

OKNGJBID00725 1185
pobA
p-Hydroxybenzoate hydroxylase Degrad
hydrox
protoca
OKNGJBID01380 1194
mhbM 3-Hydroxybenzoate 6hydroxylase
Degrad
hydrox
gentisa
and fum
OKNGJBID01381 645
nagL
Maleylpyruvate isomerase
OKNGJBID01382 642
nagK1 Fumarylpyruvate hydrolase
OKNGJBID01383 1038
nagI
Gentisate 1,2-dioxygenase
OKNGJBID01384 1359
mhbT
3-Hydroxybenzoate transporter
MhbT
OKNGJBID05070 1353
ntaA
Nitrilotriacetate monooxygenase
component A
Degrad
nitrilotr
(NTA) t
iminod
glyoxyl
OKNGJBID04245 942
ssuA1
Putative aliphatic sulfonatesbinding protein
Degrad
aliphat
(dimeth
methan
OKNGJBID04246 1407
dmoA
Dimethyl-sulfide
monooxygenase
OKNGJBID04247 1398
Hypothetical protein
OKNGJBID04248 1173
ssuD1
(msuD1)
Methanesulfonate
monooxygenase
OKNGJBID00548 576
ssuE
FMN reductase (NADPH)
OKNGJBID00549 792
ssuC1
Putative aliphatic sulfonates
transport permease protein SsuC
OKNGJBID00550 774
ssuB1
Aliphatic sulfonates import ATPbinding protein SsuB
OKNGJBID05215 597
3-Mercaptopropionate
dioxygenase
OKNGJBID05216 894
gltC8
HTH-type transcriptional
regulator GltC

## Page 44

OKNGJBID05217 1086
ssuD2
(msuD2)
Methanesulfonate
monooxygenase
OKNGJBID05218 1146
ydbM
Putative acyl-CoA
dehydrogenase YdbM
OKNGJBID05219 975
ssuA2
Putative aliphatic sulfonatesbinding protein
OKNGJBID05220 1038
Hypothetical protein
OKNGJBID05221 996
Hypothetical protein
OKNGJBID05222 789
ssuB4
Aliphatic sulfonates import ATPbinding protein SsuB
OKNGJBID05270 1224
sfnC
Putative FMNH2-dependent
monooxygenase SfnC
OKNGJBID01203 852
tauD
Alpha-ketoglutarate-dependent
taurine dioxygenase
Degrad
taurine
alterna
source 
the abs
OKNGJBID01204 828
tauC
(ssuC2)
Putative aliphatic sulfonates
transport permease protein SsuC
OKNGJBID01205 771
tauB
Taurine import ATP-binding
protein TauB
OKNGJBID01206 963
tauA
Taurine-binding periplasmic
protein
OKNGJBID03899 1038
yhhX
Putative oxidoreductase YhhX
Degrad
metabo
quercet
OKNGJBID03900 696
yhhW
Quercetin 2,3-dioxygenase
OKNGJBID05206 1668
mhpA 3-(3-Hydroxyphenyl)propionate/3hydroxycinnamic acid
hydroxylase
OKNGJBID05207 945
mhpB
2,3Dihydroxyphenylpropionate/2,3dihydroxycinnamic acid 1,2dioxygenase

## Page 45

OKNGJBID05208 867
mhpC
2-Hydroxy-6oxononadienedioate/2hydroxy-6-oxononatrienedioate
hydrolase
OKNGJBID05209 807
mhpD 2-Keto-4-pentenoate hydratase
OKNGJBID05210 951
mhpF
Acetaldehyde dehydrogenase
OKNGJBID05211 1017
mhpE
4-Hydroxy-2-oxovalerate
aldolase
OKNGJBID05212 1197
mhpT
3-(3-Hydroxy-phenyl)propionate
transporter
OKNGJBID03825 873
ectD
Ectoine dioxygenase
Glycine
threoni
OKNGJBID04586 792
ygiD
4,5-DOPA dioxygenase extradiol Tyrosin
OKNGJBID04573 315
ygiN
Putative quinol monooxygenase
YgiN
Quinon
OKNGJBID03603 306
ydhR
Putative monooxygenase YdhR
OKNGJBID01443 651
alkB
Alpha-ketoglutarate-dependent
dioxygenase AlkB
DNA re
OKNGJBID02888 1029
vldW1 Validamycin A dioxygenase
Transfo
validam
validam
OKNGJBID04037 1077
vldW2 Validamycin A dioxygenase
OKNGJBID04040 729
cloR
4-Hydroxy-3prenylphenylpyruvate
oxygenase/4-hydroxy-3prenylbenzoate synthase
Biosynt
seconda
metabo
OKNGJBID03156 1317
moxC
Putative monooxygenase MoxC
Unknow
Whole genome sequence and characterization of
Citrobacter braakii strain Tre-T
A total of 8[image]193[image]946 reads were sequenced for strain
Tre-T and the genome assembly was represented by 28 contigs with

## Page 46

a N50 value of 484[image]747 bp and a N75 value of 413[image]051
bp; the largest contig was 1[image]837[image]299 bp. The average
coverage was >90-fold. The combined length was
4[image]936[image]210 bp with a G + C content of 52.2% (Table
2). Based on Genome Taxonomy Database, genome sequence of
strain Tre-T showed the highest identity (98.71%) with the that of
Citrobacter braakii strain ATCC 51113 (accession
GCA002075345.1). Rapid genome annotation was performed with
Prokka.62 Annotation features identified with Prokka v1.14.6 (ref.
62) includes a 4581 DNA coding sequences (CDSs), 13 rRNAs, 75
tRNAs (Table 2). Annotation features identified with DFAST v5.0.2
(ref. 63) identified a total of 4563 CDSs, 9 rRNAs, 75 tRNAs and
one CRISPR locus (Table 2). According to these results, annotation
was then manually implemented, and results are reported is
reported in Table S5.†
C. braakii is a species belonging to the large Citrobacter freundii
complex.96 As with K. grimontii, concern is growing about the
environmental spread of these bacteria as they are acquiring
multidrug resistance.97,98 Indeed, the analysis of the genome of C.
braakii Tre-T with Resistance Gene Identifier allowed to predict
acquired resistance to very wide range of antibiotics, disinfecting
agents and antiseptics (Tables 3 and S6†). In particular, factors
associated with resistance to fluoroquinolones include the presence
of the acridine-resistance proteins A and B (AcrAB) and the
multidrug efflux pump outer membrane factor TolC (AcrAB-TolC)
with MarR mutations conferring resistance to ciprofloxacin and
tetracycline, also present in K. grimontii Tre-B. It may be seen that
in several members of the γ-Proteobacteria this efflux pump confers
resistance to a wide range of toxic compounds such as antibiotics,
surfactants, dyes, detergents, and disinfectants which are not found
in the natural environment of these bacteria.99–101
The analysis of the genome of C. braakii Tre-T also revealed that
the genomic region encompassing genes involved in biosynthesis of
lipoic acid from octanoic acid (lipA, lipB, pagP), fluoride ion
transport (crcB), general stress (uspG), cold stress (cspE), oxidative
stress (ahpF, ahpC), glutathione metabolism (ybeM) and methionine
metabolism (ybdO, ybdM, ybdL) have a similar arrangement with
respect to the syntenic region of K. grimontii Tre-B. However, the K.

## Page 47

grimontii Tre-B gene dmsA coding for dimethyl sulfoxide/
trimethylamine N-oxide reductase was notably replaced by a
different molybdopterin-dependent oxidoreductase that is not able
to use TMAO (Fig. 8). The analysis of the genome of C. braakii TreT revealed two BMC loci (Table S5†): the pdu BMC locus and the
eut BMC locus. At variance with K. grimontii Tre-B, the cut BMC
locus was absent (Fig. 7 and Table S3†). C. braakii Tre-T as well as
K grimontii Tre-B also present loci eut and pdu (Fig. 7). These loci
are extremely similar to those of K grimontii Tre-B and the encoded
proteins are homologous to those identified in S. enterica LT2.
Whole genome sequence of C. braakii Tre-T showed that it has the
potential to be a human pathogen. antiSMASH 7.0 (ref. 68) revealed
that it contains an aryl polyene biosynthetic gene cluster (Fig. 6 and
Table 4), 94% similar to that found in E. coli CFT073. Aryl polyene
are specialized polyunsaturated carboxylic acids that increase
protection from oxidative stress and contribute to biofilm formation
in pathogenic E. coli strains.102 VirulenceFinder-2.0 Server67
identified the gene encoding the lipoprotein NlpI precursor as
virulence factor,81 also found in also present in K. grimontii Tre-B.
PathogenFinder65,66 identified a long list of putative virulence
factors and provided a probability score of being a human pathogen
of 0.868 (as a reference, the probability score of Salmonella
enterica sv. Typhimurium LT2 is 0.937). Among the virulence
factors identified by PathogenFinder, Vi polysaccharide biosynthesis
protein TviE, also found in S. enterica subsp. enterica serovar
Parathyphi C, and hemolysin HylD were found.
In addition to nlpI, MobileElementFinder69 detected the presence of
blaCMY-82 and blaCMY-101 to many beta-lactams and their associations
(ampicillin + clavulanic acid, ceftazidime, ticarcillin + clavulanic
acid, ampicillin, piperacillin + tazobactam, cefoxitin, amoxicillin,
ticarcillin, cefotaxime, piperacillin, amoxicillin + clavulanic acid),
and traT encoding an outer membrane protein that is involved in
resistance to complement.
The genome sequence of C. braakii Tre-T also showed that it has a
potential for degradation of some aromatic compounds and
recalcitrant chemicals (Table 7). A total of 7 monooxygenase- and 8
dioxygenase-encoding genes were annotated in the genome

## Page 48

sequence, which are involved in different pathways, including: (i)
degradation of 3-phenylpropanoate; (ii) degradation of 3hydroxybenzoate via gentisate to pyruvate and fumarate; (iii)
degradation of aliphatic sulfonates to utilize dimethyl sulfide and
methanesulfonate as a carbon and energy and/or sulfur source; (iv)
degradation of taurine as an alternative sulfur source for growth in
the absence of sulfate; (v) degradation and metabolism of quercetin
and other plant flavonoids. The genome sequence also revealed
genes involved in degradation of carnitine to trimethylamine (TMA)
and malic semialdehyde, which are absent in K. grimontii Tre-B.
Thus, while K. grimontii Tre-B appears to be able to produce TMA
through the catabolism of choline in the cut BMC, C. braakii Tre-T
can produce TMA through the catabolism of carnitine.
Table 7 Pathways for degradation of aromatic compounds and
recalcitrant chemicals as inferred from C. braakii Tre-T genome
sequence
Locustag
CDS
length
(bp)
Gene
Product
Path
LBFIJIGF02889 1203
hcaD
3-Phenylpropionate/cinnamic
acid dioxygenase ferredoxin–
NAD(+) reductase component
Degradati
phenylpro
LBFIJIGF02890 813
hcaB
3-Phenylpropionatedihydrodiol/cinnamic aciddihydrodiol dehydrogenase
LBFIJIGF02891 321
hcaC
3-Phenylpropionate/cinnamic
acid dioxygenase ferredoxin
subunit
LBFIJIGF02892 519
hcaF
3-Phenylpropionate/cinnamic
acid dioxygenase subunit beta
LBFIJIGF02893 1362
hcaE
3-Phenylpropionate/cinnamic
acid dioxygenase subunit alpha
LBFIJIGF02894 882
hcaR
Hca operon transcriptional
activator HcaR

## Page 49

LBFIJIGF03255 1359
mhbT 3-Hydroxybenzoate transporter
MhbT
Degradati
hydroxyb
via gentis
pyruvate 
fumarate
LBFIJIGF03256 1038
nagI
(sdgD)
Gentisate 1,2-dioxygenase
LBFIJIGF03257 702
nagK1 Fumarylpyruvate hydrolase
LBFIJIGF03258 645
nagL
Maleylpyruvate isomerase
LBFIJIGF03259 1194
mhbM 3-Hydroxybenzoate 6hydroxylase
LBFIJIGF04525 576
ssuE
FMN reductase (NADPH)
Degradati
aliphatic
sulfonates
(dimethyl
and
methanes
LBFIJIGF04526 975
ssuA
Putative aliphatic sulfonatesbinding protein
LBFIJIGF04527 1146
ssuD
Alkanesulfonate monooxygenase
LBFIJIGF04528 792
ssuC2 Putative aliphatic sulfonates
transport permease protein SsuC
LBFIJIGF04529 768
ssuB
Aliphatic sulfonates import ATPbinding protein SsuB
LBFIJIGF00719 852
tauD1 Alpha-ketoglutarate-dependent
taurine dioxygenase
Degradati
taurine as
alternativ
source for
in the abs
sulfate
LBFIJIGF00720 831
tauC
(ssuC1)
Putative aliphatic sulfonates
transport permease protein SsuC
LBFIJIGF00721 768
tauB
Taurine import ATP-binding
protein TauB
LBFIJIGF00722 747
tauA1 Taurine-binding periplasmic
protein

## Page 50

LBFIJIGF00723 165
tauA2 Taurine-binding periplasmic
protein
LBFIJIGF01999 852
tauD2 Alpha-ketoglutarate-dependent
taurine dioxygenase
LBFIJIGF00372 1038
yhhX
Putative oxidoreductase YhhX
Degradati
metabolis
quercetin
LBFIJIGF00373 696
yhhW Quercetin 2,3-dioxygenase
LBFIJIGF00732 1212
mhpT 3-(3-Hydroxy-phenyl)propionate
transporter
LBFIJIGF00733 1014
mhpE 4-Hydroxy-2-oxovalerate
aldolase
LBFIJIGF00734 951
mhpF Acetaldehyde dehydrogenase
LBFIJIGF00735 810
mhpD 2-Keto-4-pentenoate hydratase
LBFIJIGF00736 867
mhpC 2-Hydroxy-6oxononadienedioate/2hydroxy-6-oxononatrienedioate
hydrolase
LBFIJIGF00737 945
mhpB 2,3Dihydroxyphenylpropionate/2,3dihydroxicinnamic acid 1,2dioxygenase
LBFIJIGF00738 1665
mhpA 3-(3-Hydroxyphenyl)propionate/3hydroxycinnamic acid
hydroxylase
LBFIJIGF03591 966
yeaX
Carnitine monooxygenase
reductase subunit
Degradati
carnitine 
trimethyla
and malic
semialdeh
LBFIJIGF03592 1125
yeaW Carnitine monooxygenase
oxygenase subunit
LBFIJIGF03593 1602
caiT2 L-Carnitine/gammabutyrobetaine antiporter

## Page 51

LBFIJIGF02638 789
ygiD
4,5-DOPA dioxygenase extradiol Tyrosine
metabolis
LBFIJIGF02647 315
ygiN
Putative quinol monooxygenase
YgiN
Quinone r
cycle
LBFIJIGF04036 309
ydhR
Putative monooxygenase YdhR
LBFIJIGF03175 651
alkB
Alpha-ketoglutarate-dependent
dioxygenase AlkB
DNA repa
LBFIJIGF03867 1314
moxC Putative monooxygenase MoxC
Unknown
LBFIJIGF00134 306
Putative monooxygenase
Unknown
Effect of PFOA on the antibiotic resistance genes
Whole genome sequencing of K. grimontii Tre-B and C. braakii TreT revealed the presence of numerous resistance genes, in particular
to fluoroquinolone antibiotics (Table 3). Therefore, a disk diffusion
method experiment was performed to measure the antibiotic
susceptibility of Tre-B and Tre-T strains in the MH agar
supplemented with increasing concentrations of PFOA (2 μg mL−1,
20 μg mL−1, 200 μg mL−1, 2 mg mL−1) or isopropanol as a control
(Fig. 9A).
[image]

## Page 52

Fig. 9 Impact of PFOA on antibiotic susceptibility testing
and transcription levels of antibiotic-resistance genes. (A)
Kirby–Bauer test results on Mueller–Hinton (MH) agar for
K. grimontii Tre-B and C. braakii Tre-T. The following
antibiotics were used: ampicillin 10 μg (AMP10),
cefoperazone 30 μg (CAZ30), ceftazidime 30 μg (CFP30),
amikacin 30 μg (AK30), tobramycin 10 μg (TOB10),
pefloxacin 5 μg (PEF5), pipemidic acid 20 μg (PI20),
azithromycin 15 μg (AZM15), tetracycline 30 μg (TE30),
and trimethoprim-sulfamethoxazol 25 μg (SXT25). (B)
Levels of transcript (RT-qPCR) of five antibiotic
resistance-related genes (kpnF, kpnG, adeF, oqxA, and
acrA) of K. grimontii Tre-B grown in LB supplemented
with an incremental concentration of PFOA. Asterisks
indicate statistical significance (p-value <0.05)
compared to the control.
A slight increase in sensitivity to ampicillin (AMP10) was observed
in C. braakii Tre-T at a PFOA concentration of 2 mg mL−1 (Fig. 9A).
In both strains, no appreciable effect in sensitivity to cephalosporins
(ceftazidime [CAZ] and cefepime [CFP]), tetracycline (TE) and
trimethoprim–sulfamethoxazole (SXT) was detected (Fig. 9A and
Table S7†). In contrast, increased resistance to aminoglycosides
(amikacin [AK] and tobramycin [TOB]), pefloxacin (PEF),
piperacillin (PI), and azithromycin (AZM) was observed in both
strains, with resistance levels increasing in parallel with increasing
PFOA concentrations, peaking at 2 mg mL−1 (Fig. 9A and Table
S7†).
An RT-qPCR experiment assessed the transcript levels of five
antibiotic resistance-related genes: kpnF, kpnG, adeF, oqxA, and
acrA. Strain Tre-B was cultured in LB broth supplemented with
PFOA at concentrations of 2, 20, and 200 μg mL−1, and RT-qPCR
analysis was subsequently performed to evaluate gene expression
levels (Fig. 9B). The results indicate a dose-dependent upregulation
of kpnF, kpnG, adeF and oqxA following exposure to increasing
concentrations of PFOA (2, 20, and 200 μg mL−1). Expression of
acrA was also increased at 20 μg per mL PFOA, but not further at
higher PFOA concentrations.

## Page 53

The RT-qPCR results indicate that PFOA activates the transcription
of many genes involved in multiple antibiotic resistance in the TreB strain, particularly those encoding efflux pumps. The strongest
effect was observed at the highest concentration tested (200 μg
mL−1), supporting the hypothesis that PFOA, in addition to select
specific groups of bacteria, can enhance antibiotic resistance by
upregulating specific ARGs.
Discussion
Although the increase in AMR has been mainly attributed to their
misuse or overuse in clinical practice, livestock and agriculture,103
much remains to be understood about the other drivers of AMR,
especially environmental ones. Among the factors involved in the
spread of AMR in the environment, the release of large amounts of
antibiotics into wastewater and the selective growth of antibioticresistant bacteria (ARB) in wastewater treatment plants, and the use
of contaminated organic fertilizers and irrigation water in the soils
are considered the most relevant.103,104 However, resistant and
multidrug-resistant microbial strains have also been isolated in
relatively uninhabited areas of the earth, and there is conclusive
evidence that antibiotic resistance is ancient. A highly diverse
collection of genes encoding resistance to β-lactam, tetracycline and
glycopeptide antibiotics was recovered by sequencing of ancient
DNA recovered from Late Pleistocene permafrost sediments.105
Furthermore, ARB and antibiotic resistance genes (ARG) were
recently detected in minimally human-impacted environments
including Antarctica although further research is required for better
detecting and quantifying ARB and ARG along human gradients to
better characterize the factors leading to their spread in pristine
environments.106,107 Therefore, something is still missing regarding
a complete understanding of the environmental drivers of AMR.
Understanding the environmental drivers of AMR is critical because
the circulation of bacterial ARG in different environments can be
considered a potential factor in the transfer of these genes to health
centers.108,109
In this study, the results of Winogradsky column experiments
provided some evidence for a link between PFAS contamination and
AMR. While the effects of PFOA and PFOS on soil microbial

## Page 54

communities have been extensively explored in observational
studies, Winogradsky columns as miniature ecosystem offer the
opportunity to analyze, under controlled laboratory conditions,
specific effects related to, for example, a single PFAS congener or
combination thereof, as well as possible synergistic interactions
between PFAS and other environmental contaminants. It also
provides an opportunity to analyze PFAS metabolism over time,
both in the aerobic and anaerobic zones of the column, microbial
successions, transcriptional activity, and specific microbial activities
stimulated by the presence of PFAS for computational modeling and
pathway prediction, as demonstrated for other environmental
pollutants.110,111 By using this experimental system, it is possible to
carry out biostimulation or biological enrichment experiments.
In line with a recent study,34 here we demonstrate that
environmental PFAS contamination may act as a driver for the
selection of environmental ARBs that behave as opportunistic
pathogens in humans. Microbial communities from the PFAScontaminated Trelleborg site B1 were stimulated with a high
concentration of PFOA in microcosm experiments, and this resulted
in the isolation of two bacterial species, K. grimontii and C. braakii
that are known to cause opportunistic infections. Both K. grimontii
Tre-B and C. braakii Tre-T are characterized by a large set of genes
involved in AMR, in particular to fluoroquinolones (Table 3).
Strains of K. grimontii were isolated from human blood cultures,
wound infections, antibiotic-associated colitis, as well as from feces
of healthy patients.79Klebsiella species are commonly found in
water, soil and plants and as commensals in the intestine of
animals, including humans.112 There is increasing concern about
the environmental spread of these bacteria because they are even
more frequently associated with nosocomial infections and are
developing multidrug resistance.113Klebsiella oxytoca is the second
most common Klebsiella species causing disease in humans, after K.
pneumoniae.80
Citrobacter species are commonly found in water, soil and plants
and as commensal in the intestine of animals, including humans.
Occasionally, they can cause enteric diseases but they are also
associated with extraintestinal disorders, among which the most

## Page 55

significant are neonatal meningitis and brain abscesses, and are
rarely implicated in skin or soft tissue infections.114,115C. braakii
strains have been described as plant growth-promoting rice
rhizobacteria.116 At the same time, C. braakii is a human
opportunistic pathogen that has been implicated in enteric diseases
(gastroenteritis), and rarely in sepsis and multiorgan dysfunctions in
immunocompromised patients.117,118 A recent case of bacteremia
due to carbapenem-resistant C. braakii has been reported.119
Although antibiotic resistance is of particular concern in pathogenic
bacteria, a growing number of studies draw attention to the
worrying increase in the prevalence of AMR in non-pathogenic
(commensal) bacterial species of the human microbiota.120 These
commensal bacterial species, including several members of the large
family of Enterobacteriaceae, can transfer ARG to pathogenic
species and can themselves cause opportunistic infections in
humans.121,122 Furthermore, most of them are environmental
species capable of growing or to persisting in different
environmental niches and undertaking horizontal gene transfer with
other environmental bacteria.123
It can be noted that both K. grimontii and C. braakii belong to
family of Enterobacteriaceae (γ-Proteobacteria), consistent with the
growing evidence that exposure to PFAS leads to an enrichment of
several bacterial phyla, mostly Proteobacteria, which are more
resistant to PFAS than other phyla.18,22,33 Negatively charged outer
membrane repelling negatively charged PFAS, an increased ability
to cope with oxidative damage and/or DNA damage, or even an
ability to extrude PFAS from cells or immobilize these compounds
in a biofilm are possible mechanisms of resistance of these
Enterobacteriaceae to PFAS.
From a mechanistic point of view, it is crucial to understand the
mechanisms by which the presence of PFASs can promote the
selection in the environment of bacteria resistant to antibiotics,
particularly fluoroquinolones. Fluoroquinolones target type II
bacterial topoisomerases and are widely used in the medical,
livestock and aquaculture sectors. The presence of fluoroquinolone
antibiotics is ubiquitous and poses a serious threat to
ecosystems.124 They are not readily biodegradable and can also

## Page 56

accumulate in soils and sediments due to their adsorption
properties.
Bacterial resistance to these compounds is due to multiple
mechanisms, including mutations in one or more of the genes
encoding the primary and secondary targets of these drugs (gyrA,
gyrB, parC, parE), the type II topoisomerases, permeability changes,
such as porin loss in Gram-negative bacteria or up-regulation of
chromosomal efflux systems (patAB, acrAB-tolC). Transmissible
fluoroquinolone-resistance is often associated with the acquirement
of plasmid-mediated quinolone resistance (PMQR) genes encoding
proteins that prevent binding of fluoroquinolones to type II
topoisomerases (qnrA, qnrB, qnrC, qnrD, qnrS), degrade (aac(6′)lbcr) or extrude (oqxAB and qepA efflux systems) the
fluoroquinolones.125 Possession of these resistance mechanisms
enables the survival of fluoroquinolone-resistant bacteria not only
in the infected host, but also in environments contaminated with
these antibiotics.
Among the ARGs present in both K. grimontii Tre-B and C. braakii
Tre-T, it can be noted the presence of the genes encoding AcrABTolC because this multidrug efflux pump confers resistance to a
wide range of toxic compounds such as antibiotics, surfactants,
dyes, detergents, and disinfectants which are not found in the
natural environment of these bacteria. Moreover, in C. braakii TreT, Resistance Gene Identifier predicted the presence of EmrAB-TolC
MSF efflux system, structurally similar to AcrAB-TolC, conferring
reduced susceptibility to a large variety of unrelated antimicrobial
compounds.126,127 In K. grimontii Tre-B the transmissible oqxAB
efflux system was also found. This system confers reduced
susceptibility to a multitude of substrates, including several
antibiotics (including quinolones, quinoxalines, tigecycline,
nitrofurantoin and chloramphenicol), detergents and disinfectants
(benzalkonium chloride, triclosan and sodium dodecyl
sulfate).128,129 Furthermore, in K. grimontii Tre-B, Resistance Gene
Identifier predicted the presence of the AdeFGH efflux system that
was associated with decreased susceptibility to many antibiotics,
like chloramphenicol and fluoroquinolones, and a number of
compounds, as well as biofilm formation in Acinetobacter
baumannii.130,131 Intriguingly, we found by RT-qPCR experiments

## Page 57

that the transcript levels of some of these genes (acrA, adeF, and
oqxA) increased when the Tre-B strain was grown in the presence of
PFOA. In particular, adeF and oqxA showed a dose-dependent
increase in transcript levels in response to PFOA exposure, whereas
acrA was upregulated only at low PFOA concentrations. This result
suggests that PFOA, in addition to selecting different groups of
microorganisms in polluted environments, may enhance antibiotic
resistance through upregulation of specific ARGs.
Furthermore, it would be interesting to investigate a possible
involvement of these efflux systems in PFAS resistance, to
understand if there is a mechanistic connection between these
chemicals and antibiotic resistance, particularly to
fluoroquinolones. In K. grimontii Tre-B, genes encoding the efflux
pump KpnGH of the MFS superfamily and KpnEF of the small
multidrug resistance (SMR) family were also identified. These
systems contribute to reduced susceptibility to a wide range of
antibiotics, dyes, detergents and disinfectants.132,133 In addition,
KpnEF, which is also present in C. braakii Tre-T, was directly
involved in capsule biogenesis in K. pneumoniae.132 Therefore,
some of these transport systems, including AdeFGH and KpnEF,
could mediate both antimicrobial and PFAS resistance by increasing
the production of exopolysaccharides and capsular polysaccharides.
This hypothesis could be validated by testing defective mutants in
these transport systems. It is also interesting to analyze the
expression of these transport systems in response to PFAS exposure.
In this regard, it is worth noting that the expression of the efflux
pump AcrAB-TolC was significantly up-regulated in E. coli strains
DH5α and HB101 exposed to PFOA.134 Consistent with this result,
we found by RT-qPCR experiments that the transcript levels of
kpnF, kpnG and acrA genes increased when bacteria were grown
with PFOA, and that in the case of kpnF and kpnG this increase was
striking and dependent on PFOA concentrations.
K. grimontii Tre-B and C. braakii Tre-T could be useful to further
understand the adaptive responses of bacteria to PFAS exposure by
transcriptomics studies, as was recently done using model strains of
E. coli.134 Notably, in E. coli, PFOA has been shown to induce
oxidative stress, enhance cell membrane permeability and promote
the excretion of extracellular polymeric substances.134 This latter

## Page 58

finding is consistent with our hypothesis that some of these
transport systems could mediate both antimicrobial and PFAS
resistance by increasing the production of extracellular polymers
and capsule.
An additional mechanism could be involved in the PFAS/
fluoroquinolone cross-resistance in polluted environment: resistance
to fluoride. As well as PFAS, fluoroquinolones are characterized by
the presence of fluorine atom, which forms an exceptionally strong
and highly polarized C–F bond, making them recalcitrant to
biodegradation.135 All biodegradation pathways of these
compounds involve their defluorination,124 with the release of
fluoride, a very toxic compound to microorganisms.26 Fluoride is
detrimental to biological systems mainly because of enzyme
inhibition. The electronegative F− effectively outcompetes
electronegative substrate groups, such as OH−, phosphate, or
carboxylate, for coordination by an enzyme-bound metal ion
causing broad-spectrum harm to many metabolic pathways.136,137
To survive in fluoroquinolone-contaminated environments,
fluoroquinolone-resistant bacteria must have the ability to resist
toxic fluoride. Fluoride is also released as a consequence of PFAS
defluorination, and is another important factor in determining PFAS
resistance.
Among the mechanisms of resistance to fluoride that some
microorganisms have evolved is the export of fluoride via the CLCF
family of F−/H+ antiporters.26 We found in both K. grimontii Tre-B
and C. braakii Tre-T the gene coding for fluoride ion transport
(crcB) (Fig. 8). This gene is localized in a conserved chromosome
region and co-occurs in Enterobacteriaceae with genes involved in
the biosynthesis of lipoic acid from octanoic acid, as well as with
genes involved in sulfur metabolism and oxidative stress (Fig. 8),
and it may be noted that in rat α-lipoic acid alleviate fluorideinduced damage to liver.138 α-Lipoic acid, a natural free radical
scavenger, alleviated fluoride-induced iron accumulation, increased
oxidative stress, and elevated lipid peroxidation in the liver.
Therefore, this gene locus could be involved in fluoride
detoxification, contributing to the ability of these bacteria to
survive in environments polluted by PFAS and/or fluoroquinolones.
It might be interesting to analyze the expression of this fluoride ion

## Page 59

transport in response to PFAS exposure.
In the same chromosome region, it may be also noted the presence
of two genes encoding specific and distinct oxidases in the two
bacteria: dsmA2 in K. grimontii Tre-B coding for a dimethyl
sulfoxide/trimethylamine N-oxide reductase, and ynfE1 in C.
braakii Tre-T coding for a putative dimethyl sulfoxide reductase.
Indeed, both bacteria were able to grow anaerobically using
dimethyl sulfoxide as terminal electron acceptor. Furthermore,
potential trimethylamine N-oxide reductase activity in K. grimontii
Tre-B is of particular interest because of the presence of the cut
BMC locus in the genome of this microorganism.
A characteristic of the K. grimontii Tre-B genome is the presence of
a resistance plasmid (named pKGTreB) containing many genes
involved in resistance to arsenic, copper, mercury and silver (Table
5). Especially the availability of information on a silver resistance
plasmid is useful due to the widespread antimicrobial use of silver
ions and nanoparticles against bacteria, fungi and viruses and the
need to gain further knowledge on silver ion and toxicity
mechanisms and nanoparticles.139 Furthermore, the observed crossresistance between resistance to silver and resistance to other heavy
metals and antibiotics in bacteria is also a clinically and
environmentally important issue.
Another feature of the K. grimontii Tre-B genome is the presence of
a large set of genes involved in the degradation of aromatic
compounds, including halogenated ones, and other recalcitrant
chemicals (Table 6). Many of these genes are also present in the
genome of C. braakii Tre-T (Table 7), including those involved in
the degradation of aliphatic sulfonates (dimethylsulfide and
methanesulfonate). The presence of the latter genes may be related
to the fact that the Trelleborg site has long been used industrially
for production of tires, and it is known that the rubber
vulcanization process involves the addition of a mixture of sulfur
and other additives, whose release into the environment may have
acted as a selection factor for local microbial communities.
Given the genomic and metabolic complexity of these bacteria, the
selection of antibiotic-resistant strains observed in our study may be
influenced by interactions between PFAS and other co-

## Page 60

contaminants. As reported in previous studies, such interactions can
facilitate the uptake, transport, and release of PFAS.74–76
Consequently, further research is needed to investigate these
dynamics in both polluted environments, such as the Trelleborg
landfill, and natural settings. Moreover, the microcosm and RTqPCR experiments were conducted using only PFOA. Additional
studies are therefore necessary to assess the effects of other PFAS
compounds.
Conclusion
Microbial communities are fundamental to maintaining ecosystem
health by driving key biogeochemical cycles, including carbon,
nitrogen, sulfur, and phosphorus cycles. Disruptions to these
communities, such as those caused by PFAS contamination, can
have significant consequences for microbial-mediated processes like
nutrient cycling, organic matter decomposition, and pollutant
degradation. In PFAS-contaminated environments, the altered
community structure—characterized by the dominance of PFASresistant strains such as Klebsiella grimontii and Citrobacter braakii
—signals a shift toward microbial populations that thrive under
high levels of environmental stress.
The enrichment of antibiotic-resistant bacteria (ARB) in these
ecosystems raises concerns about the broader impacts on ecosystem
function. This shift may suppress sensitive but ecologically vital
microbial species responsible for nutrient cycling and soil health,
leading to potential imbalances in nutrient availability and
ecosystem productivity. For example, nutrient-poor soils could
affect plant growth and disrupt higher trophic levels, ultimately
altering the structure of entire ecosystems.
Furthermore, the accumulation of antibiotic-resistant pathogens in
PFAS-contaminated environments may have cascading effects
throughout the food web. Wildlife exposed to contaminated water
or prey may face physiological and reproductive risks from both
PFAS and ARB exposure. The bioaccumulation of PFAS,
compounded by the spread of resistance genes via horizontal gene
transfer, could exacerbate these effects, increasing the spread of
multidrug resistance (MDR) within natural populations. The

## Page 61

simultaneous exposure to chemical pollutants and resistant
microbes creates a dual burden on wildlife and ecosystems,
complicating efforts to maintain biodiversity and ecosystem
resilience.
The ability of K. grimontii and C. braakii to resist both antibiotics
and environmental pollutants suggests that these bacteria could act
as vectors for the spread of multidrug resistance through soils,
sediments, and aquatic systems. The transfer of resistance genes to
other pathogenic or opportunistic bacteria via horizontal gene
transfer increases the risk of AMR beyond the immediate
contaminated sites. This has broad implications for public health, as
the global AMR crisis continues to be exacerbated by the spread of
environmental resistance, posing challenges for both animal and
human health.
These findings highlight the urgent need for comprehensive
environmental policies that address not only the chemical toxicity
of PFAS but also their role in promoting antimicrobial resistance.
Current regulatory frameworks focus largely on chemical
contaminants, with less attention to their ecological consequences.
Expanding these frameworks to include microbial community
monitoring in PFAS-contaminated sites could provide a more
holistic assessment of the long-term environmental and health risks
posed by these pollutants. Such policies would better account for
the complex interactions between chemical and biological factors
that drive resistance and ecosystem disruption.
Data availability
Sequencing data for this article are available under the following
accession numbers: PRJEB89934. The other data supporting this
article have been included as part of the ESI.†
Author contributions
Conceptualization: PA, FD; data curation: MC, MT, AG, DG, DR, CL,
AR; formal analysis: MC, AC, AG, CL, AR, MM, KEK; funding
acquisition: FD, PA; investigation: MC, PA, FD, AC, CL, KEK;

## Page 62

methodology: PA, MC, MT; project administration: PA, FD;
resources: KEK, FD, DR, DG; visualization: MC, AG; supervision: PA,
FD; writing – original draft: PA, MC, FD, CL; writing – review &
editing: PA, MC, FD, CL, AC, AR.
Conflicts of interest
There are no conflicts to declare.
Acknowledgements
This project has received funding from the European Union's
Horizon 2020 research and innovation program under grant
agreement no. 101037509 (SCENARIOS project).
References
1. C. F. Kwiatkowski, D. Q. Andrews, L. S. Birnbaum, T. A.
Bruton, J. C. DeWitt, D. R. U. Knappe, M. V. Maffini, M. F.
Miller, K. E. Pelch, A. Reade, A. Soehl, X. Trier, M. Venier, C.
C. Wagner, Z. Wang and A. Blum, Scientific Basis for
Managing PFAS as a Chemical Class, Environ. Sci. Technol.
Lett., 2020, 7, 532–543 CrossRef CAS PubMed.
2. J. Glüge, M. Scheringer, I. T. Cousins, J. C. DeWitt, G.
Goldenman, D. Herzke, R. Lohmann, C. A. Ng, X. Trier and Z.
Wang, An overview of the uses of per- and polyfluoroalkyl
substances (PFAS), Environ. Sci.: Processes Impacts, 2020, 22,
2345–2373 RSC.
3. R. Renner, Growing Concern Over Perfluorinated Chemicals,
Environ. Sci. Technol., 2001, 35, 154A–
160A CrossRef CAS PubMed.
4. C. Lau, K. Anitole, C. Hodes, D. Lai, A. Pfahles-Hutchens and
J. Seed, Perfluoroalkyl Acids: A Review of Monitoring and
Toxicological Findings, Toxicol. Sci., 2007, 99, 366–
394 CrossRef CAS PubMed.
5. A. M. Seacat, Subchronic Toxicity Studies on
Perfluorooctanesulfonate Potassium Salt in Cynomolgus
Monkeys, Toxicol. Sci., 2002, 68, 249–

## Page 63

264 CrossRef CAS PubMed.
6. V. Ehrlich, W. Bil, R. Vandebriel, B. Granum, M. Luijten, B.
Lindeman, P. Grandjean, A.-M. Kaiser, I. Hauzenberger, C.
Hartmann, C. Gundacker and M. Uhl, Consideration of
pathways for immunotoxicity of per- and polyfluoroalkyl
substances (PFAS), Environ. Health, 2023, 22,
19 CrossRef PubMed.
7. M. N. Ehsan, M. Riza, M. N. Pervez, M. M. O. Khyum, Y.
Liang and V. Naddeo, Environmental and health impacts of
PFAS: Sources, distribution and sustainable management in
North Carolina (USA), Sci. Total Environ., 2023, 878,
163123 CrossRef CAS PubMed.
8. G. L. Kennedy, J. L. Butenhoff, G. W. Olsen, J. C. O'Connor, A.
M. Seacat, R. G. Perkins, L. B. Biegel, S. R. Murphy and D. G.
Farrar, The Toxicology of Perfluorooctanoate, Crit. Rev.
Toxicol., 2004, 34, 351–384 CrossRef CAS PubMed.
9. U. M. Ismail, H. Elnakar and M. F. Khan, Sources, Fate, and
Detection of Dust-Associated Perfluoroalkyl and
Polyfluoroalkyl Substances (PFAS): A Review, Toxics, 2023,
11, 335 CrossRef CAS PubMed.
10. C. Lau, J. L. Butenhoff and J. M. Rogers, The developmental
toxicity of perfluoroalkyl acids and their derivatives, Toxicol.
Appl. Pharmacol., 2004, 198, 231–241 CrossRef CAS PubMed.
11. Y. Manojkumar, S. Pilli, P. V. Rao and R. D. Tyagi, Sources,
occurrence and toxic effects of emerging per- and
polyfluoroalkyl substances (PFAS), Neurotoxicol. Teratol.,
2023, 97, 107174 CrossRef CAS PubMed.
12. J. Sun, L. Xing and J. Chu, Global ocean contamination of
per- and polyfluoroalkyl substances: A review of seabird
exposure, Chemosphere, 2023, 330,
138721 CrossRef CAS PubMed.
13. Z. Sun, Y. Wen, B. Wang, S. Deng, F. Zhang, Z. Fu, Y. Yuan
and D. Zhang, Toxic effects of per- and polyfluoroalkyl
substances on sperm: Epidemiological and experimental
evidence, Front. Endocrinol., 2023, 14,
1114463 CrossRef PubMed.
14. Z.-J. Wen, Y.-J. Wei, Y.-F. Zhang and Y.-F. Zhang, A review of
cardiovascular effects and underlying mechanisms of legacy
and emerging per- and polyfluoroalkyl substances (PFAS),
Arch. Toxicol., 2023, 97, 1195–1245 CrossRef CAS PubMed.

## Page 64

15. X. Zhang, J. A. Flaws, M. J. Spinella and J. Irudayaraj, The
Relationship between Typical Environmental Endocrine
Disruptors and Kidney Disease, Toxics, 2022, 11,
32 CrossRef PubMed.
16. The Forever Pollution Project – Journalists tracking PFAS
across Europe, https://foreverpollution.eu/, accessed 17
September 2024.
17. S. Huang and P. R. Jaffé, Defluorination of Perfluorooctanoic
Acid (PFOA) and Perfluorooctane Sulfonate (PFOS) by
Acidimicrobium sp. Strain A6, Environ. Sci. Technol., 2019,
53, 11410–11419 CrossRef CAS PubMed.
18. Y. Bao, B. Li, S. Xie and J. Huang, Vertical profiles of
microbial communities in perfluoroalkyl substancecontaminated soils, Ann. Microbiol., 2018, 68, 399–
408 CrossRef CAS.
19. Y. Ke, J. Chen, X. Hu, T. Tong, J. Huang and S. Xie, Emerging
perfluoroalkyl substance impacts soil microbial community
and ammonia oxidation, Environ. Pollut., 2020, 257,
113615 CrossRef CAS PubMed.
20. Y. Sun, T. Wang, X. Peng, P. Wang and Y. Lu, Bacterial
community compositions in sediment polluted by
perfluoroalkyl acids (PFAAs) using Illumina high-throughput
sequencing, Environ. Sci. Pollut. Res., 2016, 23, 10556–
10565 CrossRef CAS PubMed.
21. M. Sun, E. Arevalo, M. Strynar, A. Lindstrom, M. Richardson,
B. Kearns, A. Pickett, C. Smith and D. R. U. Knappe, Legacy
and Emerging Perfluoroalkyl Substances Are Important
Drinking Water Contaminants in the Cape Fear River
Watershed of North Carolina, Environ. Sci. Technol. Lett.,
2016, 3, 415–419 CrossRef CAS.
22. W. Qiao, Z. Xie, Y. Zhang, X. Liu, S. Xie, J. Huang and L. Yu,
Perfluoroalkyl substances (PFASs) influence the structure and
function of soil bacterial community: Greenhouse experiment,
Sci. Total Environ., 2018, 642, 1118–
1126 CrossRef CAS PubMed.
23. D. Zhang, W. Zhang and Y. Liang, Distribution of eight
perfluoroalkyl acids in plant-soil-water systems and their
effect on the soil microbial community, Sci. Total Environ.,
2019, 697, 134146 CrossRef CAS PubMed.
24. D. Zhang, W. Zhang and Y. Liang, Bacterial community in a

## Page 65

freshwater pond responding to the presence of
perfluorooctanoic acid (PFOA), Environ. Technol., 2020, 41,
3646–3656 CrossRef CAS PubMed.
25. L. P. Wackett, Nothing lasts forever: understanding microbial
biodegradation of polyfluorinated compounds and
perfluorinated alkyl substances, Microb. Biotechnol., 2022,
15, 773–792 CrossRef CAS PubMed.
26. B. C. McIlwain, M. T. Ruprecht and R. B. Stockbridge,
Membrane Exporters of Fluoride Ion, Annu. Rev. Biochem.,
2021, 90, 559–579 CrossRef CAS PubMed.
27. E. Shahsavari, D. Rouch, L. S. Khudur, D. Thomas, A. AburtoMedina and A. S. Ball, Challenges and Current Status of the
Biological Treatment of PFAS-Contaminated Soils, Front.
Bioeng. Biotechnol., 2021, 8, 602040 CrossRef PubMed.
28. L. Philippot, C. Chenu, A. Kappler, M. C. Rillig and N. Fierer,
The interplay between microbial communities and soil
properties, Nat. Rev. Microbiol., 2024, 22, 226–
239 CrossRef CAS PubMed.
29. P. Liu, S. Wen, S. Zhu, X. Hu and Y. Wang, Microbial
Degradation of Soil Organic Pollutants: Mechanisms,
Challenges, and Advances in Forest Ecosystem Management,
Processes, 2025, 13, 916 CrossRef CAS.
30. D. V. Murphy, E. A. Stockdale, P. C. Brookes and K. W. T.
Goulding, in Soil Biological Fertility, ed. L. K. Abbott and D.
V. Murphy, Springer Netherlands, Dordrecht, 2004, pp. 37–
59 Search PubMed.
31. W. Qiao, Z. Xie, Y. Zhang, X. Liu, S. Xie, J. Huang and L. Yu,
Perfluoroalkyl substances (PFASs) influence the structure and
function of soil bacterial community: Greenhouse experiment,
Sci. Total Environ., 2018, 642, 1118–
1126 CrossRef CAS PubMed.
32. S. N. Davis, S. M. Klumker, A. A. Mitchell, M. A. Coppage, J.
M. Labonté and A. Quigg, Life in the PFAS lane: The impact
of perfluoroalkyl substances on photosynthesis, cellular
exudates, nutrient cycling, and composition of a marine
microbial community, Sci. Total Environ., 2024, 927,
171977 CrossRef CAS PubMed.
33. Y. Cai, H. Chen, R. Yuan, F. Wang, Z. Chen and B. Zhou,
Metagenomic analysis of soil microbial community under
PFOA and PFOS stress, Environ. Res., 2020, 188,

## Page 66

109838 CrossRef CAS PubMed.
34. C. Chen, Y. Fang, X. Cui and D. Zhou, Effects of trace PFOA
on microbial community and metabolisms: Microbial
selectivity, regulations and risks, Water Res., 2022, 226,
119273 CrossRef CAS PubMed.
35. C. Liu, X. Zhu, L. You, K. Y.-H. Gin, H. Chen and B. Chen,
Per/polyfluoroalkyl substances modulate plasmid transfer of
antibiotic resistance genes: A balance between oxidative
stress and energy support, Water Res., 2023, 240,
120086 CrossRef CAS PubMed.
36. Z. Xu, J. Xiong, C. Li, S. Hu, Z. Li, Y. Ma, S. Li, B. Huang, X.
Ren and X. Pan, Environmental concentrations of per/
polyfluoroalkyl substances promote the conjugative transfer
of antibiotic resistance genes, Chem. Eng. J., 2024, 498,
155500 CrossRef CAS.
37. Y. Li, Y. Zhang, X. Liu, X. Zhou, T. Ye, Q. Fu, M. Du, Q. Lu, Y.
Zheng and D. Wang, Per- and polyfluoroalkyl substances
exacerbate the prevalence of plasmid-borne antibiotic
resistance genes by enhancing natural transformation, in vivo
stability, and expression in bacteria, Water Res., 2025, 272,
122972 CrossRef CAS PubMed.
38. E. Abbasi Montazeri, A. D. Khosravi, M. Saki, M. Sirous, B.
Keikhaei and S. Seyed-Mohammadi, Prevalence of ExtendedSpectrum Beta-Lactamase-Producing Enterobacteriaceae
Causing Bloodstream Infections in Cancer Patients from
Southwest of Iran, Infect. Drug Resist., 2020, 13, 1319–
1326 CrossRef PubMed.
39. A. Farajzadeh Sheikh, M. Moradi Bandbal and M. Saki,
Emergence of multidrug-resistant Shigella species harboring
extended-spectrum beta-lactamase genes in pediatric patients
with diarrhea from southwest of Iran, Mol. Biol. Rep., 2020,
47, 7097–7106 CrossRef CAS PubMed.
40. K. Garbacz, M. Wierzbowska, E. Kwapisz, M. Kosecka-Strojek,
M. Bronk, M. Saki and J. Międzobrodzki, Distribution and
antibiotic-resistance of different Staphylococcus species
identified by matrix assisted laser desorption ionization-time
of flight mass spectrometry (MALDI-TOF MS) isolated from
the oral cavity, J. Oral Microbiol., 2021, 13,
1983322 CrossRef PubMed.
41. WHO, Antimicrobial resistance, https://www.who.int/health-

## Page 67

topics/antimicrobial-resistance, accessed 13 June 2023.
42. EU, EU Action on Antimicrobial Resistance, European
Commission, https://health.ec.europa.eu/antimicrobialresistance/eu-action-antimicrobial-resistance_en, accessed 13
June 2023.
43. Council of The European Union, Council Recommendation on
Stepping up EU Actions to Combat Antimicrobial Resistance
in a One Health Approach 2023/C 220/01,
2023 Search PubMed.
44. J. O'Neill and Grande-Bretagne, Antimicrobial Resistance:
Tackling a Crisis for the Health and Wealth of Nations,
Review on Antimicrobial Resistance, 2014 Search PubMed.
45. J. O'Neill, Tackling Drug-Resistant Infections Globally: Final
Report and Recommendations, Government of the United
Kingdom, 2016 Search PubMed.
46. NOAH, NOAH responds to the O'Neill review, ProQuest,
https://www.proquest.com/docview/1809928859?
parentSessionId=fiAy%2Bi4DqmXU9DIR5m2Y5VVWdf1bVCKLk9ML
accessed 16 September 2024.
47. M. Gao, Q. Zhang, C. Lei, T. Lu and H. Qian, Atmospheric
antibiotic resistome driven by air pollutants, Sci. Total
Environ., 2023, 902, 165942 CrossRef CAS PubMed.
48. A. H. Holmes, L. S. P. Moore, A. Sundsfjord, M. Steinbakk, S.
Regmi, A. Karkey, P. J. Guerin and L. J. V. Piddock,
Understanding the mechanisms and drivers of antimicrobial
resistance, Lancet, 2016, 387, 176–
187 CrossRef CAS PubMed.
49. R. Magnano San Lio, G. Favara, A. Maugeri, M. Barchitta and
A. Agodi, How Antimicrobial Resistance Is Linked to Climate
Change: An Overview of Two Intertwined Global Challenges,
Int. J. Environ. Res. Publ. Health, 2023, 20,
1681 CrossRef PubMed.
50. G. Pizzolante, C. Cordero, S. M. Tredici, D. Vergara, P.
Pontieri, L. Del Giudice, A. Capuzzo, P. Rubiolo, C. N.
Kanchiswamy, S. A. Zebelo, C. Bicchi, M. E. Maffei and P.
Alifano, Cultivable gut bacteria provide a pathway for
adaptation of Chrysolina herbacea to Mentha aquatica
volatiles, BMC Plant Biol., 2017, 17, 30 CrossRef PubMed.
51. M. Calcagnile, I. Jeguirim, S. M. Tredici, F. Damiano and P.
Alifano, Spiramycin Disarms Pseudomonas aeruginosa

## Page 68

without Inhibiting Growth, Antibiotics, 2023, 12,
499 CrossRef CAS PubMed.
52. E. P. Goldschmidt, M. S. Cater, T. S. Matney, M. Ann Butler
and A. Greene, Genetic Analysis Of The Histidine Operon In
Escherichia Coli K12, Genetics, 1970, 66, 219–
229 CrossRef CAS PubMed.
53. T. Kasai, Regulation of the expression of the histidine operon
in Salmonella typhimurium, Nature, 1974, 249, 523–
527 CrossRef CAS PubMed.
54. D. Vinella and R. D'Ari, Thermoinducible filamentation in
Escherichia coli due to an altered RNA polymerase beta
subunit is suppressed by high levels of ppGpp, J. Bacteriol.,
1994, 176, 966–972 CrossRef CAS PubMed.
55. A. Talà, M. Calcagnile, S. C. Resta, A. Pennetta, G. E. De
Benedetto and P. Alifano, Thiostrepton, a resurging drug
inhibiting the stringent response to counteract antibioticresistance and expression of virulence determinants in
Neisseria gonorrhoeae, Front. Microbiol., 2023, 14,
1104454 CrossRef PubMed.
56. A. Talà, A. Buccolieri, M. Calcagnile, G. Ciccarese, M.
Onorato, R. Onorato, A. Serra, F. Spedicato, S. M. Tredici, P.
Alifano and G. Belmonte, Chemotrophic profiling of
prokaryotic communities thriving on organic and mineral
nutrients in a submerged coastal cave, Sci. Total Environ.,
2021, 755, 142514 CrossRef PubMed.
57. T. Z. DeSantis, P. Hugenholtz, N. Larsen, M. Rojas, E. L.
Brodie, K. Keller, T. Huber, D. Dalevi, P. Hu and G. L.
Andersen, Greengenes, a Chimera-Checked 16S rRNA Gene
Database and Workbench Compatible with ARB, Appl.
Environ. Microbiol., 2006, 72, 5069–
5072 CrossRef CAS PubMed.
58. S. Andrews, FastQC A Quality Control tool for High
Throughput Sequence Data, https://
www.bioinformatics.babraham.ac.uk/projects/fastqc/,
accessed 16 September 2024.
59. A. V. Zimin, G. Marçais, D. Puiu, M. Roberts, S. L. Salzberg
and J. A. Yorke, The MaSuRCA genome assembler,
Bioinformatics, 2013, 29, 2669–2677 CrossRef CAS PubMed.
60. F. A. Simão, R. M. Waterhouse, P. Ioannidis, E. V. Kriventseva
and E. M. Zdobnov, BUSCO: assessing genome assembly and

## Page 69

annotation completeness with single-copy orthologs,
Bioinformatics, 2015, 31, 3210–3212 CrossRef PubMed.
61. A. Gurevich, V. Saveliev, N. Vyahhi and G. Tesler, QUAST:
quality assessment tool for genome assemblies,
Bioinformatics, 2013, 29, 1072–1075 CrossRef CAS PubMed.
62. T. Seemann, Prokka: rapid prokaryotic genome annotation,
Bioinformatics, 2014, 30, 2068–2069 CrossRef CAS PubMed.
63. Y. Tanizawa, T. Fujisawa and Y. Nakamura, DFAST: a flexible
prokaryotic genome annotation pipeline for faster genome
publication, Bioinformatics, 2018, 34, 1037–
1039 CrossRef CAS PubMed.
64. B. P. Alcock, W. Huynh, R. Chalil, K. W. Smith, A. R.
Raphenya, M. A. Wlodarski, A. Edalatmand, A. Petkau, S. A.
Syed, K. K. Tsang, S. J. C. Baker, M. Dave, M. C. McCarthy, K.
M. Mukiri, J. A. Nasir, B. Golbon, H. Imtiaz, X. Jiang, K. Kaur,
M. Kwong, Z. C. Liang, K. C. Niu, P. Shan, J. Y. J. Yang, K. L.
Gray, G. R. Hoad, B. Jia, T. Bhando, L. A. Carfrae, M. A.
Farha, S. French, R. Gordzevich, K. Rachwalski, M. M. Tu, E.
Bordeleau, D. Dooley, E. Griffiths, H. L. Zubyk, E. D. Brown,
F. Maguire, R. G. Beiko, W. W. L. Hsiao, F. S. L. Brinkman, G.
Van Domselaar and A. G. McArthur, CARD 2023: expanded
curation, support for machine learning, and resistome
prediction at the Comprehensive Antibiotic Resistance
Database, Nucleic Acids Res., 2023, 51, D690–
D699 CrossRef CAS PubMed.
65. S. Cosentino, M. Voldby Larsen, F. Møller Aarestrup and O.
Lund, Correction: PathogenFinder - Distinguishing Friend
from Foe Using Bacterial Whole Genome Sequence Data, PLoS
One, 2013, 8(12) DOI:10.1371/annotation/b84e1af7-c127-45c3be22-76abd977600f.
66. S. Cosentino, M. Voldby Larsen, F. Møller Aarestrup and O.
Lund, PathogenFinder - Distinguishing Friend from Foe Using
Bacterial Whole Genome Sequence Data, PLoS One, 2013, 8,
e77302 CrossRef CAS PubMed.
67. K. G. Joensen, F. Scheutz, O. Lund, H. Hasman, R. S. Kaas, E.
M. Nielsen and F. M. Aarestrup, Real-Time Whole-Genome
Sequencing for Routine Typing, Surveillance, and Outbreak
Detection of Verotoxigenic Escherichia coli, J. Clin.
Microbiol., 2014, 52, 1501–1510 CrossRef PubMed.
68. K. Blin, S. Shaw, H. E. Augustijn, Z. L. Reitz, F. Biermann, M.

## Page 70

Alanjary, A. Fetter, B. R. Terlouw, W. W. Metcalf, E. J. N.
Helfrich, G. P. van Wezel, M. H. Medema and T. Weber,
antiSMASH 7.0: new and improved predictions for detection,
regulation, chemical structures and visualisation, Nucleic
Acids Res., 2023, 51, W46–W50 CrossRef CAS PubMed.
69. M. H. K. Johansson, V. Bortolaia, S. Tansirichaiya, F. M.
Aarestrup, A. P. Roberts and T. N. Petersen, Detection of
mobile genetic elements associated with antibiotic resistance
in Salmonella enterica using a newly developed web tool:
MobileElementFinder, J. Antimicrob. Chemother., 2021, 76,
101–109 CrossRef CAS PubMed.
70. R. Zallot, N. Oberg and J. A. Gerlt, The EFI Web Resource for
Genomic Enzymology Tools: Leveraging Protein, Genome,
and Metagenome Databases to Discover Novel Enzymes and
Metabolic Pathways, Biochemistry, 2019, 58, 4169–
4182 CrossRef CAS PubMed.
71. N. Oberg, R. Zallot and J. A. Gerlt, EFI-EST, EFI-GNT, and
EFI-CGFP: Enzyme Function Initiative (EFI) Web Resource for
Genomic Enzymology Tools, J. Mol. Biol., 2023, 435,
168018 CrossRef CAS PubMed.
72. G. Su, J. H. Morris, B. Demchak and G. D. Bader, Biological
Network Exploration with Cytoscape 3, Curr. Protoc. Bioinf.,
2014, 47, 1–24 Search PubMed.
73. Y. Xie, H. Li, X. Luo, H. Li, Q. Gao, L. Zhang, Y. Teng, Q.
Zhao, Z. Zuo and J. Ren, IBS 2.0: an upgraded illustrator for
the visualization of biological sequences, Nucleic Acids Res.,
2022, 50, W420–W426 CrossRef CAS PubMed.
74. S. Y. Wee and A. Z. Aris, Revisiting the “forever chemicals”,
PFOA and PFOS exposure in drinking water, npj Clean Water,
2023, 6, 57 CrossRef CAS.
75. Y. Wang and K. D. Good, Microplastics and PFAS air-water
interaction and deposition, Sci. Total Environ., 2024, 954,
176247 CrossRef CAS PubMed.
76. T. A. Bruton and D. L. Sedlak, Treatment of Aqueous FilmForming Foam by Heat-Activated Persulfate Under Conditions
Representative of In Situ Chemical Oxidation, Environ. Sci.
Technol., 2017, 51, 13878–13885 CrossRef CAS PubMed.
77. W. Cai, D. A. Navarro, J. Du, P. Srivastava, Z. Cao, G. Ying
and R. S. Kookana, Effect of heavy metal co-contaminants on
the sorption of thirteen anionic per- and poly-fluoroalkyl

## Page 71

substances (PFAS) in soils, Sci. Total Environ., 2023, 905,
167188 CrossRef CAS PubMed.
78. G. A. Zavarzin, Winogradsky and modern microbiology,
Microbiology, 2006, 75, 501–511 CAS.
79. V. Passet and S. Brisse, Description of Klebsiella grimontii sp.
nov, Int. J. Syst. Evol. Microbiol., 2018, 68, 377–
381 CrossRef CAS PubMed.
80. C. A. Broberg, M. Palacios and V. L. Miller, Klebsiella: a long
way to go towards understanding this enigmatic jet-setter,
F1000Prime Rep., 2014, 6, 64,  DOI:10.12703/P6-64.
81. N. Barnich, M.-A. Bringer, L. Claret and A. DarfeuilleMichaud, Involvement of Lipoprotein NlpI in the Virulence of
Adherent Invasive Escherichia coli Strain LF82 Isolated from
a Patient with Crohn's Disease, Infect. Immun., 2004, 72,
2484–2493 CrossRef CAS PubMed.
82. K. L. Stewart, A. M. Stewart and T. A. Bobik, Prokaryotic
Organelles: Bacterial Microcompartments in E. coli and
Salmonella, EcoSal Plus, 2020, 9(1) DOI:10.1128/
ecosalplus.ESP-0025–2019.
83. B. Ferlez, M. Sutter and C. A. Kerfeld, Glycyl Radical EnzymeAssociated Microcompartments: Redox-Replete Bacterial
Organelles, mBio, 2019, 10, e02327 CrossRef CAS PubMed.
84. C. A. Huffine, L. C. Wheeler, B. Wing and J. C. Cameron,
Computational modeling and evolutionary implications of
biochemical reactions in bacterial microcompartments, Curr.
Opin. Microbiol., 2022, 65, 15–23 CrossRef CAS PubMed.
85. T. I. Herring, T. N. Harris, C. Chowdhury, S. K. Mohanty and
T. A. Bobik, A Bacterial Microcompartment Is Used for
Choline Fermentation by Escherichia coli 536, J. Bacteriol.,
2018, 200(10), e00764 CrossRef CAS PubMed.
86. E. Jameson, T. Fu, I. R. Brown, K. Paszkiewicz, K. J. Purdy, S.
Frank and Y. Chen, Anaerobic choline metabolism in
microcompartments promotes growth and swarming of P
roteus mirabilis, Environ. Microbiol., 2016, 18, 2886–
2898 CrossRef CAS PubMed.
87. G. Kalnins, J. Kuka, S. Grinberga, M. Makrecka-Kuka, E.
Liepinsh, M. Dambrova and K. Tars, Structure and Function of
CutC Choline Lyase from Human Microbiota Bacterium
Klebsiella pneumoniae, J. Biol. Chem., 2015, 290, 21732–
21740 CrossRef CAS PubMed.

## Page 72

88. G. Kalnins, E.-E. Cesle, J. Jansons, J. Liepins, A. Filimonenko
and K. Tars, Encapsulation mechanisms and structural studies
of GRM2 bacterial microcompartment particles, Nat.
Commun., 2020, 11, 388 CrossRef CAS PubMed.
89. Y. Chen, Y. Liu, R. Zhou, X. Chen, C. Wang, X. Tan, L. Wang,
R. Zheng, H. Zhang, W. Ling and H. Zhu, Associations of gutflora-dependent metabolite trimethylamine-N-oxide, betaine
and choline with non-alcoholic fatty liver disease in adults,
Sci. Rep., 2016, 6, 19076 CrossRef CAS PubMed.
90. M. Trøseid, T. Ueland, J. R. Hov, A. Svardal, I. Gregersen, C.
P. Dahl, S. Aakhus, E. Gude, B. Bjørndal, B. Halvorsen, T. H.
Karlsen, P. Aukrust, L. Gullestad, R. K. Berge and A.
Yndestad, Microbiota-dependent metabolite trimethylamineN-oxide is associated with disease severity and survival of
patients with chronic heart failure, J. Intern. Med., 2015,
277, 717–726 CrossRef PubMed.
91. W. H. W. Tang, Z. Wang, B. S. Levison, R. A. Koeth, E. B.
Britt, X. Fu, Y. Wu and S. L. Hazen, Intestinal Microbial
Metabolism of Phosphatidylcholine and Cardiovascular Risk,
N. Engl. J. Med., 2013, 368, 1575–
1584 CrossRef CAS PubMed.
92. W. H. W. Tang, Z. Wang, D. J. Kennedy, Y. Wu, J. A. Buffa, B.
Agatisa-Boyle, X. S. Li, B. S. Levison and S. L. Hazen, Gut
Microbiota-Dependent Trimethylamine N -Oxide (TMAO)
Pathway Contributes to Both Development of Renal
Insufficiency and Mortality Risk in Chronic Kidney Disease,
Circ. Res., 2015, 116, 448–455 CrossRef CAS PubMed.
93. W. H. W. Tang, Z. Wang, X. S. Li, Y. Fan, D. S. Li, Y. Wu and
S. L. Hazen, Increased Trimethylamine N-Oxide Portends
High Mortality Risk Independent of Glycemic Control in
Patients with Type 2 Diabetes Mellitus, Clin. Chem., 2017,
63, 297–306 CrossRef CAS PubMed.
94. P. Kaufmann, B. R. Duffus, B. Mitrova, C. Iobbi-Nivol, C.
Teutloff, M. Nimtz, L. Jänsch, U. Wollenberger and S.
Leimkühler, Modulating the Molybdenum Coordination
Sphere of Escherichia coli Trimethylamine N -Oxide
Reductase, Biochemistry, 2018, 57, 1130–
1143 CrossRef CAS PubMed.
95. K. Nagórska, A. Ostrowski, K. Hinc, I. B. Holland and M.
Obuchowski, Importance ofeps genes from Bacillus subtilis in

## Page 73

biofilm formation and swarming, J. Appl. Genet., 2010, 51,
369–381 CrossRef PubMed.
96. D. J. Brenner, P. A. D. Grimont, A. G. Steigerwalt, G. R.
Fanning, E. Ageron and C. F. Riddle, Classification of
Citrobacteria by DNA Hybridization: Designation of
Citrobacter farmeri sp. nov., Citrobacter youngae sp. nov.,
Citrobacter braakii sp. nov., Citrobacter werkmanii sp. nov.,
Citrobacter sedlakii sp. nov., and Three Unnamed Citrobacter
Genomospecies, Int. J. Syst. Bacteriol., 1993, 43, 645–
658 CrossRef CAS PubMed.
97. H. Han, Z. Zhao, Y. Lin, B. Lin, H. Xu and B. Zheng, CoProduction of NDM-1 and OXA-10 β-Lactamase in Citrobacter
braakii Strain Causing Urinary Tract Infection, Infect. Drug
Resist., 2022, 15, 1127–1133 CrossRef CAS PubMed.
98. A. Shnaiderman-Torban, S. Navon-Venezia, H. Baron, W. AbuAhmad, H. Arielly, G. Zizelski Valenci, I. Nissan, Y. Paitan
and A. Steinman, Prevalence and Molecular Characterization
of Extended-Spectrum β-Lactamase Producing
Enterobacterales in Healthy Community Dogs in Israel,
Antibiotics, 2022, 11, 1069 CrossRef CAS PubMed.
99. A. L. Colclough, I. Alav, E. E. Whittle, H. L. Pugh, E. M.
Darby, S. W. Legood, H. E. McNeil and J. M. Blair, RND Efflux
Pumps in Gram-Negative Bacteria; Regulation, Structure and
Role in Antibiotic Resistance, Future Microbiol., 2020, 15,
143–157 CrossRef CAS PubMed.
100. L. Hadchity, J. Houard, A. Lanois, A. Payelleville, F. Nassar,
M. Gualtieri, A. Givaudan and Z. Abi Khattar, The AcrAB
efflux pump confers self-resistance to stilbenes in
Photorhabdus laumondii, Res. Microbiol., 2023, 174,
104081 CrossRef CAS PubMed.
101. A. Pérez, M. Poza, A. Fernández, M. Del Carmen Fernández,
S. Mallo, M. Merino, S. Rumbo-Feal, M. P. Cabral and G. Bou,
Involvement of the AcrAB-TolC Efflux Pump in the
Resistance, Fitness, and Virulence of Enterobacter cloacae,
Antimicrob. Agents Chemother., 2012, 56, 2084–
2090 CrossRef PubMed.
102. I. Johnston, L. J. Osborn, R. L. Markley, E. A. McManus, A.
Kadam, K. B. Schultz, N. Nagajothi, P. P. Ahern, J. M. Brown
and J. Claesen, Identification of essential genes for
Escherichia coli aryl polyene biosynthesis and function in

## Page 74

biofilm formation, npj Biofilms Microbiomes, 2021, 7,
56 CrossRef CAS PubMed.
103. F. Prestinaci, P. Pezzotti and A. Pantosti, Antimicrobial
resistance: a global multifaceted phenomenon, Pathog. Global
Health, 2015, 109, 309–318 CrossRef PubMed.
104. M. Irfan, A. Almotiri and Z. A. AlZeyadi, Antimicrobial
Resistance and Its Drivers—A Review, Antibiotics, 2022, 11,
1362 CrossRef CAS PubMed.
105. V. M. D'Costa, C. E. King, L. Kalan, M. Morar, W. W. L. Sung,
C. Schwarz, D. Froese, G. Zazula, F. Calmels, R. Debruyne, G.
B. Golding, H. N. Poinar and G. D. Wright, Antibiotic
resistance is ancient, Nature, 2011, 477, 457–
461 CrossRef PubMed.
106. K. Hwengwere, H. Paramel Nair, K. A. Hughes, L. S. Peck, M.
S. Clark and C. A. Walker, Antimicrobial resistance in
Antarctica: is it still a pristine environment?, Microbiome,
2022, 10, 71 CrossRef CAS PubMed.
107. L. C. Scott, N. Lee and T. G. Aw, Antibiotic Resistance in
Minimally Human-Impacted Environments, Int. J. Environ.
Res. Public Health, 2020, 17, 3939 CrossRef PubMed.
108. C. Ejikeugwu, O. Nworie, M. Saki, H. O. M. Al-Dahmoshi, N.
S. K. Al-Khafaji, C. Ezeador, E. Nwakaeze, P. Eze, E. Oni, C.
Obi, I. Iroha, C. Esimone and M. U. Adikwu, Metallo-βlactamase and AmpC genes in Escherichia coli, Klebsiella
pneumoniae, and Pseudomonas aeruginosa isolates from
abattoir and poultry origin in Nigeria, BMC Microbiol., 2021,
21, 124 CrossRef CAS PubMed.
109. M. Saki, A. F. Sheikh, S. Seyed-Mohammadi, A. A. Z. Dezfuli,
M. Shahin, M. Tabasi, H. Veisi, R. Keshavarzi and P. Khani,
Publisher Correction: Occurrence of plasmid-mediated
quinolone resistance genes in Pseudomonas aeruginosa strains
isolated from clinical specimens in southwest Iran: a
multicentral study, Sci. Rep., 2022, 12,
3817 CrossRef CAS PubMed.
110. F. Abbasian, R. Lockington, M. Mallavarapu and R. Naidu, A
pyrosequencing-based analysis of microbial diversity
governed by ecological conditions in the Winogradsky
column, World J. Microbiol. Biotechnol., 2015, 31, 1115–
1126 CrossRef CAS PubMed.
111. R. Sridharan, M. Vetriselvan, V. G. Krishnaswamy, S. Jansi R,

## Page 75

H. Rishin, T. Kumar D and G. P. Doss C, Integrated approach
in LDPE degradation – An application using Winogradsky
column, computational modeling, and pathway prediction, J.
Hazard. Mater., 2021, 412, 125336 CrossRef CAS PubMed.
112. S. Brisse, F. Grimont and P. A. D. Grimont, in The
Prokaryotes, ed. M. Dworkin, S. Falkow, E. Rosenberg, K.-H.
Schleifer and E. Stackebrandt, Springer New York, New York,
NY, 2006, pp. , pp. 159–196 Search PubMed.
113. M. K. Paczosa and J. Mecsas, Klebsiella pneumoniae: Going
on the Offense with a Strong Defense, Microbiol. Mol. Biol.
Rev., 2016, 80, 629–661 CrossRef CAS PubMed.
114. T. I. Doran, The Role of Citrobacter in Clinical Disease of
Children: Review, Clin. Infect. Dis., 1999, 28, 384–394 CAS.
115. G. Samonis, D. E. Karageorgopoulos, D. P. Kofteridis, D. K.
Matthaiou, V. Sidiropoulou, S. Maraki and M. E. Falagas,
Citrobacter infections in a general hospital: characteristics
and outcomes, Eur. J. Clin. Microbiol. Infect. Dis., 2009, 28,
61–68 CrossRef CAS PubMed.
116. A. Nawaz, F. Mubeen, Z. U. Qamar, M. U. Marghoob, S. Aziz
and H. Gross, Draft Genome Sequence of the Halophilic Strain
Citrobacter braakii AN-PRR1, Isolated from Rhizospheric Soil
of Rice (Oryza sativa L.) from Pakistan, Microbiol. Resour.
Announce., 2021, 10, e00787 CAS.
117. R. Gupta, S. J. Rauf, S. Singh, J. Smith and M. L. Agraharkar,
Sepsis in a Renal Transplant Recipient due to Citrobacter
braakii, South. Med. J., 2003, 96, 796–798 CrossRef PubMed.
118. E. Tollkuci and R. Myers, Citrobacter braakii CLABSI in a
hematopoietic stem cell transplant patient, J. Oncol. Pharm.
Pract., 2021, 27, 1792–1794 CrossRef CAS PubMed.
119. V. Prasanna, R. Rana, D. K. Daunaria and N. B. Patel,
Bacteremia due to carbapenem-resistant Citrobacter braakii,
J. Fam. Med. Prim. Care, 2022, 11, 3395 CrossRef PubMed.
120. M. O. A. Sommer, G. Dantas and G. M. Church, Functional
Characterization of the Antibiotic Resistance Reservoir in the
Human Microflora, Science, 2009, 325, 1128–
1131 CrossRef CAS PubMed.
121. S. P. Brown, D. M. Cornforth and N. Mideo, Evolution of
virulence in opportunistic pathogens: generalism, plasticity,
and control, Trends Microbiol., 2012, 20, 336–
342 CrossRef CAS PubMed.

## Page 76

122. A. Salyers, A. Gupta and Y. Wang, Human intestinal bacteria
as reservoirs for antibiotic resistance genes, Trends
Microbiol., 2004, 12, 412–416 CrossRef CAS PubMed.
123. C. J. H. Von Wintersdorff, J. Penders, J. M. Van Niekerk, N.
D. Mills, S. Majumder, L. B. Van Alphen, P. H. M. Savelkoul
and P. F. G. Wolffs, Dissemination of Antimicrobial Resistance
in Microbial Ecosystems through Horizontal Gene Transfer,
Front. Microbiol., 2016, 7, 173 Search PubMed.
124. S. Bhatt and S. Chatterjee, Fluoroquinolone antibiotics:
Occurrence, mode of action, resistance, environmental
detection, and remediation – A comprehensive review,
Environ. Pollut., 2022, 315, 120440 CrossRef CAS PubMed.
125. L. S. Redgrave, S. B. Sutton, M. A. Webber and L. J. V.
Piddock, Fluoroquinolone resistance: mechanisms, impact on
bacteria, and role in evolutionary success, Trends Microbiol.,
2014, 22, 438–445 CrossRef CAS PubMed.
126. O. Lomovskaya, K. Lewis and A. Matin, EmrR is a negative
regulator of the Escherichia coli multidrug resistance pump
EmrAB, J. Bacteriol., 1995, 177, 2328–
2334 CrossRef CAS PubMed.
127. N. Yousefian, A. Ornik-Cha, S. Poussard, M. Decossas, M.
Berbon, L. Daury, J.-C. Taveau, J.-W. Dupuy, S. ĐorđevićMarquardt, O. Lambert and K. M. Pos, Structural
characterization of the EmrAB-TolC efflux complex from E.
coli, Biochim. Biophys. Acta, Biomembr., 2021, 1863,
183488 CrossRef CAS PubMed.
128. J. Li, H. Zhang, J. Ning, A. Sajid, G. Cheng, Z. Yuan and H.
Hao, The nature and epidemiology of OqxAB, a multidrug
efflux pump, Antimicrob. Resist. Infect. Control, 2019, 8,
44 CrossRef PubMed.
129. L. H. Hansen, L. B. Jensen, H. I. Sørensen and S. J. Sørensen,
Substrate specificity of the OqxAB multidrug resistance pump
in Escherichia coli and selected enteric bacteria, J.
Antimicrob. Chemother., 2007, 60, 145–
147 CrossRef CAS PubMed.
130. X. He, F. Lu, F. Yuan, D. Jiang, P. Zhao, J. Zhu, H. Cheng, J.
Cao and G. Lu, Biofilm Formation Caused by Clinical
Acinetobacter baumannii Isolates Is Associated with
Overexpression of the AdeFGH Efflux Pump, Antimicrob.
Agents Chemother., 2015, 59, 4817–

## Page 77

4825 CrossRef CAS PubMed.
131. S. Coyne, N. Rosenfeld, T. Lambert, P. Courvalin and B.
Périchon, Overexpression of Resistance-Nodulation-Cell
Division Pump AdeFGH Confers Multidrug Resistance in
Acinetobacter baumannii, Antimicrob. Agents Chemother.,
2010, 54, 4389–4393 CrossRef CAS PubMed.
132. V. B. Srinivasan and G. Rajamohan, KpnEF, a New Member of
the Klebsiella pneumoniae Cell Envelope Stress Response
Regulon, Is an SMR-Type Efflux Pump Involved in BroadSpectrum Antimicrobial Resistance, Antimicrob. Agents
Chemother., 2013, 57, 4449–4462 CrossRef CAS PubMed.
133. V. B. Srinivasan, B. B. Singh, N. Priyadarshi, N. K. Chauhan
and G. Rajamohan, Role of Novel Multidrug Efflux Pump
Involved in Drug Resistance in Klebsiella pneumoniae, PLoS
One, 2014, 9, e96288 CrossRef PubMed.
134. Z. Xu, J. Xiong, C. Li, S. Hu, Z. Li, Y. Ma, S. Li, B. Huang, X.
Ren and X. Pan, Environmental concentrations of per/
polyfluoroalkyl substances promote the conjugative transfer
of antibiotic resistance genes, Chem. Eng. J., 2024, 498,
155500 CrossRef CAS.
135. V.-A. Thai, V. D. Dang, N. T. Thuy, B. Pandit, T.-K.-Q. Vo and
A. P. Khedulkar, Fluoroquinolones: Fate, effects on the
environment and selected removal methods, J. Cleaner Prod.,
2023, 418, 137762 CrossRef CAS.
136. N. R. Johnston and S. A. Strobel, Principles of fluoride
toxicity and the cellular response: a review, Arch. Toxicol.,
2020, 94, 1051–1069 CrossRef CAS PubMed.
137. O. Barbier, L. Arreola-Mendoza and L. M. Del Razo, Molecular
mechanisms of fluoride toxicity, Chem.-Biol. Interact., 2010,
188, 319–333 CrossRef CAS PubMed.
138. Y. Zhao, X. Liu, C. Liang, T. Pei, M. Guo, J. Wang and J.
Zhang, α-Lipoic Acid Alleviated Fluoride-Induced Hepatocyte
Injury via Inhibiting Ferroptosis, J. Agric. Food Chem., 2022,
70, 15962–15971 CrossRef CAS PubMed.
139. E. Terzioğlu, M. Arslan, B. G. Balaban and Z. P. Çakar,
Microbial silver resistance mechanisms: recent developments,
World J. Microbiol. Biotechnol., 2022, 38,
158 CrossRef PubMed.

## Page 78

Footnotes
† Electronic supplementary information (ESI) available. See DOI:
https://doi.org/10.1039/d4va00359d
‡ These authors contributed equally.
This journal is © The Royal Society of Chemistry 2025
Click here to see how this site uses Cookies. View our privacy
policy here.

