# sciadv.ado2957

**Source PDF**: sciadv.ado2957.pdf

---

## Page 1

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
1 of 10
B I O C H E M I S T R Y
Electron bifurcation and fluoride efflux systems 
implicated in defluorination of perfluorinated 
unsaturated carboxylic acids by Acetobacterium spp
Yaochun Yu1†, Fengjun Xu1, Weiyang Zhao1, Calvin Thoma2, Shun Che1‡, Jack E. Richman2,  
Bosen Jin1, Yiwen Zhu1, Yue Xing1, Lawrence Wackett2, Yujie Men1*
Enzymatic cleavage of C─F bonds in per-­ and polyfluoroalkyl substances (PFAS) is largely unknown but avidly 
sought to promote systems biology for PFAS bioremediation. Here, we report the reductive defluorination of α, 
β-­unsaturated per-­ and polyfluorocarboxylic acids by Acetobacterium spp. The microbial defluorination products 
were structurally confirmed and showed regiospecificity and stereospecificity, consistent with their formation by 
enzymatic reactions. A comparison of defluorination activities among several Acetobacterium species indicated 
that a functional fluoride exporter was required for the detoxification of the released fluoride. Results from both 
in vivo inhibition tests and in silico enzyme modeling suggested the involvement of enzymes of the flavin-­based 
electron-­bifurcating caffeate reduction pathway [caffeoyl-­CoA reductase (CarABCDE)] in the reductive defluorination. This is a report on specific microorganisms carrying out enzymatic reductive defluorination of PFAS, which 
could be linked to electron-­bifurcating reductases that are environmentally widespread.
INTRODUCTION
Anthropogenic organofluorines, particularly per-­ and polyfluoroalkyl substances (PFAS), are among the most strictly regulated and 
concerning emerging contaminants (1, 2). The perfluorinated structure makes C─F bond dissociation energies high enough to resist 
oxidative, reductive, and hydrolytic enzymes (3, 4). Although C─F 
bond reduction is thermodynamically favorable overall (5), the high 
energy (i.e., low reduction potential) required, together with fluoride toxicity and the short evolutionary time span of PFAS exposure 
to microbes, likely constrains PFAS biodefluorination (4). Currently, PFAS defluorination has largely been shown to be indirectly triggered when weaker C–H or C–Cl linkages are present, which are 
more physiologically feasible for existing enzymes to attack (5–12). 
Unstable intermediates such as gem-­fluoroalcohols may be formed, 
leading to spontaneous C─F bond cleavage (6, 7, 13). In contrast, 
direct enzymatic cleavage of C─F bonds in perfluorinated structures is rare (5). The defluorination of perfluorinated acids by Acidimicrobium sp. A6 has been reported, but no enzymes responsible for 
these processes have been identified (14–16). Because bioremediation is considered to be a more cost-­effective and environmentally 
friendly approach for in situ cleanup of PFAS-­impacted sites, it is 
vital to identify defluorinating microorganisms/enzymes and elucidate molecular mechanisms of C─F bond cleavage to allow systems 
biology and enzyme engineering to progress.
In previous studies, we observed reductive defluorination of an 
unsaturated perfluorinated compound, E-­perfluoro-­4-­methylpent-­
2-­enoic acid [PFMeUPA, (CF3)2CFCF═CFCOOH], by an anaerobic 
microbial community grown on lactate (17). The defluorination was 
not catalyzed by Dehalococcoides, capable of reductive dechlorination, in the community (17). This indicated the involvement of yet-­
to-­be-­uncovered microbes in the reductive defluorination and led 
us to further identify them here. Other key members of the community included a variety of heterotrophic fermenters, along with 
autotrophic methanogens and homoacetogens capable of utilizing 
H2 and CO2 (18, 19). Some acetogens have been reported to be involved in cometabolic dechlorination and debromination (20, 21). 
Here, we selected the two metabolic distinct microorganisms, methanogens and homoacetogens, and tested the PFMeUPA defluorination activities of representative species. We found that specific 
Acetobacterium species, commonly occurring acetogens, can catalyze reductive defluorination of PFMeUPA.
Following the identification of species capable of reductive defluorination, we delved into the mechanisms underlying this process. Defluorination occurring intracellularly accumulates toxic F−, 
thereby requiring functional F− transporters. In the defluorinating 
Acetobacterium species, a heterodimeric Fluc F− channel (CrcB1 and 
CrcB2) was used for F− detoxification. Both in vivo experiments and 
in silico modeling suggested that the electron-­bifurcating caffeate 
reduction complex [caffeoyl-­CoA reductase (CarABCDE)] (22) was 
involved in the reductive defluorination. Metagenomic screening of 
CarC homologs revealed a ubiquitous occurrence of potential defluorinating microorganisms in wastewater environments. Here, the 
identification of defluorinating pure cultures and the proposed molecular mechanisms of microbial reductive defluorination offer 
valuable insights into rational design and optimization of biocatalysts, which could be potentially applied for bioremediation of per-­ 
and polyfluorinated contaminants.
RESULTS
PFMeUPA is reductively defluorinated at the α-­C position by 
Acetobacterium bakii
Previous studies on PFMeUPA defluorination by an anaerobic community did not identify the responsible microorganism, possibly 
1Department of Chemical and Environmental Engineering, University of California, 
Riverside, Riverside, CA 92521, USA. 2Department of Biochemistry, Molecular Biology and Biophysics and Biotechnology Institute, University of Minnesota, Twin Cities, 
MN 55108, USA.
*Corresponding author. Email: ymen@​engr.​ucr.​edu
†Present address: Department of Environmental Chemistry, Swiss Federal Institute 
of Aquatic Science and Technology (Eawag), 8600 Dübendorf, Switzerland.
‡Present address: SINOPEC Research Institute of Safety Engineering, Qingdao, 
266071 Shandong, China.
Copyright © 2024 The 
Authors, some rights 
reserved; exclusive 
licensee American 
Association for the 
Advancement of 
Science. No claim to 
original U.S. 
Government Works. 
Distributed under a 
Creative Commons 
Attribution 
NonCommercial 
License 4.0 (CC BY-­NC). 
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 2

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
2 of 10
due to the challenge in enriching the species carrying out the cometabolic defluorination of PFMeUPA (13, 17). Here, we took an 
alternative approach and did a screening test using two pure cultures, Methanosarcina barkeri and A. bakii, a methanogen and 
homoacetogen, respectively, of the type found in the anaerobic 
community. While M. barkeri exhibited very minor defluorination (5 μM fluoride ion from 55 μM PFMeUPA after 50 days of 
incubation) (fig. S1), A. bakii achieved complete biotransformation of PFMeUPA (~90 μM) within 3 weeks when grown on fructose (Fig. 1A).
The defluorination was substantially faster than that reported 
previously (~120 days) in the anaerobic community (17). Despite 
the different rates, A. bakii transformed PFMeUPA into the same 
intermediates and end products as the community with the same 
stoichiometry of fluoride release at a near 1:1 molar ratio to the 
transformed PFMeUPA (Fig. 1A). By comparing MS2 and 19F-­NMR 
spectra of biological samples and isomeric products of chemical 
catalytic reduction of PFMeUPA, we confirmed the regio-­specific 
single isomer structure of the two microbial reductive defluorination transformation products (TPs): (i) the unsaturated form, 
(CF3)2CFCF═CHCOO−, with a theoretical mass/charge ratio (m/z) 
of 256.9849 and designated “TP256” (the same nomenclature hereafter for other TPs), and (ii) its secondary hydrogenation product 
[TP259, (CF3)2CFCHFCH2COO−] (Fig. 1, A and B, and figs. S2 and 
S3). This demonstrated an α-­C defluorination in the microbial 
system. Unlike the parallel chemically catalytic reduction of PFMeUPA 
that we conducted here (Supplementary Text), only one of the 
separable diastereomers from chemical synthesis was detected as a 
biohydrogenation product [TP276, (CF3)2CF(CHF)2COO−] during 
PFMeUPA biotransformation by A. bakii (Fig.  1, A and B, and 
fig. S4). The regiospecificity and stereospecificity of the TPs demonstrate that those reactions carried out by A. bakii were enzyme mediated. Quantifying the two end TPs (TP259 and TP276) in the 
mixture by 19F-­NMR spectra (Supplementary Text) indicated that 
the majority (~82%) of PFMeUPA was transformed via reductive 
defluorination (Fig. 1, B and C).
A. bakii also transformed two other unsaturated PFAS with 
structures similar to PFMeUPA, PFUPA (CF3CF2CF═CFCOOH) 
and 6:2 FTUCA (CF3(CF2)4CF═CHCOOH) (Fig. 1D), via reductive 
defluorination and hydrogenation (fig. S5, A to D). The total defluorination was similar to what we previously observed in an anaerobic 
microbial community (13). While the community slightly defluorinated FTMeUPA [(CF3)2CFCH═CHCOOH] (17), A. bakii only hydrogenated this structure without any F− formation (Fig. 1D and 
fig. S5, E and F). This indicated that while A. bakii carried out the 
reduction of a number of α, β-­unsaturated per-­ and polyfluorinated 
carboxylic acids, it only specifically cleaved C─F bonds on the unsaturated carbons. Thus, the previously reported defluorination of 
FTMeUPA at the tertiary C─F by the anaerobic community (17) 
should be attributed to other pathways with different mechanisms.
Fig. 1. Defluorination of per-­ and polyfluorinated unsaturated carboxylic acids by A. bakii. (A) PFMeUPA removal and the formation of fluoride and defluorination 
intermediate (TP256) and stable end products (TP276 and TP259). (B) 19F-­NMR spectra identification and quantification of products by assignment of resonances, integrations (in red numbers), and atom coupling. The additional NMR data on biological and chemical reduction products is included in the Supplementary Materials (for confirmatory LC-­HRMS/MS data, see figs. S2 to S4). (C) Confirmed biotransformation pathways of PFMeUPA. (D) Substrate specificity of A. bakii for reductive defluorination.
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 3

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
3 of 10
A functional F− transporter is a prerequisite for 
intracellular defluorination
We did not observe defluorination activity in the cell-­free spent medium (fig. S6), suggesting that the defluorination occurred within 
the outer membrane of the bacterium. Whether this process takes 
place in the periplasm, inner membrane, or cytoplasm, fluoride 
would become concentrated within the cellular environment. If 
formed in the periplasm, fluoride is known to spontaneously enter 
cells as hydrogen fluoride (HF). If formed in the cytoplasm, fluoride 
would be directly toxic, inhibiting critical enzymes at concentrations as low as 70 μM (23, 24). Bacteria catalyzing defluorination 
have been shown to be rendered nonviable due to fluoride release 
(25). Thus, for defluorination occurring in the cellular environment 
such as PFMeUPA defluorination by A. bakii, a functional F− transporter would be required for the detoxification of fluoride (26). 
A. bakii has a Fluc family F− channel, which is a heterodimer composed of two distinct proteins encoded by crcB1 and crcB2 genes 
(26, 27). Both genes were significantly (multivariate t test, adjusted 
P < 0.05) up-­regulated in the presence of PFMeUPA compared to 
the nonspiked control (~2-­fold change) (fig. S7D and table S1). The 
expression of the Fluc channel in A. bakii is controlled by a transcriptional F− riboswitch located upstream of crcB1 (fig. S8A) (24, 
26). The up-­regulation of crcB genes indicated the activation of fluoride detoxification by A. bakii in response to PFMeUPA defluorination, which is consistent with the response to NaF (fig. S8B). Thus, a 
functional fluoride exporter is an important component for sustaining defluorination.
We then expanded the screening to four more Acetobacterium 
species (A. woodii, A. paludosum, A. malicum, and A. fimetarium) 
(Fig. 2A). Except for A. fimetarium, all the other strains exhibited 
defluorination activity (Fig. 2, A and B). In the genome of A. fimetarium, 
while crcB1 is intact and homologous to that of A. bakii, crcB2 was 
found to be truncated by 64 bp from the 3′ tail (Fig. 2A). Given 
that both subunits must be present for a functional heterodimer 
Fluc F− channel (27), we attributed A. fimetarium’s incapability of 
defluorination to its nonfunctional Fluc operon. To test this hypothesis, we introduced into a plasmid the crcB1 and crcB2 genes 
from five Acetobacterium species and transformed the plasmid 
into an Escherichia coli K-­12 MG1655 ΔcrcB mutant strain. That 
strain is sensitive to F− and exhibited limited growth in the presence of 250 μM F−. Except for E. coli ΔcrcB with crcB1 and truncated crcB2 from A. fimetarium, all other strains were rescued 
from F− toxicity (Fig. 2C). This supports the idea that a functional 
F− exporter is essential for microorganisms to be capable of intracellular defluorination.
Flavin-­based electron-­bifurcating caffeate reduction 
complex as a proposed enzymatic system involved in the 
PFMeUPA defluorination
Besides the need for an active F− exporter for cell detoxification, we 
also explored potential enzymes involved in reductive defluorination. We demonstrated that the reductive defluorination occurred 
to α, β-­unsaturated fluorinated carboxylic acids (Fig. 1D). Given 
that Acetobacterium spp. were able to reduce caffeate, which is also 
an α, β-­unsaturated carboxylic acid (22), we hypothesized that enzymes for caffeate reduction, encoded by the car operon (carABCDE) 
(28–30), could also be involved in the reductive defluorination 
of PFMeUPA.
Here, we obtained both in vivo and in silico results suggesting 
the involvement of the car operon in the reductive defluorination of 
PFMeUPA: (i) PFMeUPA defluorination was inhibited by caffeate 
and restored after caffeate was depleted; (ii) defluorination was observed with multiple Acetobacterium species possessing CarABCDE-­
dependent caffeate reduction pathways; (iii) the consistency between 
Fig. 2. Comparison in PFMeUPA defluorination capability and fluoride resistance among Acetobacterium species. (A) Amino acid sequence similarities of the CrcB 
fluoride ion transporter and Car operon, CarABCDE among five Acetobacterium species, and the corresponding total fluoride formation from ~80 μM PFMeUPA; (B) Temporal fluoride formation from ~80 μM PFMeUPA; (C) Growth restoration of E. coli crcB-­knockout mutant (i.e., MG1655 ΔcrcB) supplemented by crcB from different origins, 
including the native crcB from E. coli, those from the five Acetobacterium species, and individual subunit genes, crcB1 and crcB2 from A. woodii. The cells were cultivated in 
LB medium containing 250 μM NaF. The color legend is arranged from top to bottom, corresponding to the order of the lines at the final time point shown in the figure.
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 4

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
4 of 10
the biochemical reactions observed for PFMeUPA and the pathways 
was inferred based on the function of each gene in the car operon; 
(iv) computational modeling of the CarC active site showed that the 
coenzyme A (CoA) thioesters of PFMeUPA and caffeate align their 
carbon-­carbon double bonds identically with respect to the CarC 
flavin cofactor known to catalyze substrate (28).
First, the addition of 5 mM caffeate in A. bakii completely inhibited the defluorination of PFMeUPA during the first week (Fig. 3A) 
without substantial inhibition on cell growth (fig. S9A). Defluorination activity was revived after caffeate was depleted after 12 days 
(Fig. 3A and fig. S9B). One possible reason for the inhibition was 
that caffeate competed with PFMeUPA for the binding site of the 
enzyme catalyzing the reductive defluorination. The enzyme system 
that caffeate reacts with is the caffeate reduction complex. This suggests that enzymes in the caffeate reduction pathway were involved 
in the reductive defluorination. We were not able to observe differential expression of carABCDE in A. bakii by PFMeUPA (fig. S7 and 
tables S1 to S3). This was probably due to the maximum level of 
PFMeUPA that could be added without causing toxicity, which was 
about 10-­fold lower than the caffeate concentration typically used 
(31). The low concentration of PFMeUPA could result in low levels 
of transcripts and proteins, which would make them difficult to detect and compare by RNA sequencing or proteomics.
We then compared PFMeUPA defluorination activities of the 
four Acetobacterium species (i.e., A. bakii, A. woodii, A. paludosum, 
and A. malicum) with a functional Fluc F− channel and similar 
caffeate reduction operons (Fig. 2A). Similar defluorination activities 
were observed for A. bakii, A. woodii, and A. paludosum, while 
A. malicum exhibited partial defluorination and parent compound 
biotransformation (Fig. 2B and fig. S10A). The lower defluorination 
of PFMeUPA by A. malicum might be due to its slower kinetics of 
caffeate reduction. Although A. malicum has a complete caffeate 
reduction operon, its caffeate reduction activity was significantly 
lower than the other Acetobacterium species (t test, P < 0.05, 
fig. S10, B and C).
Using the x-­ray structure coordinates of the CarCDE enzyme 
complex (28), we modeled the caffeoyl-­CoA binding site of CarC to 
determine whether caffeoyl-­CoA and an analogous perfluorinated 
acyl-­CoA could compete for binding and position the 2,3-­enoyl 
double bond near the CarC flavin for reduction. Figure 3B shows 
the docking model of each CoA thioester in the surface cavity proposed for the x-­ray structure of CarC to harbor caffeoyl-­CoA (28), 
consistent with the structure of other enoyl-­CoA reductases from 
Clostridium spp. (32). The modeling showed sufficient space in the 
binding site to accommodate the branched -­CF(CF3)2 moiety in the 
same region as the catecholic aromatic ring of caffeoyl-­CoA 
(Fig. 3B). The double bond of each substrate is nearly superimposable with respect to the flavin cofactor previously identified to carry 
out the reduction reaction (28). The modeling also identified a residue that is conserved across many enoyl-­CoA reductases, Glu362, 
that was poised to within 2.2 Å of C-­3 on caffeoyl-­CoA or, as shown, 
C-­3 of the analogous perfluorinated acyl thioester (Fig. 3, C and D). 
This immediately suggested a mechanism for defluorination in 
which Glu362 serves as a general acid to deliver a proton to the 
Fig. 3. Reductive defluorination of PFMeUPA by enzymes involved in the caffeate reduction pathway. (A) Caffeate inhibition of PFMeUPA defluorination; orange, 
PFMeUPA; green, fluoride; blue, caffeate; solid line, with 5 mM caffeate; dashed line, without caffeate. (B) Docking of CoA thioesters of hydrocaffeic acid (orange) and 
PFMeUPA (purple) into CarC active site (pink) with the relative position of FAD shown (green). (C) Proposed reaction pathway showing reduction of fluorinated substrate 
by FAD (bottom) and elimination of fluoride with glutamate362 serving as both the active site acid in the reduction step and, after deprotonation, as the base during the 
defluorination step. (D) The close contact between the docked CoA thioester of PFMeUPA and Glu362 as proposed participating in the defluorination reaction.
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 5

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
5 of 10
opposite face of the double bond as the hydride equivalent is delivered by the flavin cofactor to reduce the double bond (Fig. 3C). After 
reduction, the deprotonated acid can then serve as a general base to 
stabilize removal of the proton from C-­3 and facilitate the β-­
elimination of a fluoride anion from C-­2, forming TP256. This 
enzyme-­assisted defluorination explains the regiospecific fluoride 
elimination observed for the microbiological defluorination, whereas we see the elimination of fluorine from either C-­3 or C-­2, or both, 
in chemical reductions (figs. S2 and S4). This is consistent with the 
identification of TP259 as the sole product from an additional reduction of the double bond in TP256. On the basis of our modeling, 
no further defluorination is possible for TP256 because there is no 
fluorine at C-­2 to undergo defluorination. The minor TP, TP276, 
would derive from the exit of the hydrogenation intermediate from 
the CarC active site before deprotonation occurred. This enzyme-­
assisted defluorination pathway (fig. S11) is also supported by the 
TP formation trend (Fig. 1A), with TP256 first increasing then decreasing while both TP276 and TP259 increase over time. This argues against a precursor-­product relationship between TP276 and 
TP256/TP259 that would be observed if defluorination were occurring freely in solution.
Chain-­length specificity and prevalence of microorganisms 
possessing car operon (CarC) in nature
There are several microbial groups, including some Clostridium 
species, known to have electron-­bifurcating enoyl-­CoA reductases. 
We further tested Clostridium homopropionicum (DSM5847), a 
species that utilizes electron bifurcation in a fermentation pathway 
yielding propionic acid (33). This pathway involves α, β-­enoyl–CoA 
reduction reactions with substrates notably shorter than caffeoyl-­
CoA. Analysis of the published genome sequence (34) reveals 
gene constellations analogous to those in Acetobacterium spp., including homologs of CarCDE and fluoride ion export genes. Thus, 
C. homopropionicum is potentially capable of reductive defluorination, akin to Acetobacterium spp., albeit with shorter-­chain substrates. As expected, C. homopropionicum did not exhibit any 
transformation activity against PFMeUPA (fig. S12). However, it did 
achieve slow defluorination (20 μM of F− formation from 100 μM of 
parent compound over 15 days) for PFUPA (CF3CF2CF═CFCOOH) 
that is one carbon shorter than PFMeUPA (fig. S12). Considering 
that many other Clostridium spp. are known to catalyze the reduction of short-­chain acyl-­CoA substrates using genomically identifiable CarCDE homologs (35, 36), we think it is reasonable to suggest 
that other Clostridia could also react with short-­chain unsaturated 
perfluorinated acids. In light of the above, we collected sequences 
with >55% identity to A. woodii and C. homopropionicum from the 
nonredundant protein sequence database using the BLAST algorithm, created multiple sequence alignments, and used the multiple 
sequence alignments to generate hidden Markov models (HMMs) 
and sequence logos, shown in their entirety in fig.  S13. The sequence logos were then searched for critical regions in CarC linked 
to electron bifurcation and defluorination. Figure 4A depicts sequence logos around F155, shown by Demmer et al. (28) to be important in electron bifurcation electron transfer and the highly 
conserved region surrounding E362, that we have proposed as a 
key residue in defluorination (Fig. 3, C and D). The conservation 
surrounding F155 is consistent with an important role in catalysis, 
following its proposed role in facilitating electron transfer from the 
Fig. 4. Sequence signatures of CarC and the presence of CarC homologs in global wastewater treatment plant (WWTP) metagenomes. (A) Key regions of sequence 
LOGOs and their significance in electron bifurcation and defluorination found Acetobacterium and Clostridium derived from BLAST search as described in Materials and 
Methods. The left side shows sequence patterns containing phenylalanine (F) 155 that is reported to bridge the electron-­bifurcating flavin in CarC to its flavin electron 
acceptor in CarE (28). The right side shows the sequence patterns containing glutamate (E) 362 that is implicated in this study as facilitating defluorination following 
double-­bond reduction. (B) A search of the MGnify database retrieved 723 WWTP metagenomes of which 499 had sufficient location metadata. The multiple sequence 
alignments in part A were used to create HMMs to search 499 metagenome sites containing genes with sequence signatures encoding putative electron-­bifurcating reductases having the potential to catalyze defluorination. In most instances, multiple metagenomic sequences were identified at a given site and the number is represented by the size of circles. Sites were identified as having Acetobacterium-­like patterns, Clostridium-­like patterns, or both, represented as red, blue, or green circles, 
respectively. Specific details of bioinformatic methods for building the searchable database and creating the HMMs are in Materials and Methods. Finally, proteins retrieved in the search were used to create new sequence logos, highlighted here as universal logos from WWTPs.
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 6

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
6 of 10
electron-­bifurcating flavin in CarC to a flavin electron acceptor in 
CarE (32). The region around E362 shows complete conservation in 
Acetobacterium spp. and high conservation with Clostridium.
Building on that, we used the HMMs to search global wastewater 
treatment plant (WWTP) metagenomic sequences obtained from 
the MGnify Database (Fig. 4B). We searched WWTP metagenomes 
because many anaerobic bacteria found to contain Car operon gene 
homologs had originally been isolated from WWTPs on different 
continents (37–41). Sequences with sample-­origin metadata were 
used to create a map showing WWTPs containing CarC sequence 
patterns strongly linking their functionality to those found in Acetobacterium, Clostridium, or both. In total, 723 metagenomes were 
identified with homologs; a breakdown by continent is shown in 
table S4.
The focus here was on WWTPs, which are known to be enriched 
in Acetobacterium and Clostridium sequences, but the HMMs 
(fig. S13) can be used to search many additional environments. 
We have detected PFMeUPA defluorination activity in aqueous 
film forming foam-­impacted soil and groundwater communities 
(fig.  S14), suggesting that other environments may contain 
electron-­bifurcating, defluorinating reductases of the type characterized in this study.
DISCUSSION
The vast majority of organofluorines like PFAS are industrial chemicals, new to the biosphere, and among the most pervasive are perfluorinated acids (1, 42–45). Some consider these chemicals to be 
nonbiodegradable given the high bond dissociation energy and low 
reduction potential of C─F bonds. Moreover, cytoplasmic defluorination, if occurring, would result in fluoride formation, which is 
highly toxic to cells. Here, we identified specific bacterial species 
capable of reductive defluorination of certain PFAS structures. This 
is pivotal to guide the screening of more defluorinating microorganisms, further understand the defluorination mechanism, and biochemically pinpoint the responsible enzymes. Our findings also 
suggest that combining a unique noncentral metabolic trait (flavin-­
based electron bifurcation) and a co-­opted detoxification trait (fluoride export) could allow Acetobacterium and Clostridium species 
to defluorinate unsaturated perfluorinated acids with different 
chain lengths.
Intracellular fluoride release from obligately cytoplasmic defluorinating enzymes will immediately lead to fluoride binding to sensitive magnesium and calcium centers of enzymes, preventing cell 
growth (46). It is necessary to couple C─F cleavage enzymes with 
rapid fluoride expulsion. Fluoride is abundant in Earth’s crust (47, 
48). Thus, nature has evolved effective proteins to protect against 
fluoride leaking into cells from mineral fluoride (24, 49). Under 
fluoride stress, many native microorganisms express one of two 
families of membrane proteins to export F−, i.e., the CLCF family of 
F−/H+ antiporters and the Fluc (CrcB) family of F− exporting channels. CLCF family proteins are homodimers, while Fluc family proteins include homodimers and heterodimers. Although the two 
types of F− exporters differ in sequence, structure, and mechanism, they share the same physiological role in F− detoxification. 
A. fimetarium, which has an intact Car operon (Fig. 2A and fig. 
S10) but a nonfunctional CrcB (Fig. 2C and fig. S8), was able to 
reduce caffeate (fig. S10A) but not able to defluorinate PFMeUPA at 
all (Fig. 2B). Such an incapability underscores the essential role 
F− exporters play in enabling cytoplasmic defluorination. This will 
be one critical design criterion for efforts to bioengineer efficient 
PFAS-­degrading microorganisms.
Previous studies have demonstrated reductive defluorination of 
α, β-­unsaturated perfluorocarboxylic acids by anaerobic microbial 
communities (17), but common reductive dechlorinating bacteria 
and purified reductive dehalogenases did not react with fluorinated 
substrates (50), suggesting different reduction systems. Some biological reductions with the lowest reduction potential (<−500 mV) 
are carried out by flavin-­based electron-­bifurcating complexes (51, 
52). These enzyme complexes are widespread in anaerobic Bacteria 
and Archaea where they couple adenosine triphosphate (ATP) formation or utilization with the transformation of aryl/alkyl/alkenyl-­
CoA substrates, aldehydes, ketones, acids, hydrogen/protons, and 
carbon monoxide (53). The shared feature is the separation, or bifurcation, of electron pair energies such that one is boosted to a 
higher energy (lower potential) and the other to a lower energy 
(higher potential) (51). Here, we implicate one such system, 
CarCDE, in reductive defluorination. CarCDE is naturally evolved 
to reduce plant aromatic enoic acids such as caffeic acid, coupled 
with formation of reducing power for autotrophic growth or sodium 
transport to generate incremental ATP in energy-­marginalized environments (30). While the similarity of caffeic acid and PFMeUPA 
is not immediately apparent, modeling the respective CoA esters 
showed an ideal fit in the active site of CarC. This is also consistent 
with caffeate competition experiments implicating PFMeUPA as a 
possible electron acceptor alternative to caffeate and undergoing 
enzyme-­assisted α-­defluorination within the electron-­bifurcating 
enzyme complex. The CarCDE complex was purified from A. woodii 
under strict anaerobic conditions by Demmer et al. (28). A clean 
in-­frame gene knockout system in A. woodii was recently reported 
by Baker et al. in 2022 (54). As the sophisticated molecular methods 
are further established and applied, a full demonstration could be 
done in the future using carC-­knockout mutants in vivo and purified CarCDE complex in  vitro under appropriate conditions. Although such an electron-­bifurcating system we proposed here was 
specific to unsaturated PFAS structures, given the occurrence of unsaturated PFAS in the environment, the findings could help with a 
better understanding of their environmental fate. It also suggests 
potential for the rational design and optimization of other electron-­
bifurcating systems with diverse substrate specificities in catalyzing 
reductive defluorination reactions.
In different prokaryotes utilizing electron bifurcation, the high-­
energy electron reduces ferredoxins or flavodoxins that, in turn, 
reduce protons, dinitrogen gas, carboxylates to alcohols, and benzenoid rings, and activate 2-­hydroxyacyl-­CoA dehydratases (51). 
The benzoyl-­CoA reductase in Thauera aromatica has been shown 
to reductively defluorinate 4-­fluoro-­benzoyl-­CoA via HF elimination and further reduction, a reaction analogous to that observed 
here with perfluoroalkenoic acids (55). While this class I system 
uses ATP to drive the formation of high-­energy electrons, a class II 
benzoyl-­CoA reductase uses two flavin-­based electron bifurcation 
steps consecutively to catalyze benzene ring reduction (56). This 
1-­MDa enzyme complex uses flavin, tungsten, selenium, and more 
than 50 iron-­sulfur clusters to achieve the Eo′ potential required for 
benzene ring reduction. These studies highlight an as-­yet-­unrealized 
prospect for endergonic reduction of a broader range of PFAS 
compounds by serving as the low-­potential electron acceptor in 
the electron-­bifurcating system. It warrants future discovery or 
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 7

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
7 of 10
engineering of those enzyme complexes, overcoming the mechanistic/thermodynamic barriers, thus substantially shortening the 
time it could have taken by natural evolution.
MATERIALS AND METHODS
Fluorinated compounds
Four fluorinated carboxylic acids were investigated based on their 
previously reported biodefluorination feasibilities in an anaerobic 
enrichment community (13). The chemicals were purchased from 
SynQuest Laboratories (Alachua, FL) and used without further purification. For all authentic standards, 10 mM stock solutions were 
prepared anaerobically in autoclaved Milli-­Q water (160 ml) sealed 
serum bottles and stored at room temperature until use. The limit of 
quantification for each standard compound was determined as the 
lowest concentration of calibration standards with a detection variation within ±20%. The detailed compound information is provided 
in table S5.
The hydrogenation product of PFMeUPA was synthesized in-­
house using chemical catalysis. The catalyst, 39 mg of 5% Pd/C, and 
stir bar were charged to a Pyrex 6-­ml 14/20 RB flask, which was, in 
turn, positioned in the bottom of a 50-­ml Parr pressure reactor. The 
sealed reactor was evacuated (1.3 × 10−4 atm) and then filled with 
H2 gas. The reactor was maintained under a slow purge of H2 while 
116 mg of PFMeUPA, followed by 2× washes totaling 2 ml of 
methyl-­t-­butyl ether, was charged and magnetically stirred. The reactor was sealed and pressured to 3.74 atm with H2 and maintained 
for 4 hours. The reactor was stirred overnight while the pressure was 
slowly released. Anhydrous HF was produced. After settling, the 
clear colorless supernatant was sampled (0.3 ml) and diluted to 
0.7 ml with CDCl3 for the acquisition of 19F-­NMR spectra using 
a Varian Unity Inova 400-­MHz NMR Spectrometer for standard 
5-­mm outside diameter tubes. The products were also analyzed for 
structure confirmation by liquid chromatography coupled to high-­
resolution tandem mass spectrometry (LC-­HRMS/MS; as described 
below), and the LC chromatograph and MS2 fragmentation profiles 
(figs. S2 to S4) were used to compare with those of the products 
from biological reductive defluorination. Details of palladium-­
catalyzed hydrogenation of PFMeUPA are included in the Supplementary Text.
Cultures and growth conditions
Five Acetobacterium, one Clostridium, and one Methanosarcina 
species were used to study the defluorination activities of the selected fluorinated compounds. A. bakii (DSM 8239) and M. barkeri 
(DSM 800) were purchased from the American Type Culture Collection (ATCC); A. m woodii (DSM 1030), A. paludosum (DSM 
8237), A. malicum (DSM 4132), A. fimetarium (DSM 8238), and 
C. homopropionicum (DSM 5847) were purchased from Deutsche 
Sammlung von Mikroorganismen und Zellkulturen (DSMZ). To 
test the biotransformation capabilities of individual species, we cultivated all cultures under their optimal growing conditions as instructed by DSMZ. Specifically, A. bakii was grown in DSMZ 
Medium 900 (pH 7.0), supplemented with yeast extract (1 g/liter) 
and fructose (5 g/liter), and cultures were maintained at 20°C. A. 
woodii was cultivated at 30°C in DSMZ Medium 135 (pH 8.0) containing yeast extract (2 g/liter) and fructose (10 g/liter) as the primary substrate. A. malicum was also grown at 30°C in DSMZ 
Medium 135 but with fructose (1 g/liter) as the primary substrate. 
A. paludosum was kept at 20°C and grown in DSMZ Medium 
614 (pH 7.0), which contains yeast extract (0.5 g/liter) and fructose (10 g/liter). A. fimetarium was cultivated at 30°C in DSMZ 
Medium 900 with lactate (4 g/liter) as the primary substrate. 
C. homopropionicum was incubated at 35°C with DSMZ Medium 
503 (pH 7.2), containing fructose (2 g/liter) as the primary substrate. Last, M. barkeri was grown in DSMZ Medium 120 (pH 7.0) at 
35°C with methanol (15.5 g/liter) as primary substrate. The comprehensive list of Media for Microorganisms can be found in MediaDive 
(https://mediadive.dsmz.de/).
Biotransformation of fluorinated compounds by 
Acetobacterium spp. and C. homopropionicum
After the full growth of individual species, each culture was 
transferred (5%, v/v) into six serum bottles with 95 ml of fresh 
culture medium containing optimal primary substrates as suggested by DSMZ (i.e., fructose for A. woodii, A. paludosum, A. bakii, 
A. malicum, C. homopropionicum; lactate for A. fimetarium; and 
methanol for M. barkeri; see details in the “Cultures and growth conditions” section). Two conditions were set up with triplicates: (i) 
growth positive control with no fluorinated compounds addition and 
(ii) biotransformation group spiked with 50 to 90 μM of individual 
fluorinated compounds (i.e., one compound was spiked for each 
test). A triplicated heat-­inactivated biomass adsorption control was 
also set up by inoculating 5 ml of autoclaved (at 121°C for 20 min for 
two cycles) culture into the same culture medium spiked with the 
same concentration of individual fluorinated compounds. Samples 
were taken subsequently during the incubation period for the measurement of the parent compound, TPs, and F−. Briefly, at each sampling time, 3 ml of aqueous suspension was centrifuged at 16,000g 
for 30 min (4°C). Two milliliters of supernatant was used for F− 
measurement. The remainder (~1 ml) was stored at 4°C in the dark 
for subsequent LC-­HRMS/MS measurement. The cell pellets from 
the biotransformation experiments were stored at −20°C for DNA 
extraction. The DNA extraction was done using a DNeasy PowerSoil 
Kit (QIAGEN, Germantown, MD) according to the manufacturer’s 
instructions. Cell growth was measured using quantitative PCR 
(qPCR), and the detailed procedure was provided in the Supplementary Text.
Differential gene expression by transcriptomic and 
proteomic analyses
Gene regulation at the transcriptional and translational level in 
A. bakii grown on d-­fructose (5 g/liter) with and without the addition of PFMeUPA (300 μM) was examined using RNA and protein 
sequencing. Details are included in the Supplementary Text.
Functional validation of the fluoride ion transporter genes 
in Acetobacterium spp.
The mutant strain E. coli K-­12 MG1655 ΔcrcB was constructed using the λ-­red homologous recombination system and is sensitive to 
high fluoride concentration. The gene crcB from different Acetobacterium spp. was obtained using PCR. Purified PCR products were 
ligated to the linearized pBAD24 backbone. Recombinant plasmids 
were introduced into the mutant strain for rescue tests. Each strain 
was tested with LB medium supplemented with 250 μM of NaF. Cell 
growth was tracked via OD600 measurement. A detailed description 
of strain construction and rescue tests can be found in the Supplementary Text.
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 8

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
8 of 10
Substrate competition between caffeate and PFMeUPA 
in A. bakii
Caffeate (5 mM) was added into six 160-­ml serum bottles containing 
95 ml of culture medium and incubated overnight until caffeate was completely dissolved. Pregrown A. bakii was then transferred (5%, v/v) into 
the above caffeate-­containing medium bottles. Three of the six bottles 
were randomly picked as caffeate-­only control and the remaining three 
were amended with ~90 μM PFMeUPA. In addition, A. bakii (5%, v/v) 
was also inoculated into another bottle with growth medium only (without any addition of caffeate or PFMeUPA), serving as the positive 
control to confirm the normal growth activity. Samples were taken 
subsequently during the incubation period for the measurement of 
the parent compounds (i.e., PFMeUPA and caffeate), TPs, and F− as 
described below. Cells were collected and stored at −20°C for subsequent DNA extractions and biomass growth measurements.
UHPLC-­HRMS/MS analysis
For ultra-­high performance liquid chromatography (UHPLC) analysis, 2 μl of sample was loaded onto an Hypersil GOLD column 
(particle size 1.9 μm, 100 × 2.1 mm, Thermo Fisher Scientific) connected to a Hypersil GOLD guard column (particle size 3.0 μm, 
10 × 2.1 mm, Thermo Fisher Scientific) and eluted with Milli-­Q water (A) and methanol (B) (both with 10 mM ammonium acetate) at 
a flow rate of 280 μl/min, with a gradient as follows: 95% A: 0 to 
1 min, 95%-­5% A: 1 to 6 min, 5% A: 6 to 8 min, and 95% A: 8 to 
10 min. For HRMS/MS, mass full scan was performed in the negative 
mode of electrospray ionization at a resolution of 70,000 at m/z 200 
and a scan range of m/z 50 to 750, and MS2 spectra were acquired in 
the same negative mode at a resolution of 14,000 at m/z 200 with a 
normalized collision energy of 25. Xcalibur 4.0 (Thermo Fisher Scientific) was used for data acquisition and analysis. TP screening was 
conducted using Compound Discoverer 3.1 (Thermo Fisher Scientific) as previously described (6). ChemDraw Professional 20.0 was 
used for drawing, displaying, and characterizing chemical structures.
19F NMR
19F-­NMR spectra on the biotransformation product of PFMeUPA by 
A. bakii (fig. S15) were acquired on a Bruker Avance NEO 600-­MHz 
NMR spectrometer with a 5-­mm H&F-­C/N-­D cryoprobe and adapter 
to a 3-­mm milli-­tube. All 19F spectra on products of PFMeUPA 
from abiotic catalytic hydrogenation were acquired on a Varian 
Unity Inova 400-­Hz NMR spectrometer with a standard 5-­mm 
four-­nuclei (1H, 31P, 13C, and 19F) probe.
Fluoride ion measurement
The concentration of fluoride ion (F−) in the culture supernatant 
was measured by an ion-­selective electrode (ISE, HACH, Loveland, 
CO) connected to an HQ30D Portable Multi Meter (HACH). The 
accuracy of the fluoride measurement by ISE was cross-­validated 
using ion chromatography as described in our previous study (17). 
For the ISE measurement, a 100-­μg fluoride ionic strength adjustment powder (HACH) was added into 2 ml of culture supernatant; 
the F− concentration was then measured by the ISE-­Multi Meter system.
Enzyme modeling
Models of the CarC active site were created by docking C5F9CO-­CoA 
and caffeoyl-­CoA. The published crystal structure of A. woodii CarC 
[Protein Data Bank (PDB) 6FAH] does not include caffeoyl-­CoA in 
complex with the enzyme. Therefore, a highly similar CarC homolog 
from a Rat protein complexed with acetoacetyl-­CoA (PDB 1JQI) 
was aligned to the A. woodii structure. After alignment, caffeoyl-­
CoA and C5F9CO-­CoA were fitted to the electron density of the 
acetoacetyl-­CoA using Coot (57). Restraint dictionaries for ligands 
were created on the Grade web Server (58). The protein-­ligand complex structures were then subjected to geometric refinement using 
the Refmac5 tool within Collaborative Computational Project 
Number 4 (CCP4) with default parameters (except the number of 
refinement cycles was set to 5 and the matrix weighting term was set 
to 0.0000001) (59). Visualization was done in PyMOL (The PyMOL 
Molecular Graphics System).
Bioinformatics of CarC homologs in WWTP metagenomes
HMMs were created to search a wastewater metagenome database 
for CarC homologs. These HMMs originated from two different 
seed sequences: caffeoyl-­CoA reductase CarC (A. woodii DSM 1030, 
NCBI AFA48354) and acyl-­CoA dehydrogenase family protein 
(C. homopropionicum, NCBI WP_052222365.1). For each seed, sequence databases with varying confidence levels were created using 
BLAST by gathering sequences at different percent identity thresholds 
from the nonredundant database. Proteins with 50% identity and 
higher have been reported to have less than 1-­Å difference between 
their backbones (60). Therefore, the lowest cutoff score was 55% identity for building a sequence library and constructing HMMs. In addition, 74% identity was selected as the other cutoff, because that was 
the minimum similarity score identified in this study for other Acetobacterium CarC proteins capable of defluorination. These sequence 
databases were used for multiple alignments created with Clustal 
Omega (61). Sequence logos for multiple alignments were created using WebLogo3 using the online web server and full versions are available in the Supplementary Materials (62). HMMs were generated 
from multiple alignments using hmmbuild version 3.1b2 (hmmer.org).
The newly created HMMs were used to identify CarC homologs 
in wastewater metagenomes, and their locations were placed on the 
globalmap (Fig.  4). A protein sequence database containing full-­
length cluster-­representative proteins from wastewater was provided by MGnify from the May 2022 Protein database release (63). 
MGnify database was searched using hmmsearch version 3.1b2. A 
sequence was considered a CarC homolog if the HMM score was 
higher than the lowest scoring sequence from which the database 
was built. Metadata for each sample were fetched using MgnifyR, 
and samples without longitude and latitudes were removed (63). 
Maps were built using ggplot2 in R (64).
Supplementary Materials
This PDF file includes:
Supplementary Text
Figs. S1 to S16
Legends for data S1 to S4
References
Other Supplementary Material for this manuscript includes the following:
Data S1 to S4
REFERENCES AND NOTES
	 1.	 M. G. Evich, M. J. B. Davis, J. P. McCord, B. Acrey, J. A. Awkerman, D. R. U. Knappe,  
A. B. Lindstrom, T. F. Speth, C. Tebes-­Stevens, M. J. Strynar, Z. Wang, E. J. Weber,  
W. M. Henderson, J. W. Washington, Per-­ and polyfluoroalkyl substances in the 
environment. Science 375, eabg9065 (2022).
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 9

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
9 of 10
	 2.	 Z. Wang, A. M. Buser, I. T. Cousins, S. Demattio, W. Drost, O. Johansson, K. Ohno,  
G. Patlewicz, A. M. Richard, G. W. Walker, G. S. White, E. Leinala, A new OECD definition for 
per-­ and polyfluoroalkyl substances. Environ. Sci. Technol. 55, 15575–15578 (2021).
	 3.	 L. P. Wackett, Why is the biodegradation of polyfluorinated compounds so rare? mSphere 
6, e0072121 (2021).
	 4.	 L. P. Wackett, Nothing lasts forever: Understanding microbial biodegradation of 
polyfluorinated compounds and perfluorinated alkyl substances. Microb. Biotechnol. 15, 
773–792 (2022).
	 5.	 J. R. Parsons, M. Sáez, J. Dolfing, P. de Voogt, in Rev. Environ. Contam. Toxicol., 
D. M. Whitacre, Ed. (Springer US, New York, NY, 2008), vol. 196, pp. 53–71.
	 6.	 S. Che, B. Jin, Z. Liu, Y. Yu, J. Liu, Y. Men, Structure-­specific aerobic defluorination of 
short-­chain fluorinated carboxylic acids by activated sludge communities. Environ. Sci. 
Technol. Lett. 8, 668–674 (2021).
	 7.	 B. Jin, H. Liu, S. Che, J. Gao, Y. Yu, J. Liu, Y. Men, Substantial defluorination of 
polychlorofluorocarboxylic acids triggered by anaerobic microbial hydrolytic 
dechlorination. Nat. Water 1, 451–461 (2023).
	 8.	 J. Liu, S. Mejia Avendano, Microbial degradation of polyfluoroalkyl chemicals in the 
environment: A review. Environ. Int. 61, 98–114 (2013).
	 9.	 C. M. Butt, D. C. Muir, S. A. Mabury, Biotransformation pathways of fluorotelomer-­based 
polyfluoroalkyl substances: A review. Environ. Toxicol. Chem. 33, 243–267 (2014).
	10.	 S. Joudan, S. A. Mabury, Aerobic biotransformation of a novel highly functionalized 
polyfluoroether-­based surfactant using activated sludge from a wastewater treatment 
plant. Environ. Sci. Process. Impacts 24, 62–71 (2022).
	11.	 S.-­H. Yang, Y. Shi, M. Strynar, K.-­H. Chu, Desulfonation and defluorination of 6:2 
fluorotelomer sulfonic acid (6:2 FTSA) by Rhodococcus jostii RHA1: Carbon and sulfur 
sources, enzymes, and pathways. J. Hazard. Mater. 423, 127052 (2022).
	12.	 J. D. Van Hamme, E. M. Bottos, N. J. Bilbey, S. E. Brewer, Genomic and proteomic 
characterization of Gordonia sp. NB4-­1Y in relation to 6 : 2 fluorotelomer sulfonate 
biodegradation. Microbiology 159, 1618–1628 (2013).
	13.	 Y. Yu, S. Che, C. Ren, B. Jin, Z. Tian, J. Liu, Y. Men, Microbial defluorination of unsaturated 
per-­ and polyfluorinated carboxylic acids under anaerobic and aerobic conditions: A 
structure specificity study. Environ. Sci. Technol. 56, 4894–4904 (2022).
	14.	 S. Huang, P. R. Jaffé, Defluorination of perfluorooctanoic acid (PFOA) and perfluorooctane 
sulfonate (PFOS) by Acidimicrobium sp. strain A6. Environ. Sci. Technol. 53, 11410–11419 
(2019).
	15.	 J. Liu, E. Edwards, J. Van Hamme, M. Manefield, C. P. Higgins, J. Blotevogel, J. Liu, L. S. Lee, 
Correspondence on “Defluorination of perfluorooctanoic acid (PFOA) and 
perfluorooctane sulfonate (PFOS) by Acidimicrobium sp. strain A6”. Environ. Sci. Technol. 
57, 20440–20442 (2023).
	16.	 P. R. Jaffé, S. Huang, Rebuttal to correspondence on “Defluorination of perfluorooctanoic 
acid (PFOA) and perfluorooctane sulfonate (PFOS) by Acidimicrobium sp. strain A6”. 
Environ. Sci. Technol. 57, 20443–20447 (2023).
	17.	 Y. Yu, K. Zhang, Z. Li, C. Ren, J. Chen, Y. H. Lin, J. Liu, Y. Men, Microbial cleavage of C-­F 
bonds in two C6 per-­ and polyfluorinated compounds via reductive defluorination. 
Environ. Sci. Technol. 54, 14393–14402 (2020).
	18.	 L. A. Hug, R. G. Beiko, A. R. Rowe, R. E. Richardson, E. A. Edwards, Comparative 
metagenomics of three Dehalococcoides-­containing enrichment cultures: The role of the 
non-­dechlorinating community. BMC Genomics 13, 327 (2012).
	19.	 M. Duhamel, E. A. Edwards, Growth and yields of dechlorinators, acetogens, and 
methanogens during reductive dechlorination of chlorinated ethenes and 
dihaloelimination of 1,2-­dichloroethane. Environ. Sci. Technol. 41, 2303–2310 (2007).
	20.	 D. P. Terzenbach, M. Blaut, Transformation of tetrachloroethylene to trichloroethylene by 
homoacetogenic bacteria. FEMS Microbiol. Lett. 123, 213–218 (1994).
	21.	 C. Ding, W. L. Chow, J. He, Isolation of Acetobacterium sp. strain AG, which reductively 
debrominates octa-­ and pentabrominated diphenyl ether technical mixtures. Appl. 
Environ. Microbiol. 79, 1110–1117 (2013).
	22.	 J. Bertsch, A. Parthasarathy, W. Buckel, V. Müller, An electron-­bifurcating caffeyl-­CoA 
reductase. J. Biol. Chem. 288, 11304–11311 (2013).
	23.	 R. E. Marquis, S. A. Clock, M. Mota-­Meira, Fluoride and organic weak acids as modulators 
of microbial physiology. FEMS Microbiol. Rev. 26, 493–510 (2003).
	24.	 J. L. Baker, N. Sudarsan, Z. Weinberg, A. Roth, R. B. Stockbridge, R. R. Breaker, Widespread 
genetic switches and toxicity resistance proteins for fluoride. Science 335, 233–235 
(2012).
	25.	 A. G. Dodge, C. J. Thoma, M. R. O’Connor, L. P. Wackett, Recombinant Pseudomonas 
growing on non-­natural fluorinated substrates shows stress but overall tolerance to 
cytoplasmically released fluoride anion. mBio 15, e0278523 (2024).
	26.	 B. C. McIlwain, M. T. Ruprecht, R. B. Stockbridge, Membrane exporters of fluoride ion. 
Annu. Rev. Biochem. 90, 559–579 (2021).
	27.	 R. B. Stockbridge, J. L. Robertson, L. Kolmakova-­Partensky, C. Miller, A family of 
fluoride-­specific ion channels with dual-­topology architecture. eLife 2, e01084 
(2013).
	28.	 J. K. Demmer, J. Bertsch, C. Öppinger, H. Wohlers, K. Kayastha, U. Demmer, U. Ermler,  
V. Müller, Molecular basis of the flavin-­based electron-­bifurcating caffeyl-­CoA reductase 
reaction. FEBS Lett. 592, 332–342 (2018).
	29.	 V. Hess, S. Vitt, V. Müller, A caffeyl-­coenzyme A synthetase initiates caffeate activation 
prior to caffeate reduction in the acetogenic bacterium Acetobacterium woodii. J. 
Bacteriol. 193, 971–978 (2011).
	30.	 V. Hess, J. M. González, A. Parthasarathy, W. Buckel, V. Müller, Caffeate respiration in the 
acetogenic bacterium Acetobacterium woodii: A coenzyme A loop saves energy for 
caffeate activation. Appl. Environ. Microbiol. 79, 1942–1947 (2013).
	31.	 S. Dilling, F. Imkamp, S. Schmidt, V. Müller, Regulation of caffeate respiration in the 
acetogenic bacterium Acetobacterium woodii. Appl. Environ. Microbiol. 73, 3630–3636 
(2007).
	32.	 J. K. Demmer, N. Pal Chowdhury, T. Selmer, U. Ermler, W. Buckel, The semiquinone swing 
in the bifurcating electron transferring flavoprotein/butyryl-­CoA dehydrogenase 
complex from Clostridium difficile. Nat. Commun. 8, 1577 (2017).
	33.	 W. Buckel, R. K. Thauer, Flavin-­based electron bifurcation, a new mechanism of biological 
energy coupling. Chem. Rev. 118, 3862–3886 (2018).
	34.	 M. H. Beck, A. Poehlein, F. R. Bengelsdorf, B. Schiel-­Bengelsdorf, R. Daniel, P. Durre, Draft 
genome sequence of the strict anaerobe Clostridium homopropionicum LuHBu1 (DSM 
5847). Genome Announc. 3, e01112–e01115 (2015).
	35.	 M. Hetzel, M. Brock, T. Selmer, A. J. Pierik, B. T. Golding, W. Buckel, Acryloyl-­CoA reductase 
from Clostridium propionicum. An enzyme complex of propionyl-­CoA dehydrogenase 
and electron-­transferring flavoprotein. Eur. J. Biochem. 270, 902–910 (2003).
	36.	 E. Munier, H. Licandro, E. Beuvier, R. Cachon, Bioinformatics and metabolic flux analysis 
highlight a new mechanism involved in lactate oxidation in Clostridium tyrobutyricum. 
Int. Microbiol. 26, 501–511 (2023).
	37.	 N. R. Adrian, C. M. Arnett, Anaerobic biodegradation of hexahydro-­1,3,5-­trinitro-­1,3,5-­
triazine (RDX) by Acetobacterium malicum strain HAAP-­1 isolated from a methanogenic 
mixed culture. Curr. Microbiol. 48, 332–340 (2004).
	38.	 R. Bache, N. Pfennig, Selective isolation of Acetobacterium woodii on methoxylated 
aromatic acids and determination of growth yields. Arch. Microbiol. 130, 255–261 (1981).
	39.	 W.-­M. Chen, Z.-­J. Tseng, K.-­S. Lee, J.-­S. Chang, Fermentative hydrogen production with 
Clostridium butyricum CGS5 isolated from anaerobic sewage sludge. Int. J. Hydrogen 
Energy. 30, 1063–1070 (2005).
	40.	 T. Mechichi, M. Labat, T. H. Woo, P. Thomas, J. L. Garcia, B. K. Patel, Eubacterium aggregans 
sp. nov., a new homoacetogenic bacterium from olive mill wastewater treatment 
digestor. Anaerobe 4, 283–291 (1998).
	41.	 C. C. Wang, C. W. Chang, C. P. Chu, D. J. Lee, B. V. Chang, C. S. Liao, Producing hydrogen 
from wastewater sludge by Clostridium bifermentans. J. Biotechnol. 102, 83–92 (2003).
	42.	 B. D. Key, R. D. Howell, C. S. Criddle, Fluorinated organics in the biosphere. Environ. Sci. 
Technol. 31, 2445–2454 (1997).
	43.	 C. F. Kwiatkowski, D. Q. Andrews, L. S. Birnbaum, T. A. Bruton, J. C. DeWitt, D. R. U. Knappe, 
M. V. Maffini, M. F. Miller, K. E. Pelch, A. Reade, A. Soehl, X. Trier, M. Venier, C. C. Wagner,  
Z. Wang, A. Blum, Scientific basis for managing PFAS as a chemical class. Environ. Sci. 
Technol. Lett. 7, 532–543 (2020).
	44.	 Z. Wang, J. C. DeWitt, C. P. Higgins, I. T. Cousins, A never-­ending story of per-­ and 
polyfluoroalkyl substances (PFASs)? Environ. Sci. Technol. 51, 2508–2518 (2017).
	45.	 J. Glüge, M. Scheringer, I. T. Cousins, J. C. DeWitt, G. Goldenman, D. Herzke, R. Lohmann,  
C. A. Ng, X. Trier, Z. Wang, An overview of the uses of per-­ and polyfluoroalkyl substances 
(PFAS). Environ. Sci. Process. Impacts 22, 2345–2373 (2020).
	46.	 A. Wiseman, in Pharmacology of Fluorides: Part 2, F. A. Smith, Ed. (Springer Berlin 
Heidelberg, 1970), pp. 48–97.
	47.	 C. J. Allègre, J.-­P. Poirier, E. Humler, A. W. Hofmann, The chemical composition of the 
Earth. Earth Planet. Sci. Lett. 134, 515–526 (1995).
	48.	 M. Amini, K. Mueller, K. C. Abbaspour, T. Rosenberg, M. Afyuni, K. N. Møller, M. Sarr,  
C. A. Johnson, Statistical modeling of global geogenic fluoride contamination in 
groundwaters. Environ. Sci. Technol. 42, 3662–3668 (2008).
	49.	 R. B. Stockbridge, H. H. Lim, R. Otten, C. Williams, T. Shane, Z. Weinberg, C. Miller, Fluoride 
resistance and transport by riboswitch-­controlled CLC antiporters. Proc. Natl. Acad. Sci. 
U.S.A. 109, 15289–15294 (2012).
	50.	 K. A. Payne, C. P. Quezada, K. Fisher, M. S. Dunstan, F. A. Collins, H. Sjuts, C. Levy, S. Hay,  
S. E. Rigby, D. Leys, Reductive dehalogenase structure suggests a mechanism for 
B12-­dependent dehalogenation. Nature 517, 513–516 (2015).
	51.	 W. Buckel, R. K. Thauer, Flavin-­based electron bifurcation, ferredoxin, flavodoxin, and 
anaerobic respiration with protons (Ech) or NAD+ (Rnf) as electron acceptors: A historical 
review. Front. Microbiol. 9, 401 (2018).
	52.	 R. Rabus, M. Boll, J. Heider, R. U. Meckenstock, W. Buckel, O. Einsle, U. Ermler, B. T. Golding, 
R. P. Gunsalus, P. M. H. Kroneck, M. Krüger, T. Lueders, B. M. Martins, F. Musat,  
H. H. Richnow, B. Schink, J. Seifert, M. Szaleniec, T. Treude, G. M. Ullmann, C. Vogt,  
M. von Bergen, H. Wilkes, Anaerobic microbial degradation of hydrocarbons: From 
enzymatic reactions to the environment. J. Mol. Microbiol. Biotechnol. 26, 5–28 (2016).
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

## Page 10

Yu et al., Sci. Adv. 10, eado2957 (2024)     17 July 2024
S c i e n c e  A d va n c e s  | R e s e a r c h  A r t i c l e
10 of 10
	53.	 G. J. Schut, D. K. Haja, X. Feng, F. L. Poole, H. Li, M. W. W. Adams, An abundant and diverse 
new family of electron bifurcating enzymes with a non-­canonical catalytic mechanism. 
Front. Microbiol. 13, 946711 (2022).
	54.	 J. P. Baker, J. Saez-­Saez, S. I. Jensen, A. T. Nielsen, N. P. Minton, A clean in-­frame knockout 
system for gene deletion in Acetobacterium woodii. J. Biotechnol. 353, 9–18 (2022).
	55.	 O. Tiedt, M. Mergelsberg, K. Boll, M. Müller, L. Adrian, N. Jehmlich, M. von Bergen, M. Boll, 
ATP-­dependent C–F bond cleavage allows the complete degradation of 
4-­fluoroaromatics without oxygen. mBio 7, e00990-­16 (2016).
	56.	 S. E. L. Anselmann, C. Löffler, H.-­J. Stärk, N. Jehmlich, M. von Bergen, T. Brüls, M. Boll, The 
class II benzoyl-­coenzyme A reductase complex from the sulfate-­reducing Desulfosarcina 
cetonica. Environ. Microbiol. 21, 4241–4252 (2019).
	57.	 P. Emsley, B. Lohkamp, W. G. Scott, K. Cowtan, Features and development of Coot. Acta 
Crystallogr. D Biol. Crystallogr. 66, 486–501 (2010).
	58.	 O. S. Smart, A. Sharff, J. Holstein, T. O. Womack, C. Flensburg, P. Keller, W. Paciorek, 
C. Vonrhein, G. Bricogne, Grade2 version 1.4.0. (Global Phasing Ltd., 2021).
	59.	 A. A. Vagin, R. A. Steiner, A. A. Lebedev, L. Potterton, S. McNicholas, F. Long,  
G. N. Murshudov, REFMAC5 dictionary: Organization of prior chemical knowledge and 
guidelines for its use. Acta Crystallogr. D Biol. Crystallogr. 60, 2184–2195 (2004).
	60.	 P. Koehl, M. Levitt, Sequence variations within protein families are linearly related to 
structural variations. J. Mol. Biol. 323, 551–562 (2002).
	61.	 F. Sievers, A. Wilm, D. Dineen, T. J. Gibson, K. Karplus, W. Li, R. Lopez, H. McWilliam,  
M. Remmert, J. Soding, J. D. Thompson, D. G. Higgins, Fast, scalable generation of 
high-­quality protein multiple sequence alignments using Clustal Omega. Mol. Syst. Biol. 
7, 539 (2011).
	62.	 G. E. Crooks, G. Hon, J. M. Chandonia, S. E. Brenner, WebLogo: A sequence logo generator. 
Genome Res. 14, 1188–1190 (2004).
	63.	 L. Richardson, B. Allen, G. Baldi, M. Beracochea, M. L. Bileschi, T. Burdett, J. Burgin,  
J. Caballero-­Perez, G. Cochrane, L. J. Colwell, T. Curtis, A. Escobar-­Zepeda, T. A. Gurbich,  
V. Kale, A. Korobeynikov, S. Raj, A. B. Rogers, E. Sakharova, S. Sanchez, D. J. Wilkinson,  
R. D. Finn, MGnify: The microbiome sequence data analysis resource in 2023. Nucleic Acids 
Res. 51, D753–D759 (2023).
	64.	 H. Wickham, ggplot2: Elegant Graphics for Data Analysis (Springer-­Verlag, 2016).
	65.	 F. R. Blattner, G. Plunkett III, C. A. Bloch, N. T. Perna, V. Burland, M. Riley, J. Collado-­Vides,  
J. D. Glasner, C. K. Rode, G. F. Mayhew, J. Gregor, N. W. Davis, H. A. Kirkpatrick,  
M. A. Goeden, D. J. Rose, B. Mau, Y. Shao, The complete genome sequence of Escherichia 
coli K-­12. Science 277, 1453–1462 (1997).
	66.	 K. A. Datsenko, B. L. Wanner, One-­step inactivation of chromosomal genes in Escherichia 
coli K-­12 using PCR products. Proc. Natl. Acad. Sci. U. S. A. 97, 6640–6645 (2000).
	67.	 K. C. Murphy, K. G. Campellone, Lambda Red-­mediated recombinogenic engineering of 
enterohemorrhagic and enteropathogenic E. coli. BMC Mol. Biol. 4, 11 (2003).
	68.	 F. Malagon, RNase III is required for localization to the nucleoid of the 5' pre-­rRNA leader 
and for optimal induction of rRNA synthesis in E. coli. RNA 19, 1200–1207 (2013).
	69.	 Y. Men, E. C. Seth, S. Yi, R. H. Allen, M. E. Taga, L. Alvarez-­Cohen, Sustainable growth of 
Dehalococcoides mccartyi 195 by corrinoid salvaging and remodeling in defined 
lactate-­fermenting consortia. Appl. Environ. Microbiol. 80, 2133–2141 (2014).
	70.	 A. P. Arkin, R. W. Cottingham, C. S. Henry, N. L. Harris, R. L. Stevens, S. Maslov, P. Dehal,  
D. Ware, F. Perez, S. Canon, M. W. Sneddon, M. L. Henderson, W. J. Riehl, D. Murphy-­Olson, 
S. Y. Chan, R. T. Kamimura, S. Kumari, M. M. Drake, T. S. Brettin, E. M. Glass, D. Chivian,  
D. Gunter, D. J. Weston, B. H. Allen, J. Baumohl, A. A. Best, B. Bowen, S. E. Brenner,  
C. C. Bun, J. M. Chandonia, J. M. Chia, R. Colasanti, N. Conrad, J. J. Davis, B. H. Davison,  
M. DeJongh, S. Devoid, E. Dietrich, I. Dubchak, J. N. Edirisinghe, G. Fang, J. P. Faria,  
P. M. Frybarger, W. Gerlach, M. Gerstein, A. Greiner, J. Gurtowski, H. L. Haun, F. He, R. Jain, 
M. P. Joachimiak, K. P. Keegan, S. Kondo, V. Kumar, M. L. Land, F. Meyer, M. Mills,  
P. S. Novichkov, T. Oh, G. J. Olsen, R. Olson, B. Parrello, S. Pasternak, E. Pearson, S. S. Poon, 
G. A. Price, S. Ramakrishnan, P. Ranjan, P. C. Ronald, M. C. Schatz, S. M. D. Seaver,  
M. Shukla, R. A. Sutormin, M. H. Syed, J. Thomason, N. L. Tintle, D. Wang, F. Xia, H. Yoo,  
S. Yoo, D. Yu, KBase: The United States Department of Energy Systems Biology 
Knowledgebase. Nat. Biotechnol. 36, 566–569 (2018).
	71.	 D. Kim, J. M. Paggi, C. Park, C. Bennett, S. L. Salzberg, Graph-­based genome alignment 
and genotyping with HISAT2 and HISAT-­genotype. Nat. Biotechnol. 37, 907–915 
(2019).
	72.	 M. Pertea, G. M. Pertea, C. M. Antonescu, T. C. Chang, J. T. Mendell, S. L. Salzberg, StringTie 
enables improved reconstruction of a transcriptome from RNA-­seq reads. Nat. Biotechnol. 
33, 290–295 (2015).
	73.	 M. Pertea, D. Kim, G. M. Pertea, J. T. Leek, S. L. Salzberg, Transcript-­level expression 
analysis of RNA-­seq experiments with HISAT, StringTie and Ballgown. Nat. Protoc. 11, 
1650–1667 (2016).
	74.	 M. I. Love, W. Huber, S. Anders, Moderated estimation of fold change and dispersion for 
RNA-­seq data with DESeq2. Genome Biol. 15, 550 (2014).
	75.	 A. Wiechmann, S. Ciurus, F. Oswald, V. N. Seiler, V. Müller, It does not always take two to 
tango: “Syntrophy” via hydrogen cycling in one bacterial cell. ISME J. 14, 1561–1570 
(2020).
Acknowledgments 
Funding: This study is supported by the Strategic Environmental Research and Development 
Program (ER20-­1541 for Y.Y., B.J., W.Z., and Y.M.), the National Science Foundation (award no. 
1931941 for S.C.), and the National Institute of Environmental Health Sciences (award no. 
R01ES032668 for S.C. and Y.M.), and partly by a grant from MnDRIVE Industry and the 
Environment Program (to L.W.). Author contributions: Conceptualization: Y.Y. and Y.M. 
Methodology: Y.Y., F.X., C.T., J.E.R., and Y.X. Software: C.T. Formal analysis: Y.Y., C.T., L.W., and Y.M. 
Investigation: Y.Y., F.X., W.Z., C.T., S.C., J.E.R., B.J., and Y.Z. Data curation: Y.Y., C.T., L.W., and Y.M. 
Writing—original draft: Y.Y. and Y.M. Writing—review and editing: Y.Y., F.X., W.Z., C.T., S.C., J.E.R., 
B.J., Y.Z., Y.X., L.W., and Y.M. Visualization: Y.Y., C.T., J.E.R., Y.X., L.W., and Y.M. Supervision: L.W. and 
Y.M. Funding acquisition: L.W. and Y.M. Competing interests: The authors declare that they 
have no competing interests. Data and materials availability: All other data needed to 
evaluate the conclusions in the paper are present in the paper and/or the Supplementary 
Materials. The sequencing raw reads of transcriptomes at two time points were deposited to 
the National Center for Biotechnology Information (NCBI) SRA database under accession no. 
PRJNA1029002 (20 days) and PRJNA1029168 (90 min).
Submitted 26 January 2024 
Accepted 12 June 2024 
Published 17 July 2024 
10.1126/sciadv.ado2957
Downloaded from https://www.science.org at University of California Berkeley on November 18, 2025

